{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"name":"Result.ipynb","provenance":[],"collapsed_sections":[]},"kernelspec":{"name":"python3","display_name":"Python 3"},"accelerator":"GPU","widgets":{"application/vnd.jupyter.widget-state+json":{"c666525d535e4626b8e55900c77e677b":{"model_module":"@jupyter-widgets/controls","model_name":"HBoxModel","state":{"_view_name":"HBoxView","_dom_classes":[],"_model_name":"HBoxModel","_view_module":"@jupyter-widgets/controls","_model_module_version":"1.5.0","_view_count":null,"_view_module_version":"1.5.0","box_style":"","layout":"IPY_MODEL_80be2dfba24641a6be4ebc43c33554d6","_model_module":"@jupyter-widgets/controls","children":["IPY_MODEL_f182eb364fca48d59d0cdc609a54a5b1","IPY_MODEL_43b0937d8fa643f78490f55a387a6677"]}},"80be2dfba24641a6be4ebc43c33554d6":{"model_module":"@jupyter-widgets/base","model_name":"LayoutModel","state":{"_view_name":"LayoutView","grid_template_rows":null,"right":null,"justify_content":null,"_view_module":"@jupyter-widgets/base","overflow":null,"_model_module_version":"1.2.0","_view_count":null,"flex_flow":null,"width":null,"min_width":null,"border":null,"align_items":null,"bottom":null,"_model_module":"@jupyter-widgets/base","top":null,"grid_column":null,"overflow_y":null,"overflow_x":null,"grid_auto_flow":null,"grid_area":null,"grid_template_columns":null,"flex":null,"_model_name":"LayoutModel","justify_items":null,"grid_row":null,"max_height":null,"align_content":null,"visibility":null,"align_self":null,"height":null,"min_height":null,"padding":null,"grid_auto_rows":null,"grid_gap":null,"max_width":null,"order":null,"_view_module_version":"1.2.0","grid_template_areas":null,"object_position":null,"object_fit":null,"grid_auto_columns":null,"margin":null,"display":null,"left":null}},"f182eb364fca48d59d0cdc609a54a5b1":{"model_module":"@jupyter-widgets/controls","model_name":"FloatProgressModel","state":{"_view_name":"ProgressView","style":"IPY_MODEL_afc82981315f4ab89e515075ed3c3c91","_dom_classes":[],"description":"Downloading: 100%","_model_name":"FloatProgressModel","bar_style":"success","max":440473133,"_view_module":"@jupyter-widgets/controls","_model_module_version":"1.5.0","value":440473133,"_view_count":null,"_view_module_version":"1.5.0","orientation":"horizontal","min":0,"description_tooltip":null,"_model_module":"@jupyter-widgets/controls","layout":"IPY_MODEL_c9f5f1994985410c8eed0f9406423f14"}},"43b0937d8fa643f78490f55a387a6677":{"model_module":"@jupyter-widgets/controls","model_name":"HTMLModel","state":{"_view_name":"HTMLView","style":"IPY_MODEL_5acfe71daf5d42b093ccc963b6d65b94","_dom_classes":[],"description":"","_model_name":"HTMLModel","placeholder":"â€‹","_view_module":"@jupyter-widgets/controls","_model_module_version":"1.5.0","value":" 440M/440M [00:06&lt;00:00, 67.8MB/s]","_view_count":null,"_view_module_version":"1.5.0","description_tooltip":null,"_model_module":"@jupyter-widgets/controls","layout":"IPY_MODEL_482ce836853e40d3a3836257a5cc7293"}},"afc82981315f4ab89e515075ed3c3c91":{"model_module":"@jupyter-widgets/controls","model_name":"ProgressStyleModel","state":{"_view_name":"StyleView","_model_name":"ProgressStyleModel","description_width":"initial","_view_module":"@jupyter-widgets/base","_model_module_version":"1.5.0","_view_count":null,"_view_module_version":"1.2.0","bar_color":null,"_model_module":"@jupyter-widgets/controls"}},"c9f5f1994985410c8eed0f9406423f14":{"model_module":"@jupyter-widgets/base","model_name":"LayoutModel","state":{"_view_name":"LayoutView","grid_template_rows":null,"right":null,"justify_content":null,"_view_module":"@jupyter-widgets/base","overflow":null,"_model_module_version":"1.2.0","_view_count":null,"flex_flow":null,"width":null,"min_width":null,"border":null,"align_items":null,"bottom":null,"_model_module":"@jupyter-widgets/base","top":null,"grid_column":null,"overflow_y":null,"overflow_x":null,"grid_auto_flow":null,"grid_area":null,"grid_template_columns":null,"flex":null,"_model_name":"LayoutModel","justify_items":null,"grid_row":null,"max_height":null,"align_content":null,"visibility":null,"align_self":null,"height":null,"min_height":null,"padding":null,"grid_auto_rows":null,"grid_gap":null,"max_width":null,"order":null,"_view_module_version":"1.2.0","grid_template_areas":null,"object_position":null,"object_fit":null,"grid_auto_columns":null,"margin":null,"display":null,"left":null}},"5acfe71daf5d42b093ccc963b6d65b94":{"model_module":"@jupyter-widgets/controls","model_name":"DescriptionStyleModel","state":{"_view_name":"StyleView","_model_name":"DescriptionStyleModel","description_width":"","_view_module":"@jupyter-widgets/base","_model_module_version":"1.5.0","_view_count":null,"_view_module_version":"1.2.0","_model_module":"@jupyter-widgets/controls"}},"482ce836853e40d3a3836257a5cc7293":{"model_module":"@jupyter-widgets/base","model_name":"LayoutModel","state":{"_view_name":"LayoutView","grid_template_rows":null,"right":null,"justify_content":null,"_view_module":"@jupyter-widgets/base","overflow":null,"_model_module_version":"1.2.0","_view_count":null,"flex_flow":null,"width":null,"min_width":null,"border":null,"align_items":null,"bottom":null,"_model_module":"@jupyter-widgets/base","top":null,"grid_column":null,"overflow_y":null,"overflow_x":null,"grid_auto_flow":null,"grid_area":null,"grid_template_columns":null,"flex":null,"_model_name":"LayoutModel","justify_items":null,"grid_row":null,"max_height":null,"align_content":null,"visibility":null,"align_self":null,"height":null,"min_height":null,"padding":null,"grid_auto_rows":null,"grid_gap":null,"max_width":null,"order":null,"_view_module_version":"1.2.0","grid_template_areas":null,"object_position":null,"object_fit":null,"grid_auto_columns":null,"margin":null,"display":null,"left":null}}}}},"cells":[{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"UIrdzSGcAgJB","executionInfo":{"status":"ok","timestamp":1608027913025,"user_tz":-330,"elapsed":12437,"user":{"displayName":"Akash Gangwar","photoUrl":"","userId":"02081727011117790241"}},"outputId":"68df9540-0c48-4aa0-e44e-a7dd1b5fb0c6"},"source":["!pip install transformers==3.5.0\r\n","!pip install nltk\r\n","!pip install biobert-pytorch\r\n","!pip install wget\r\n","!pip install cookiecutter"],"execution_count":null,"outputs":[{"output_type":"stream","text":["Requirement already satisfied: transformers==3.5.0 in /usr/local/lib/python3.6/dist-packages (3.5.0)\n","Requirement already satisfied: numpy in /usr/local/lib/python3.6/dist-packages (from transformers==3.5.0) (1.18.5)\n","Requirement already satisfied: sentencepiece==0.1.91 in /usr/local/lib/python3.6/dist-packages (from transformers==3.5.0) (0.1.91)\n","Requirement already satisfied: tokenizers==0.9.3 in /usr/local/lib/python3.6/dist-packages (from transformers==3.5.0) (0.9.3)\n","Requirement already satisfied: protobuf in /usr/local/lib/python3.6/dist-packages (from transformers==3.5.0) (3.12.4)\n","Requirement already satisfied: filelock in /usr/local/lib/python3.6/dist-packages (from transformers==3.5.0) (3.0.12)\n","Requirement already satisfied: sacremoses in /usr/local/lib/python3.6/dist-packages (from transformers==3.5.0) (0.0.43)\n","Requirement already satisfied: requests in /usr/local/lib/python3.6/dist-packages (from transformers==3.5.0) (2.23.0)\n","Requirement already satisfied: packaging in /usr/local/lib/python3.6/dist-packages (from transformers==3.5.0) (20.7)\n","Requirement already satisfied: tqdm>=4.27 in /usr/local/lib/python3.6/dist-packages (from transformers==3.5.0) (4.41.1)\n","Requirement already satisfied: regex!=2019.12.17 in /usr/local/lib/python3.6/dist-packages (from transformers==3.5.0) (2019.12.20)\n","Requirement already satisfied: dataclasses; python_version < \"3.7\" in /usr/local/lib/python3.6/dist-packages (from transformers==3.5.0) (0.8)\n","Requirement already satisfied: six>=1.9 in /usr/local/lib/python3.6/dist-packages (from protobuf->transformers==3.5.0) (1.15.0)\n","Requirement already satisfied: setuptools in /usr/local/lib/python3.6/dist-packages (from protobuf->transformers==3.5.0) (50.3.2)\n","Requirement already satisfied: joblib in /usr/local/lib/python3.6/dist-packages (from sacremoses->transformers==3.5.0) (0.17.0)\n","Requirement already satisfied: click in /usr/local/lib/python3.6/dist-packages (from sacremoses->transformers==3.5.0) (7.1.2)\n","Requirement already satisfied: chardet<4,>=3.0.2 in /usr/local/lib/python3.6/dist-packages (from requests->transformers==3.5.0) (3.0.4)\n","Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /usr/local/lib/python3.6/dist-packages (from requests->transformers==3.5.0) (1.24.3)\n","Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.6/dist-packages (from requests->transformers==3.5.0) (2.10)\n","Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.6/dist-packages (from requests->transformers==3.5.0) (2020.12.5)\n","Requirement already satisfied: pyparsing>=2.0.2 in /usr/local/lib/python3.6/dist-packages (from packaging->transformers==3.5.0) (2.4.7)\n","Requirement already satisfied: nltk in /usr/local/lib/python3.6/dist-packages (3.2.5)\n","Requirement already satisfied: six in /usr/local/lib/python3.6/dist-packages (from nltk) (1.15.0)\n","Requirement already satisfied: biobert-pytorch in /usr/local/lib/python3.6/dist-packages (0.9)\n","Requirement already satisfied: wget in /usr/local/lib/python3.6/dist-packages (3.2)\n","Requirement already satisfied: cookiecutter in /usr/local/lib/python3.6/dist-packages (1.7.2)\n","Requirement already satisfied: Jinja2<3.0.0 in /usr/local/lib/python3.6/dist-packages (from cookiecutter) (2.11.2)\n","Requirement already satisfied: binaryornot>=0.4.4 in /usr/local/lib/python3.6/dist-packages (from cookiecutter) (0.4.4)\n","Requirement already satisfied: requests>=2.23.0 in /usr/local/lib/python3.6/dist-packages (from cookiecutter) (2.23.0)\n","Requirement already satisfied: MarkupSafe<2.0.0 in /usr/local/lib/python3.6/dist-packages (from cookiecutter) (1.1.1)\n","Requirement already satisfied: six>=1.10 in /usr/local/lib/python3.6/dist-packages (from cookiecutter) (1.15.0)\n","Requirement already satisfied: poyo>=0.5.0 in /usr/local/lib/python3.6/dist-packages (from cookiecutter) (0.5.0)\n","Requirement already satisfied: python-slugify>=4.0.0 in /usr/local/lib/python3.6/dist-packages (from cookiecutter) (4.0.1)\n","Requirement already satisfied: click>=7.0 in /usr/local/lib/python3.6/dist-packages (from cookiecutter) (7.1.2)\n","Requirement already satisfied: jinja2-time>=0.2.0 in /usr/local/lib/python3.6/dist-packages (from cookiecutter) (0.2.0)\n","Requirement already satisfied: chardet>=3.0.2 in /usr/local/lib/python3.6/dist-packages (from binaryornot>=0.4.4->cookiecutter) (3.0.4)\n","Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.6/dist-packages (from requests>=2.23.0->cookiecutter) (2020.12.5)\n","Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /usr/local/lib/python3.6/dist-packages (from requests>=2.23.0->cookiecutter) (1.24.3)\n","Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.6/dist-packages (from requests>=2.23.0->cookiecutter) (2.10)\n","Requirement already satisfied: text-unidecode>=1.3 in /usr/local/lib/python3.6/dist-packages (from python-slugify>=4.0.0->cookiecutter) (1.3)\n","Requirement already satisfied: arrow in /usr/local/lib/python3.6/dist-packages (from jinja2-time>=0.2.0->cookiecutter) (0.17.0)\n","Requirement already satisfied: python-dateutil>=2.7.0 in /usr/local/lib/python3.6/dist-packages (from arrow->jinja2-time>=0.2.0->cookiecutter) (2.8.1)\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"W4JNdJmaAoP4","executionInfo":{"status":"ok","timestamp":1608027929859,"user_tz":-330,"elapsed":29254,"user":{"displayName":"Akash Gangwar","photoUrl":"","userId":"02081727011117790241"}},"outputId":"460ec92d-017e-4cb3-9f80-cbe68af3474f"},"source":["!wget --load-cookies /tmp/cookies.txt \"https://docs.google.com/uc?export=download&confirm=$(wget --quiet --save-cookies /tmp/cookies.txt --keep-session-cookies --no-check-certificate 'https://docs.google.com/uc?export=download&id=1R84voFKHfWV9xjzeLzWBbmY1uOMYpnyD' -O- | sed -rn 's/.*confirm=([0-9A-Za-z_]+).*/\\1\\n/p')&id=1R84voFKHfWV9xjzeLzWBbmY1uOMYpnyD\" -O biobert_weights && rm -rf /tmp/cookies.txt\r\n","\r\n","!tar -xzf biobert_weights\r\n","\r\n","!transformers-cli convert --model_type bert --tf_checkpoint biobert_v1.1_pubmed/model.ckpt-1000000 --config biobert_v1.1_pubmed/bert_config.json --pytorch_dump_output biobert_v1.1_pubmed/pytorch_model.bin\r\n","\r\n","!mv biobert_v1.1_pubmed/bert_config.json biobert_v1.1_pubmed/config.json"],"execution_count":null,"outputs":[{"output_type":"stream","text":["--2020-12-15 10:25:12--  https://docs.google.com/uc?export=download&confirm=CHdY&id=1R84voFKHfWV9xjzeLzWBbmY1uOMYpnyD\n","Resolving docs.google.com (docs.google.com)... 74.125.142.113, 74.125.142.101, 74.125.142.100, ...\n","Connecting to docs.google.com (docs.google.com)|74.125.142.113|:443... connected.\n","HTTP request sent, awaiting response... 302 Moved Temporarily\n","Location: https://doc-14-9k-docs.googleusercontent.com/docs/securesc/prbajjq90jv86var83iu444vhsc8dgeu/f2p3h1hcf5sh15cf3u5r3ortd0mplq7d/1608027900000/13799006341648886493/00568481803443554857Z/1R84voFKHfWV9xjzeLzWBbmY1uOMYpnyD?e=download [following]\n","--2020-12-15 10:25:12--  https://doc-14-9k-docs.googleusercontent.com/docs/securesc/prbajjq90jv86var83iu444vhsc8dgeu/f2p3h1hcf5sh15cf3u5r3ortd0mplq7d/1608027900000/13799006341648886493/00568481803443554857Z/1R84voFKHfWV9xjzeLzWBbmY1uOMYpnyD?e=download\n","Resolving doc-14-9k-docs.googleusercontent.com (doc-14-9k-docs.googleusercontent.com)... 74.125.142.132, 2607:f8b0:400e:c08::84\n","Connecting to doc-14-9k-docs.googleusercontent.com (doc-14-9k-docs.googleusercontent.com)|74.125.142.132|:443... connected.\n","HTTP request sent, awaiting response... 302 Found\n","Location: https://docs.google.com/nonceSigner?nonce=r6aqlp5ot7kl4&continue=https://doc-14-9k-docs.googleusercontent.com/docs/securesc/prbajjq90jv86var83iu444vhsc8dgeu/f2p3h1hcf5sh15cf3u5r3ortd0mplq7d/1608027900000/13799006341648886493/00568481803443554857Z/1R84voFKHfWV9xjzeLzWBbmY1uOMYpnyD?e%3Ddownload&hash=e5ujuqca6df70bipb8l89i7f80jskjbn [following]\n","--2020-12-15 10:25:12--  https://docs.google.com/nonceSigner?nonce=r6aqlp5ot7kl4&continue=https://doc-14-9k-docs.googleusercontent.com/docs/securesc/prbajjq90jv86var83iu444vhsc8dgeu/f2p3h1hcf5sh15cf3u5r3ortd0mplq7d/1608027900000/13799006341648886493/00568481803443554857Z/1R84voFKHfWV9xjzeLzWBbmY1uOMYpnyD?e%3Ddownload&hash=e5ujuqca6df70bipb8l89i7f80jskjbn\n","Connecting to docs.google.com (docs.google.com)|74.125.142.113|:443... connected.\n","HTTP request sent, awaiting response... 302 Found\n","Location: https://doc-14-9k-docs.googleusercontent.com/docs/securesc/prbajjq90jv86var83iu444vhsc8dgeu/f2p3h1hcf5sh15cf3u5r3ortd0mplq7d/1608027900000/13799006341648886493/00568481803443554857Z/1R84voFKHfWV9xjzeLzWBbmY1uOMYpnyD?e=download&nonce=r6aqlp5ot7kl4&user=00568481803443554857Z&hash=piqiiu3gkcjhkk71856ng32ccviju923 [following]\n","--2020-12-15 10:25:12--  https://doc-14-9k-docs.googleusercontent.com/docs/securesc/prbajjq90jv86var83iu444vhsc8dgeu/f2p3h1hcf5sh15cf3u5r3ortd0mplq7d/1608027900000/13799006341648886493/00568481803443554857Z/1R84voFKHfWV9xjzeLzWBbmY1uOMYpnyD?e=download&nonce=r6aqlp5ot7kl4&user=00568481803443554857Z&hash=piqiiu3gkcjhkk71856ng32ccviju923\n","Connecting to doc-14-9k-docs.googleusercontent.com (doc-14-9k-docs.googleusercontent.com)|74.125.142.132|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: unspecified [application/x-gzip]\n","Saving to: â€˜biobert_weightsâ€™\n","\n","biobert_weights         [        <=>         ] 382.81M   261MB/s    in 1.5s    \n","\n","2020-12-15 10:25:14 (261 MB/s) - â€˜biobert_weightsâ€™ saved [401403346]\n","\n","2020-12-15 10:25:21.128531: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1\n","Building PyTorch model from configuration: BertConfig {\n","  \"attention_probs_dropout_prob\": 0.1,\n","  \"gradient_checkpointing\": false,\n","  \"hidden_act\": \"gelu\",\n","  \"hidden_dropout_prob\": 0.1,\n","  \"hidden_size\": 768,\n","  \"initializer_range\": 0.02,\n","  \"intermediate_size\": 3072,\n","  \"layer_norm_eps\": 1e-12,\n","  \"max_position_embeddings\": 512,\n","  \"model_type\": \"bert\",\n","  \"num_attention_heads\": 12,\n","  \"num_hidden_layers\": 12,\n","  \"pad_token_id\": 0,\n","  \"type_vocab_size\": 2,\n","  \"vocab_size\": 28996\n","}\n","\n","Converting TensorFlow checkpoint from /content/biobert_v1.1_pubmed/model.ckpt-1000000\n","Loading TF weight bert/embeddings/LayerNorm/beta with shape [768]\n","Loading TF weight bert/embeddings/LayerNorm/gamma with shape [768]\n","Loading TF weight bert/embeddings/position_embeddings with shape [512, 768]\n","Loading TF weight bert/embeddings/token_type_embeddings with shape [2, 768]\n","Loading TF weight bert/embeddings/word_embeddings with shape [28996, 768]\n","Loading TF weight bert/encoder/layer_0/attention/output/LayerNorm/beta with shape [768]\n","Loading TF weight bert/encoder/layer_0/attention/output/LayerNorm/gamma with shape [768]\n","Loading TF weight bert/encoder/layer_0/attention/output/dense/bias with shape [768]\n","Loading TF weight bert/encoder/layer_0/attention/output/dense/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_0/attention/self/key/bias with shape [768]\n","Loading TF weight bert/encoder/layer_0/attention/self/key/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_0/attention/self/query/bias with shape [768]\n","Loading TF weight bert/encoder/layer_0/attention/self/query/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_0/attention/self/value/bias with shape [768]\n","Loading TF weight bert/encoder/layer_0/attention/self/value/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_0/intermediate/dense/bias with shape [3072]\n","Loading TF weight bert/encoder/layer_0/intermediate/dense/kernel with shape [768, 3072]\n","Loading TF weight bert/encoder/layer_0/output/LayerNorm/beta with shape [768]\n","Loading TF weight bert/encoder/layer_0/output/LayerNorm/gamma with shape [768]\n","Loading TF weight bert/encoder/layer_0/output/dense/bias with shape [768]\n","Loading TF weight bert/encoder/layer_0/output/dense/kernel with shape [3072, 768]\n","Loading TF weight bert/encoder/layer_1/attention/output/LayerNorm/beta with shape [768]\n","Loading TF weight bert/encoder/layer_1/attention/output/LayerNorm/gamma with shape [768]\n","Loading TF weight bert/encoder/layer_1/attention/output/dense/bias with shape [768]\n","Loading TF weight bert/encoder/layer_1/attention/output/dense/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_1/attention/self/key/bias with shape [768]\n","Loading TF weight bert/encoder/layer_1/attention/self/key/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_1/attention/self/query/bias with shape [768]\n","Loading TF weight bert/encoder/layer_1/attention/self/query/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_1/attention/self/value/bias with shape [768]\n","Loading TF weight bert/encoder/layer_1/attention/self/value/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_1/intermediate/dense/bias with shape [3072]\n","Loading TF weight bert/encoder/layer_1/intermediate/dense/kernel with shape [768, 3072]\n","Loading TF weight bert/encoder/layer_1/output/LayerNorm/beta with shape [768]\n","Loading TF weight bert/encoder/layer_1/output/LayerNorm/gamma with shape [768]\n","Loading TF weight bert/encoder/layer_1/output/dense/bias with shape [768]\n","Loading TF weight bert/encoder/layer_1/output/dense/kernel with shape [3072, 768]\n","Loading TF weight bert/encoder/layer_10/attention/output/LayerNorm/beta with shape [768]\n","Loading TF weight bert/encoder/layer_10/attention/output/LayerNorm/gamma with shape [768]\n","Loading TF weight bert/encoder/layer_10/attention/output/dense/bias with shape [768]\n","Loading TF weight bert/encoder/layer_10/attention/output/dense/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_10/attention/self/key/bias with shape [768]\n","Loading TF weight bert/encoder/layer_10/attention/self/key/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_10/attention/self/query/bias with shape [768]\n","Loading TF weight bert/encoder/layer_10/attention/self/query/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_10/attention/self/value/bias with shape [768]\n","Loading TF weight bert/encoder/layer_10/attention/self/value/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_10/intermediate/dense/bias with shape [3072]\n","Loading TF weight bert/encoder/layer_10/intermediate/dense/kernel with shape [768, 3072]\n","Loading TF weight bert/encoder/layer_10/output/LayerNorm/beta with shape [768]\n","Loading TF weight bert/encoder/layer_10/output/LayerNorm/gamma with shape [768]\n","Loading TF weight bert/encoder/layer_10/output/dense/bias with shape [768]\n","Loading TF weight bert/encoder/layer_10/output/dense/kernel with shape [3072, 768]\n","Loading TF weight bert/encoder/layer_11/attention/output/LayerNorm/beta with shape [768]\n","Loading TF weight bert/encoder/layer_11/attention/output/LayerNorm/gamma with shape [768]\n","Loading TF weight bert/encoder/layer_11/attention/output/dense/bias with shape [768]\n","Loading TF weight bert/encoder/layer_11/attention/output/dense/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_11/attention/self/key/bias with shape [768]\n","Loading TF weight bert/encoder/layer_11/attention/self/key/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_11/attention/self/query/bias with shape [768]\n","Loading TF weight bert/encoder/layer_11/attention/self/query/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_11/attention/self/value/bias with shape [768]\n","Loading TF weight bert/encoder/layer_11/attention/self/value/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_11/intermediate/dense/bias with shape [3072]\n","Loading TF weight bert/encoder/layer_11/intermediate/dense/kernel with shape [768, 3072]\n","Loading TF weight bert/encoder/layer_11/output/LayerNorm/beta with shape [768]\n","Loading TF weight bert/encoder/layer_11/output/LayerNorm/gamma with shape [768]\n","Loading TF weight bert/encoder/layer_11/output/dense/bias with shape [768]\n","Loading TF weight bert/encoder/layer_11/output/dense/kernel with shape [3072, 768]\n","Loading TF weight bert/encoder/layer_2/attention/output/LayerNorm/beta with shape [768]\n","Loading TF weight bert/encoder/layer_2/attention/output/LayerNorm/gamma with shape [768]\n","Loading TF weight bert/encoder/layer_2/attention/output/dense/bias with shape [768]\n","Loading TF weight bert/encoder/layer_2/attention/output/dense/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_2/attention/self/key/bias with shape [768]\n","Loading TF weight bert/encoder/layer_2/attention/self/key/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_2/attention/self/query/bias with shape [768]\n","Loading TF weight bert/encoder/layer_2/attention/self/query/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_2/attention/self/value/bias with shape [768]\n","Loading TF weight bert/encoder/layer_2/attention/self/value/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_2/intermediate/dense/bias with shape [3072]\n","Loading TF weight bert/encoder/layer_2/intermediate/dense/kernel with shape [768, 3072]\n","Loading TF weight bert/encoder/layer_2/output/LayerNorm/beta with shape [768]\n","Loading TF weight bert/encoder/layer_2/output/LayerNorm/gamma with shape [768]\n","Loading TF weight bert/encoder/layer_2/output/dense/bias with shape [768]\n","Loading TF weight bert/encoder/layer_2/output/dense/kernel with shape [3072, 768]\n","Loading TF weight bert/encoder/layer_3/attention/output/LayerNorm/beta with shape [768]\n","Loading TF weight bert/encoder/layer_3/attention/output/LayerNorm/gamma with shape [768]\n","Loading TF weight bert/encoder/layer_3/attention/output/dense/bias with shape [768]\n","Loading TF weight bert/encoder/layer_3/attention/output/dense/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_3/attention/self/key/bias with shape [768]\n","Loading TF weight bert/encoder/layer_3/attention/self/key/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_3/attention/self/query/bias with shape [768]\n","Loading TF weight bert/encoder/layer_3/attention/self/query/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_3/attention/self/value/bias with shape [768]\n","Loading TF weight bert/encoder/layer_3/attention/self/value/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_3/intermediate/dense/bias with shape [3072]\n","Loading TF weight bert/encoder/layer_3/intermediate/dense/kernel with shape [768, 3072]\n","Loading TF weight bert/encoder/layer_3/output/LayerNorm/beta with shape [768]\n","Loading TF weight bert/encoder/layer_3/output/LayerNorm/gamma with shape [768]\n","Loading TF weight bert/encoder/layer_3/output/dense/bias with shape [768]\n","Loading TF weight bert/encoder/layer_3/output/dense/kernel with shape [3072, 768]\n","Loading TF weight bert/encoder/layer_4/attention/output/LayerNorm/beta with shape [768]\n","Loading TF weight bert/encoder/layer_4/attention/output/LayerNorm/gamma with shape [768]\n","Loading TF weight bert/encoder/layer_4/attention/output/dense/bias with shape [768]\n","Loading TF weight bert/encoder/layer_4/attention/output/dense/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_4/attention/self/key/bias with shape [768]\n","Loading TF weight bert/encoder/layer_4/attention/self/key/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_4/attention/self/query/bias with shape [768]\n","Loading TF weight bert/encoder/layer_4/attention/self/query/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_4/attention/self/value/bias with shape [768]\n","Loading TF weight bert/encoder/layer_4/attention/self/value/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_4/intermediate/dense/bias with shape [3072]\n","Loading TF weight bert/encoder/layer_4/intermediate/dense/kernel with shape [768, 3072]\n","Loading TF weight bert/encoder/layer_4/output/LayerNorm/beta with shape [768]\n","Loading TF weight bert/encoder/layer_4/output/LayerNorm/gamma with shape [768]\n","Loading TF weight bert/encoder/layer_4/output/dense/bias with shape [768]\n","Loading TF weight bert/encoder/layer_4/output/dense/kernel with shape [3072, 768]\n","Loading TF weight bert/encoder/layer_5/attention/output/LayerNorm/beta with shape [768]\n","Loading TF weight bert/encoder/layer_5/attention/output/LayerNorm/gamma with shape [768]\n","Loading TF weight bert/encoder/layer_5/attention/output/dense/bias with shape [768]\n","Loading TF weight bert/encoder/layer_5/attention/output/dense/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_5/attention/self/key/bias with shape [768]\n","Loading TF weight bert/encoder/layer_5/attention/self/key/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_5/attention/self/query/bias with shape [768]\n","Loading TF weight bert/encoder/layer_5/attention/self/query/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_5/attention/self/value/bias with shape [768]\n","Loading TF weight bert/encoder/layer_5/attention/self/value/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_5/intermediate/dense/bias with shape [3072]\n","Loading TF weight bert/encoder/layer_5/intermediate/dense/kernel with shape [768, 3072]\n","Loading TF weight bert/encoder/layer_5/output/LayerNorm/beta with shape [768]\n","Loading TF weight bert/encoder/layer_5/output/LayerNorm/gamma with shape [768]\n","Loading TF weight bert/encoder/layer_5/output/dense/bias with shape [768]\n","Loading TF weight bert/encoder/layer_5/output/dense/kernel with shape [3072, 768]\n","Loading TF weight bert/encoder/layer_6/attention/output/LayerNorm/beta with shape [768]\n","Loading TF weight bert/encoder/layer_6/attention/output/LayerNorm/gamma with shape [768]\n","Loading TF weight bert/encoder/layer_6/attention/output/dense/bias with shape [768]\n","Loading TF weight bert/encoder/layer_6/attention/output/dense/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_6/attention/self/key/bias with shape [768]\n","Loading TF weight bert/encoder/layer_6/attention/self/key/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_6/attention/self/query/bias with shape [768]\n","Loading TF weight bert/encoder/layer_6/attention/self/query/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_6/attention/self/value/bias with shape [768]\n","Loading TF weight bert/encoder/layer_6/attention/self/value/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_6/intermediate/dense/bias with shape [3072]\n","Loading TF weight bert/encoder/layer_6/intermediate/dense/kernel with shape [768, 3072]\n","Loading TF weight bert/encoder/layer_6/output/LayerNorm/beta with shape [768]\n","Loading TF weight bert/encoder/layer_6/output/LayerNorm/gamma with shape [768]\n","Loading TF weight bert/encoder/layer_6/output/dense/bias with shape [768]\n","Loading TF weight bert/encoder/layer_6/output/dense/kernel with shape [3072, 768]\n","Loading TF weight bert/encoder/layer_7/attention/output/LayerNorm/beta with shape [768]\n","Loading TF weight bert/encoder/layer_7/attention/output/LayerNorm/gamma with shape [768]\n","Loading TF weight bert/encoder/layer_7/attention/output/dense/bias with shape [768]\n","Loading TF weight bert/encoder/layer_7/attention/output/dense/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_7/attention/self/key/bias with shape [768]\n","Loading TF weight bert/encoder/layer_7/attention/self/key/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_7/attention/self/query/bias with shape [768]\n","Loading TF weight bert/encoder/layer_7/attention/self/query/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_7/attention/self/value/bias with shape [768]\n","Loading TF weight bert/encoder/layer_7/attention/self/value/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_7/intermediate/dense/bias with shape [3072]\n","Loading TF weight bert/encoder/layer_7/intermediate/dense/kernel with shape [768, 3072]\n","Loading TF weight bert/encoder/layer_7/output/LayerNorm/beta with shape [768]\n","Loading TF weight bert/encoder/layer_7/output/LayerNorm/gamma with shape [768]\n","Loading TF weight bert/encoder/layer_7/output/dense/bias with shape [768]\n","Loading TF weight bert/encoder/layer_7/output/dense/kernel with shape [3072, 768]\n","Loading TF weight bert/encoder/layer_8/attention/output/LayerNorm/beta with shape [768]\n","Loading TF weight bert/encoder/layer_8/attention/output/LayerNorm/gamma with shape [768]\n","Loading TF weight bert/encoder/layer_8/attention/output/dense/bias with shape [768]\n","Loading TF weight bert/encoder/layer_8/attention/output/dense/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_8/attention/self/key/bias with shape [768]\n","Loading TF weight bert/encoder/layer_8/attention/self/key/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_8/attention/self/query/bias with shape [768]\n","Loading TF weight bert/encoder/layer_8/attention/self/query/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_8/attention/self/value/bias with shape [768]\n","Loading TF weight bert/encoder/layer_8/attention/self/value/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_8/intermediate/dense/bias with shape [3072]\n","Loading TF weight bert/encoder/layer_8/intermediate/dense/kernel with shape [768, 3072]\n","Loading TF weight bert/encoder/layer_8/output/LayerNorm/beta with shape [768]\n","Loading TF weight bert/encoder/layer_8/output/LayerNorm/gamma with shape [768]\n","Loading TF weight bert/encoder/layer_8/output/dense/bias with shape [768]\n","Loading TF weight bert/encoder/layer_8/output/dense/kernel with shape [3072, 768]\n","Loading TF weight bert/encoder/layer_9/attention/output/LayerNorm/beta with shape [768]\n","Loading TF weight bert/encoder/layer_9/attention/output/LayerNorm/gamma with shape [768]\n","Loading TF weight bert/encoder/layer_9/attention/output/dense/bias with shape [768]\n","Loading TF weight bert/encoder/layer_9/attention/output/dense/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_9/attention/self/key/bias with shape [768]\n","Loading TF weight bert/encoder/layer_9/attention/self/key/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_9/attention/self/query/bias with shape [768]\n","Loading TF weight bert/encoder/layer_9/attention/self/query/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_9/attention/self/value/bias with shape [768]\n","Loading TF weight bert/encoder/layer_9/attention/self/value/kernel with shape [768, 768]\n","Loading TF weight bert/encoder/layer_9/intermediate/dense/bias with shape [3072]\n","Loading TF weight bert/encoder/layer_9/intermediate/dense/kernel with shape [768, 3072]\n","Loading TF weight bert/encoder/layer_9/output/LayerNorm/beta with shape [768]\n","Loading TF weight bert/encoder/layer_9/output/LayerNorm/gamma with shape [768]\n","Loading TF weight bert/encoder/layer_9/output/dense/bias with shape [768]\n","Loading TF weight bert/encoder/layer_9/output/dense/kernel with shape [3072, 768]\n","Loading TF weight bert/pooler/dense/bias with shape [768]\n","Loading TF weight bert/pooler/dense/kernel with shape [768, 768]\n","Initialize PyTorch weight ['bert', 'embeddings', 'LayerNorm', 'beta']\n","Initialize PyTorch weight ['bert', 'embeddings', 'LayerNorm', 'gamma']\n","Initialize PyTorch weight ['bert', 'embeddings', 'position_embeddings']\n","Initialize PyTorch weight ['bert', 'embeddings', 'token_type_embeddings']\n","Initialize PyTorch weight ['bert', 'embeddings', 'word_embeddings']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_0', 'attention', 'output', 'LayerNorm', 'beta']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_0', 'attention', 'output', 'LayerNorm', 'gamma']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_0', 'attention', 'output', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_0', 'attention', 'output', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_0', 'attention', 'self', 'key', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_0', 'attention', 'self', 'key', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_0', 'attention', 'self', 'query', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_0', 'attention', 'self', 'query', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_0', 'attention', 'self', 'value', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_0', 'attention', 'self', 'value', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_0', 'intermediate', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_0', 'intermediate', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_0', 'output', 'LayerNorm', 'beta']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_0', 'output', 'LayerNorm', 'gamma']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_0', 'output', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_0', 'output', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_1', 'attention', 'output', 'LayerNorm', 'beta']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_1', 'attention', 'output', 'LayerNorm', 'gamma']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_1', 'attention', 'output', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_1', 'attention', 'output', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_1', 'attention', 'self', 'key', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_1', 'attention', 'self', 'key', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_1', 'attention', 'self', 'query', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_1', 'attention', 'self', 'query', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_1', 'attention', 'self', 'value', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_1', 'attention', 'self', 'value', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_1', 'intermediate', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_1', 'intermediate', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_1', 'output', 'LayerNorm', 'beta']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_1', 'output', 'LayerNorm', 'gamma']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_1', 'output', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_1', 'output', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_10', 'attention', 'output', 'LayerNorm', 'beta']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_10', 'attention', 'output', 'LayerNorm', 'gamma']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_10', 'attention', 'output', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_10', 'attention', 'output', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_10', 'attention', 'self', 'key', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_10', 'attention', 'self', 'key', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_10', 'attention', 'self', 'query', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_10', 'attention', 'self', 'query', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_10', 'attention', 'self', 'value', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_10', 'attention', 'self', 'value', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_10', 'intermediate', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_10', 'intermediate', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_10', 'output', 'LayerNorm', 'beta']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_10', 'output', 'LayerNorm', 'gamma']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_10', 'output', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_10', 'output', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_11', 'attention', 'output', 'LayerNorm', 'beta']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_11', 'attention', 'output', 'LayerNorm', 'gamma']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_11', 'attention', 'output', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_11', 'attention', 'output', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_11', 'attention', 'self', 'key', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_11', 'attention', 'self', 'key', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_11', 'attention', 'self', 'query', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_11', 'attention', 'self', 'query', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_11', 'attention', 'self', 'value', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_11', 'attention', 'self', 'value', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_11', 'intermediate', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_11', 'intermediate', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_11', 'output', 'LayerNorm', 'beta']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_11', 'output', 'LayerNorm', 'gamma']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_11', 'output', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_11', 'output', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_2', 'attention', 'output', 'LayerNorm', 'beta']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_2', 'attention', 'output', 'LayerNorm', 'gamma']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_2', 'attention', 'output', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_2', 'attention', 'output', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_2', 'attention', 'self', 'key', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_2', 'attention', 'self', 'key', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_2', 'attention', 'self', 'query', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_2', 'attention', 'self', 'query', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_2', 'attention', 'self', 'value', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_2', 'attention', 'self', 'value', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_2', 'intermediate', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_2', 'intermediate', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_2', 'output', 'LayerNorm', 'beta']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_2', 'output', 'LayerNorm', 'gamma']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_2', 'output', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_2', 'output', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_3', 'attention', 'output', 'LayerNorm', 'beta']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_3', 'attention', 'output', 'LayerNorm', 'gamma']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_3', 'attention', 'output', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_3', 'attention', 'output', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_3', 'attention', 'self', 'key', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_3', 'attention', 'self', 'key', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_3', 'attention', 'self', 'query', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_3', 'attention', 'self', 'query', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_3', 'attention', 'self', 'value', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_3', 'attention', 'self', 'value', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_3', 'intermediate', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_3', 'intermediate', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_3', 'output', 'LayerNorm', 'beta']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_3', 'output', 'LayerNorm', 'gamma']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_3', 'output', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_3', 'output', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_4', 'attention', 'output', 'LayerNorm', 'beta']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_4', 'attention', 'output', 'LayerNorm', 'gamma']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_4', 'attention', 'output', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_4', 'attention', 'output', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_4', 'attention', 'self', 'key', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_4', 'attention', 'self', 'key', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_4', 'attention', 'self', 'query', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_4', 'attention', 'self', 'query', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_4', 'attention', 'self', 'value', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_4', 'attention', 'self', 'value', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_4', 'intermediate', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_4', 'intermediate', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_4', 'output', 'LayerNorm', 'beta']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_4', 'output', 'LayerNorm', 'gamma']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_4', 'output', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_4', 'output', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_5', 'attention', 'output', 'LayerNorm', 'beta']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_5', 'attention', 'output', 'LayerNorm', 'gamma']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_5', 'attention', 'output', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_5', 'attention', 'output', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_5', 'attention', 'self', 'key', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_5', 'attention', 'self', 'key', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_5', 'attention', 'self', 'query', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_5', 'attention', 'self', 'query', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_5', 'attention', 'self', 'value', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_5', 'attention', 'self', 'value', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_5', 'intermediate', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_5', 'intermediate', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_5', 'output', 'LayerNorm', 'beta']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_5', 'output', 'LayerNorm', 'gamma']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_5', 'output', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_5', 'output', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_6', 'attention', 'output', 'LayerNorm', 'beta']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_6', 'attention', 'output', 'LayerNorm', 'gamma']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_6', 'attention', 'output', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_6', 'attention', 'output', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_6', 'attention', 'self', 'key', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_6', 'attention', 'self', 'key', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_6', 'attention', 'self', 'query', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_6', 'attention', 'self', 'query', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_6', 'attention', 'self', 'value', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_6', 'attention', 'self', 'value', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_6', 'intermediate', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_6', 'intermediate', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_6', 'output', 'LayerNorm', 'beta']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_6', 'output', 'LayerNorm', 'gamma']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_6', 'output', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_6', 'output', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_7', 'attention', 'output', 'LayerNorm', 'beta']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_7', 'attention', 'output', 'LayerNorm', 'gamma']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_7', 'attention', 'output', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_7', 'attention', 'output', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_7', 'attention', 'self', 'key', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_7', 'attention', 'self', 'key', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_7', 'attention', 'self', 'query', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_7', 'attention', 'self', 'query', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_7', 'attention', 'self', 'value', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_7', 'attention', 'self', 'value', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_7', 'intermediate', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_7', 'intermediate', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_7', 'output', 'LayerNorm', 'beta']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_7', 'output', 'LayerNorm', 'gamma']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_7', 'output', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_7', 'output', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_8', 'attention', 'output', 'LayerNorm', 'beta']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_8', 'attention', 'output', 'LayerNorm', 'gamma']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_8', 'attention', 'output', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_8', 'attention', 'output', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_8', 'attention', 'self', 'key', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_8', 'attention', 'self', 'key', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_8', 'attention', 'self', 'query', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_8', 'attention', 'self', 'query', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_8', 'attention', 'self', 'value', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_8', 'attention', 'self', 'value', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_8', 'intermediate', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_8', 'intermediate', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_8', 'output', 'LayerNorm', 'beta']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_8', 'output', 'LayerNorm', 'gamma']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_8', 'output', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_8', 'output', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_9', 'attention', 'output', 'LayerNorm', 'beta']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_9', 'attention', 'output', 'LayerNorm', 'gamma']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_9', 'attention', 'output', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_9', 'attention', 'output', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_9', 'attention', 'self', 'key', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_9', 'attention', 'self', 'key', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_9', 'attention', 'self', 'query', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_9', 'attention', 'self', 'query', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_9', 'attention', 'self', 'value', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_9', 'attention', 'self', 'value', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_9', 'intermediate', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_9', 'intermediate', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_9', 'output', 'LayerNorm', 'beta']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_9', 'output', 'LayerNorm', 'gamma']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_9', 'output', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'encoder', 'layer_9', 'output', 'dense', 'kernel']\n","Initialize PyTorch weight ['bert', 'pooler', 'dense', 'bias']\n","Initialize PyTorch weight ['bert', 'pooler', 'dense', 'kernel']\n","Save PyTorch model to biobert_v1.1_pubmed/pytorch_model.bin\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"7kfoQwc3AwVo","executionInfo":{"status":"ok","timestamp":1608027929860,"user_tz":-330,"elapsed":29242,"user":{"displayName":"Akash Gangwar","photoUrl":"","userId":"02081727011117790241"}},"outputId":"3dc8ab74-08b4-4bf8-bb9e-cd3ec3d48e16"},"source":["import os\r\n","import io\r\n","import numpy as np\r\n","import pandas as pd\r\n","import torch\r\n","from transformers import BertTokenizer, BertModel, BertForPreTraining\r\n","import nltk\r\n","from sklearn.model_selection import train_test_split\r\n","import re\r\n","from transformers import BertTokenizer, BertModel\r\n","from transformers import BertForTokenClassification, AdamW\r\n","from transformers import tokenization_utils_base\r\n","import torch\r\n","import torch.nn.functional as F\r\n","import torch.nn as nn\r\n","import pickle\r\n","from torch.utils.data import TensorDataset, DataLoader\r\n","nltk.download('punkt')"],"execution_count":null,"outputs":[{"output_type":"stream","text":["[nltk_data] Downloading package punkt to /root/nltk_data...\n","[nltk_data]   Package punkt is already up-to-date!\n"],"name":"stdout"},{"output_type":"execute_result","data":{"text/plain":["True"]},"metadata":{"tags":[]},"execution_count":9}]},{"cell_type":"code","metadata":{"id":"d8RuawfTA2Hn"},"source":["TEXT_PATH=\"/content/drive/My Drive/MeasEval-main/data/train/test/text\"\r\n","TSV_PATH=\"/content/drive/My Drive/MeasEval-main/data/train/dev/tsv\"\r\n","MEASURED_ENTITY_MODEL_PATH=\"/content/drive/My Drive/Model/biobert_measured_entity.pt\"\r\n","MEASURED_QUANTITY_MODEL_PATH=\"/content/drive/My Drive/Model/quantity_mixedloss_t55_sh.pt\"\r\n","MEASURED_PROPERTY_MODEL_PATH=\"/content/drive/My Drive/Model/biobert_measured_prop.pt\"\r\n","QUALIFIER_PATH=\"/content/drive/My Drive/Model/qualifier_20f1_t995.pt\"\r\n","COUNT_MEASUREMENT_MODEL_PATH=\"/content/drive/My Drive/Model/count_t95_f70_ep20_sirdata.pt\"\r\n","INFORMATION_DETECTION_PATH=\"/content/drive/My Drive/Model/info_60f1_t325.pt\"\r\n","UNIT_DETECTION_PATH=\"/content/drive/My Drive/Model/units_t75_96f1.pt\"\r\n","VOCAB_PATH=\"/content/drive/My Drive/Model/vocab.Field\""],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"--YBsEMlBndI"},"source":["device = torch.device(\"cuda\" if torch.cuda.is_available() else \"cpu\")\r\n","model_BioBert = BertModel.from_pretrained('/content/biobert_v1.1_pubmed')\r\n","tokenizer_BioBert = BertTokenizer(vocab_file='biobert_v1.1_pubmed/vocab.txt', do_lower_case=False)"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"Jl2jnRNfA4ne","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1608027932630,"user_tz":-330,"elapsed":31985,"user":{"displayName":"Akash Gangwar","photoUrl":"","userId":"02081727011117790241"}},"outputId":"70b2fcc1-2789-44cd-d93e-60be52d249e6"},"source":["from google.colab import drive\r\n","drive.mount('/content/drive')"],"execution_count":null,"outputs":[{"output_type":"stream","text":["Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount(\"/content/drive\", force_remount=True).\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"eIanfdEtBQw8"},"source":["files=os.listdir(TEXT_PATH)"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"X60ZcTVHA6yu"},"source":["test_text=[]\r\n","id=[]\r\n","j=1\r\n","for i in files:\r\n","\r\n","  pathtxt=os.path.join(TEXT_PATH,i)\r\n","  \r\n","  with open(pathtxt, 'r') as f:\r\n","    text=f.read()\r\n","  \r\n","  sent_text = nltk.sent_tokenize(text)\r\n","  for l in sent_text:\r\n","    test_text.append(l)\r\n","    id.append(j)\r\n","  j+=1"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"IhukswVKBNF-","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1608027948476,"user_tz":-330,"elapsed":47806,"user":{"displayName":"Akash Gangwar","photoUrl":"","userId":"02081727011117790241"}},"outputId":"40f194da-e04c-45f5-c62d-7b471cae1106"},"source":["test_text"],"execution_count":null,"outputs":[{"output_type":"execute_result","data":{"text/plain":["['(F) Adult hRPE cultures display expression of markers typical of native RPE in their polarized localization.',\n"," 'DAPI is cyan, whereas all other immunofluorescence is gold.',\n"," 'Claudin 19, ezrin, ZO1, and MCT1 are preferentially located on the apical side.',\n"," 'RPE65 and CRALBP are cytoplasmic.',\n"," 'The scale bars represent 10 Î¼m.',\n"," 'GFP-expressing H9, BBHX8, and A1ATD-1 cells were generated by stable transfection using lipofectamine 2000 (Invitrogen) as described previously (Vallier et al., 2001).',\n"," 'GFP-positive cells were differentiated into foregut cells and then dissociated into single cells.',\n"," 'An individually isolated GFP cell was then transferred into a well containing non-GFP-positive hFSCs.',\n"," 'Wells were visually inspected 12 hr after plating, and wells containing a single GFP-positive hFSC were selected for clonal expansion.',\n"," 'For group analyses, the following procedures were applied.',\n"," '(1) T1-weighted sMRIs from each subject (Fig.',\n"," '1A) were registered to MNI space (Montreal Neurological Institute, MNI-152 atlas as in Fig.',\n"," \"1B) using an affine transformation (FLIRTâ€“FMRIB's Linear Image Registration Tool) (Jenkinson and Smith, 2001) in FSL (www.fmrib.ox.ac.uk/fsl/).\",\n"," '(2) The cortical (Fig.',\n"," \"1C) and subcortical masks with pre-defined brain regions from the standard atlas were transferred to the individual's headspace (Fig.\",\n"," '1D), using the inverse of the transformation obtained in the first step: the Harvard-Oxford Atlas, part of the FSL software with masks of 96 cortical gray-matter regions (48 regions in each hemisphere), 21 sub-cortical regions, and cerebellum, was used.',\n"," '(3) The regional masks were down-sampled to a cubic source grid with voxels of 5 mm per side (Fig.',\n"," '1E).',\n"," '(4) VESTAL MEG source imaging used the source grid from step 3.',\n"," 'This step permits group-based analyses.',\n"," \"In the shown example, MEG responses evoked by S1 localized to left and right Heschl's gyri (Fig.\",\n"," '1F).',\n"," '(5) Finally, for regions of interest (ROIs), the source time course was obtained by summing activity from all ROI voxels.',\n"," 'Fig.',\n"," \"1H shows the time course from left Heschl's gyrus (dark blue region in Fig.\",\n"," '1C and D).',\n"," 'Another important finding is that, in spite of the immune responses mounted by the host brain, a substantial number of TH+ cells survived in the allografts.',\n"," 'This is consistent with previous clinical reports of human fetal cell transplantation.',\n"," 'Postmortem analyses of the patients revealed robust survival of DA neurons in spite of the fact that numerous immune cells were present around the graft (Kordower et al., 1997).',\n"," 'In two double-blind clinical trials, immunosuppressive drugs were never used (Freed et al., 2001) or were withdrawn after 6 months (Olanow et al., 2003).',\n"," 'In these cases, the cells from multiple fetuses were used without HLA matching, but more than 50,000 TH+ cells had survived after several years.',\n"," 'Our quantitative PCR (qPCR) study in vitro showed that the expression of MHC-I increased in response to IFN-Î³, but the expression level was still 1/10 that of untreated monkey peripheral blood cells (Figure 2B).',\n"," 'The in vivo studies revealed that the serum level of IFN-Î³ increased at 2 months, and CD45+ cells (including CD8+ cells) accumulated in the allografts 3.5â€“4 months after the transplant.',\n"," 'On the other hand, the levels of INF-Î³ in the cerebrospinal fluid (CSF) and the levels of tumor necrosis factor Î± (TNF-Î±) in both the serum and CSF were below the limit of detection by ELISA (data not shown).',\n"," 'An immunofluorescence study did not reveal any apparent expression of MHC-I by the grafted cells (Figure S4A).',\n"," 'Therefore, it is possible that the immune response in the primate brain was not strong enough to reject all of the donor cells.',\n"," 'These findings closely correlate with the results of previous murine experiments (Hudson et al., 1994; Shinoda et al., 1995).',\n"," 'To apply our findings to a more clinically relevant setting, we investigated the expression of HLA-I during neural differentiation of human ESCs (hESCs) and iPSCs by qPCR (Figure S4D).',\n"," 'The expression level was 1/100 compared with that of human peripheral blood cells in both hESCs and iPSCs, and it was similarly elevated in response to IFN-Î³.',\n"," 'It is difficult to precisely compare immunogenicity in monkeys with that in humans, but the low expression level of MHC-I by the donor cells may account for the mild rejection in both monkey and human neural transplantation.',\n"," 'We next addressed whether hESC-derived trigeminal neurons can engraft in the adult mouse CNS and project toward their physiological target.',\n"," 'The trigeminal nuclei in the brainstem receive afferent innervation from the trigeminal sensory ganglion that is relayed to the contralateral thalamus.',\n"," 'The pons was selected as site for transplantation, because it is surgically accessible and located within proximity of the trigeminal brain stem nuclei that receive afferent input from the trigeminal ganglia.',\n"," 'Hence, GFP+ human trigeminal neuron clusters were injected into adult NOD/SCID mice via stereotactic surgery (see Experimental Procedures).',\n"," 'Histological analysis 4 weeks after transplantation showed survival of GFP+ human cell graft in the ventral pons (Figure S5E).',\n"," 'Although GFP+ cell bodies remained tightly clustered at injection site, GFP+ fibers showed extensive projections into the host brain (n = 6) including the endogenous trigeminal nuclei (Figure S5F).',\n"," 'Expression of BRN3A confirmed the sensory neuron identity of the cells (Figure S5G).',\n"," 'Graft-derived human fiber bundles (hNCAM+ and GFP+) were observed emanating from the graft core (Figure S5H).',\n"," 'These data demonstrate in vivo survival of trigeminal placode derivatives, differentiation along sensory neuron lineage, and the establishment of axonal projections toward relevant endogenous targets in the embryonic chick and adult mouse brain.',\n"," '(B) Fetal hRPE stained for SC121 (red) and MCT1 (green; scale bars, 125 Î¼m and 25 Î¼m [inset]).',\n"," '(F) Intracellular PGRN levels in iPSC-derived neurons after medium collection.',\n"," 'Values of control line 20 were set to 100% (n = 3â€“4 independent cultures).',\n"," 'Prior to VESTAL analyses, a 5â€“55 Hz bandpass filter was applied.',\n"," 'VESTAL analyses examined activity 30â€“130 ms post-stimulus producing a 4D activation map (3D volumes across time) as well as a 2D source time-course matrix.',\n"," 'The average percent variance explained for gradiometer data using VESTAL was 95.81% for HC and 94.38% for SZ.',\n"," 'The average percent variance explained for magnetometer data using VESTAL program was 96.24% for HC and 93.17% for SZ.',\n"," 'There were no group differences in percent variance explained for gradiometer data (t(39) = 1.16, p = 0.25) or magnetometer data (t(39) = 1.31, p = 0.20).',\n"," '(C) Control lines 16 and 17, sporadic lines 12 and 23, and PGRN S116X lines 1 and 14 were immunostained for AFP (endoderm), desmin (mesoderm), and Î²III-tubulin (ectoderm), and counterstained with DAPI (nuclei).',\n"," 'All lines showed a normal karyotype.',\n"," 'Scale bar: 50 Î¼m.',\n"," '(Iâ€“M) Teratoma formation at 3 months after transplantation in the testes of SCID mice.',\n"," 'H&E staining of the sections showed histological features of the neuroepithelium (J), cartilage (K), muscle (L) and gut-like epithelium (M).',\n"," 'Using a recombinant adeno-associated viral (rAAV)-based gene-targeting method, we inserted the gene-encoding GFP into the SOX2 locus in H9 hESCs (Figure 1A).',\n"," 'Proper homologous recombination led to the replacement of the SOX2 open reading frame with that of GFP and a neomycin selection cassette (SV40-Neo).',\n"," 'After infection with rAAV and G418 drug selection, a total of 36 clones were expanded and screened by Southern blotting for homologous recombination events.',\n"," 'Among these clones, 26 (72%) were found to carry the GFP-Neo cassette in the SOX2 locus (Figure S1A available online).',\n"," 'No clones in which both SOX2 alleles were disrupted were isolated.',\n"," 'Our subsequent analysis focused on one of these clones, clone 23 (hSOX2-23).',\n"," 'We confirmed appropriate gene targeting in this clone using multiple restriction digests followed by Southern blotting (Figures 1B, S1B, and S1C).',\n"," 'We did not observe nontargeted insertions of the rAAV sequences, and cells exhibited a normal karyotype (data not shown).',\n"," 'Flow cytometry of hSOX2-23 revealed that the majority of the cells expressed GFP (Figure 1C).',\n"," 'By comparison, a drug-selected clone, hSOX2-25, which was negative for targeted insertion (Figure S1A), showed no detectable GFP (Figure S2A).',\n"," 'Despite only having one copy of SOX2, hSOX2-23 had similar levels of SOX2, OCT4, and NANOG expression as hSOX2-25 and wild-type (WT) hESCs (Figure S2B).',\n"," 'Moreover, the percentage of GFP-positive (GFP+) cells in hSOX2-23 was constant over more than 20 passages.',\n"," 'Immunofluorescence (IF) staining of hSOX2-23 showed that 100% of GFP+ cells expressed SOX2 protein (Figure S2C).',\n"," 'Additionally, hSOX2-23 colonies had characteristic hESC morphology (Figure S2D) and expressed markers of the undifferentiated state, such as NANOG (Figure S2E).',\n"," 'These results show that this rAAV-based gene-targeting method can be used to efficiently disrupt genes by homologous recombination.',\n"," 'In addition, the SOX2-GFP hESC marker line can be used to monitor SOX2 expression in undifferentiated hESCs.',\n"," 'Based on several previous publications (Khan et al., 2010, 2011), we explored the utility of adeno-associated virus (AAV) as a method to improve gene targeting efficiencies in hPSCs.',\n"," 'For SOX2, a gene that is highly expressed in undifferentiated hPSCs, gene targeting rates were greater than 70%.',\n"," 'Similar targeting efficiencies in hPSCs using AAV have been reported by others (Asuri et al., 2012; Khan et al., 2010, 2011; Smith-Arica et al., 2003), indicating that AAV offers a highly efficient and robust approach to target genes for HR in hPSCs.',\n"," '(C) C-peptide and PDX1 expression was confirmed by immunocytochemistry of cells differentiated for 25 days.',\n"," 'â€¢Adult hRPESC-derived RPE had comparable in vitro characteristics to fetal hRPEâ€¢hRPE monolayers survived 4 weeks on PET carriers under the rabbit retinaâ€¢Better xenograft survival may be due to the maintained hRPE cell polarityâ€¢Atrophy of the retina overlaying the hRPE xenograft remains a future challenge',\n"," '(O) GH plasma levels using a human specific ELISA (6 weeks after transplantation).',\n"," '(R) Quantification of immunocytochemical analyses for each iPSC line.',\n"," 'Data are shown as the means Â± SD (n = 3 independent experiments).',\n"," 'SER, serotonin, TUBÎ²III, Î²-tubulin class III.',\n"," 'Scale bars: 200 Î¼m in (A)â€“(H), 50 Î¼m in insets of (C)â€“(F), 100 Î¼m in (J)â€“(M) and (Q).',\n"," 'The two FTD patients under investigation in this study were part of a longitudinal dementia research program at the Memory and Aging Center, University of California, San Francisco.',\n"," 'Both had an 8-year history of behavioral changes and memory impairment at the time of tissue collection for this study.',\n"," 'One patient, a 67-year-old male with sporadic FTD, tested negative for mutations in GRN, MAPT, and C9ORF72.',\n"," 'The other patient, a 64-year-old male with a significant family history of dementia, had behavioral variant FTD.',\n"," 'MRI in this patient demonstrated severe bifrontal and temporal atrophy associated with gliosis in the frontal lobes (greater on the right).',\n"," 'One year later, MRI scans showed progression of atrophy and gliosis.',\n"," 'Genetic testing revealed a novel nonsense mutation in GRN, p.S116X (g.4627C > A, c.347C > A), which is predicted to result in a premature stop codon.',\n"," 'Both FTD patients had parkinsonism, which is typical of all FTD patients with PGRN mutations.',\n"," 'An age-matched subject, a clinically normal 64-year-old male with no mutations in GRN, MAPT, or C9ORF72, served as a control.',\n"," '(Qâ€“S) Magnetic resonance images of a representative animal (No.',\n"," '6, autograft) at 3 months after the transplant.',\n"," 'The arrowheads indicate the directions of the cell injections.',\n"," '(Q) coronal, (R) axial, and (S) sagittal.',\n"," 'The letter L indicates the left side.',\n"," '(A) Fetal hRPE stained for pan-cytokeratin (scale bar, 50 Î¼m); inset shows section overview stained with hematoxylin/eosin (scale bar, 200 Î¼m).',\n"," 'To assess the in vivo properties of hESC-derived trigeminal placode precursors, PIP-induced neuronal clusters, derived from a constitutively GFP-positive hESC line (Figures S5A and S5B), were injected into the developing chick embryo targeting the early trigeminal anlage at H&H stage 10â€“12 (Figure S5C).',\n"," 'Human cells were identified based on GFP expression and use of human specific antibodies against cytoplasmic antigen (hCA).',\n"," 'Two days after in ovo transplantation, surviving GFP+ cells were found dispersed in the area of the endogenous chick trigeminal ganglion (Figure 4P).',\n"," 'We observed extensive GFP+ human fiber bundles coexpressing hCA and peripherin (Figures 4Q and 4R).',\n"," 'In contrast, no hCA or peripherin expression was detected in the neural tube of the embryo (Figure S5D).',\n"," 'The in vivo fiber outgrowth 2 days after transplantation was reminiscent of the extensive in vitro fiber outgrowth of replated trigeminal neuron clusters (Figure S5A).',\n"," 'Peripherin expression in vivo (Figure 4S) confirmed the peripheral neuron identity of the grafted cells.',\n"," 'For the first two animals (Nos.',\n"," '1 and 4), we established iPSCs from fibroblasts derived from the oral mucosa using retroviral vectors (Okita et al., 2011).',\n"," 'For the other two animals (Nos.',\n"," '6 and 8), we used peripheral blood mononuclear cells (PBMCs) with nonintegrating episomal vectors (Okita et al., 2013).',\n"," 'We selected the best clone from each animal according to the following criteria: a stable embryonic stem cell (ESC)-like morphology of the colonies after passaging, expression of pluripotent markers, few or no integrated transgenes (Figures 1Aâ€“1F; Figure S1 available online), and the potential for stable neural differentiation.',\n"," 'A PCR analysis revealed that all of the clones with retroviral vectors showed apparent expression of remaining transgenes (Figures S1C and S1D), whereas the clones with episomal vectors never did (Figure S1F).',\n"," 'To detect the iPSC-derived cells in a brain, we introduced GFP (Figures 1G and 1H).',\n"," 'The selected clones of iPSCs had the potential to generate teratomas in the testes of a severe combined immunodeficiency (SCID) mouse within 12 weeks (Figures 1Iâ€“1M).',\n"," '(A) Flow-cytometric analyses for MHC-I (HLA-A, HLA-B, and HLA-C).',\n"," 'Incubation of the cells with IFN-Î³ for 48 hr increased the MHC-I expression (green).',\n"," 'Our group has developed a defined culture system to direct the differentiation of hPSCs into a near-homogenous population of definitive endoderm (DE) cells that have the capacity to differentiate into hepatocytes and pancreatic progenitors (Brown et al., 2011; Cho et al., 2012; Rashid et al., 2010; Touboul et al., 2010; Vallier et al., 2009a; Yusa et al., 2011).',\n"," 'Cells grown in these culture conditions successively express primitive streak markers (T and Mixl1), downregulate pluripotency markers (NANOG, SOX2, and POU5F1) and progressively upregulate definitive endoderm markers (CXCR4, FOXA2, GATA4, CERB, and SOX17) (Figures S1Aâ€“S1C available online).',\n"," 'Flow cytometry analyses showed that 80% of the resulting DE population coexpresses CXCR4 and SOX17 (Figure S1D).',\n"," 'Interestingly, the resulting population of DE cells is negative for genes marking the foregut (SOX2), the midgut/hindgut (CDX2), the pancreas (PDX1), the liver (AFP), and the lungs (HOXA1) (Figures S1E and S1F) This confirms that DE cells generated in vitro could correspond to early endoderm progenitor cells prior to anteroposterior patterning or organogenesis.',\n"," 'In sequential PET studies, we observed increased uptake of [11C]PK11195 in one allograft (animal No.',\n"," '10) at 3 months (Figures 3A and 3B).',\n"," 'We could not detect any apparent uptake in the other animals or at any other time points (Figure S2).',\n"," 'Intriguingly, the serum level of IFN-Î³ temporarily increased at 2 months after the transplant in three animals (Figure 3C).',\n"," 'An immunofluorescence study conducted at 3.5â€“4 months showed that MHC-II+ cells were more frequently found in allografts than in autografts, especially in the monkey with increased uptake of [11C]PK11195 (Figure 3D, No.',\n"," '10).',\n"," 'The MHC-II staining never overlapped with that of GFP of the donor cells (Figure 3F), whereas it generally overlapped with that of IBA1 (Figure 3G), indicating that MHC-II was expressed by host-derived microglia.',\n"," 'Consistently, the number and density of IBA1+ cells were higher in allografts than in autografts (Figures 3E, 3H, and S4C).',\n"," 'An increase in the expression of MHC might trigger the recruitment of circulating immune cells, including T cells.',\n"," 'An immunofluorescence study revealed that more CD45+ cells (a marker for pan-leukocytes) accumulated in allografts compared with autografts (Figures 3I and 3J).',\n"," 'Most of the CD45+ cells were CD3+ T cells, and 60% of them were CD8+ killer T cells (Figures 3K and 3L).',\n"," 'These findings suggest that an acquired immune response was elicited only in the allografts in the primate brain.',\n"," 'This study was approved by the Institutional Review Board and Ethics Committees of the University of California, San Francisco, and written informed consent was obtained in all cases.',\n"," 'The patient with the PGRN S116X mutation followed the classic clinical progression for FTD and developed parkinsonism, as do all FTD patients with PGRN mutations, but he did not show typical features of PD dementia.',\n"," 'The patient with sporadic FTD also showed parkinsonism.',\n"," 'Skin biopsies were collected, cut into small pieces, and placed on culture dishes to allow the fibroblasts to expand.',\n"," \"The cells were maintained in Dulbecco's modified Eagle's medium supplemented with 10% fetal bovine serum, 1X nonessential amino acids, and penicillin/streptomycin (100 U/ml).\",\n"," 'iPSCs were generated as described previously (Takahashi et al., 2007).',\n"," 'Please see Supplemental Information for more details.',\n"," '(Dâ€“H) Adult hRPE stained for SC121 (red).',\n"," '(D) Adult hRPE stained for MCT1 (green).',\n"," '(E) Human adult RPE stained for ezrin (green).',\n"," 'hRPEs transplanted into rabbit SRS show absence of expression of ki67 (F), phosphohistone H3 (G), and caspase-3 (H).',\n"," 'Polarized fetal and adult hRPE cells were found in TEM (I and J).',\n"," 'Nuclei with regular chromatin were found in the basal compartment, a basal lamina ([I], large black arrowhead) had formed between the xenograft and PET carrier (black asterisks).',\n"," 'Melanosomes (M) in multiple stages, some microvilli abutting to the atrophic neural retina (NR), and junctional structures with desmosomes (small black arrowhead) and tight junctions (red arrowhead) were discerned apically.',\n"," 'Mitochondria (MC) were seen in the basolateral part of the cell.',\n"," 'Detachment from cell carrier (asterisk) in (J) is a histologic processing artifact.',\n"," 'Left images in (I) and (J) taken at 10,500Ã— magnification; right micrographs are rectangular zone in left at 25,000Ã—; scale bars represent 2 Î¼m/inset 0.2 Î¼m distance in (I) and (J).',\n"," 'iPSC colonies were detached with accutase (Millipore) and grown as embryoid bodies (EBs) in suspension for 5â€“6 days in iPSC medium without basic fibroblast growth factor.',\n"," 'EBs were allowed to attach and form rosettes.',\n"," 'Ten-day-old rosettes were collected and grown in suspension as neurospheres.',\n"," 'Neurospheres were dissociated after 3â€“4 weeks, and the cells were plated on glass coverslips (BD Biosciences) or plates coated with poly-D-lysine (0.1 mg/ml) and laminin (10 Î¼g/m).',\n"," 'Neurons were used after 2-4 weeks in culture.',\n"," 'For cell differentiation, FAP-specific iPS cells were cultured with serial changes of media as shown in Fig.',\n"," '3A.',\n"," 'To test whether FAP-specific iPS cells can differentiate into hepatocyte-like cells, we analyzed several markers via real-time PCR analysis on Day 5 (D5), D13 and D20 differentiated FAP-specific iPS cells (Fig.',\n"," '3B).',\n"," 'A decrease in expression of the pluripotency marker Oct3/4 was accompanied by differentiation of FAP-specific iPS cells.',\n"," 'Expression of the endoderm marker Sox17 was observed on D5 differentiation and decreased gradually after the medium was changed to hepatic differentiation medium on D7 (Fig.',\n"," '3B).',\n"," 'The hepatic progenitor marker AFP and the mature hepatocyte marker ALB were obviously expressed on D13 and D20.',\n"," 'In addition, immunocytochemical analyses showed Sox17 expression on D5, both HNF-4Î± and AFP expression on D13, and ALB cytoplasmic staining on D20 (Fig.',\n"," '3C).',\n"," 'Quantitative imaging analysis revealed that approximately 78 Â± 0.6% of cells were Sox17-positive on D5 and approximately 88 Â± 1.1% of cells were AFP-positive on D13 and approximately 29 Â± 0.9% of cells were ALB-positive on D20 (Fig.',\n"," '3D).',\n"," 'The ALB secretion in the media of differentiated FAP-specific iPS cells on D20 was approximately 20 Î¼g/ml (Fig.',\n"," '3E).',\n"," 'Moreover, these D20 differentiated FAP-specific iPS cells were also periodic acid-Schiff (PAS)-positive, indicating cytoplasmic glycogen storage (Fig.',\n"," '3F).',\n"," 'These results clearly indicated that FAP-specific iPS cells had the potential to differentiate into hepatocyte-like cells.']"]},"metadata":{"tags":[]},"execution_count":15}]},{"cell_type":"code","metadata":{"id":"3brnsvJ5BWAd"},"source":["class BERT_Arch(nn.Module):\r\n","\r\n","    def __init__(self, bert, embed_dim, hidden_dim, drop_prob, n_layers, out_dim):\r\n","      \r\n","      super(BERT_Arch, self).__init__()\r\n","\r\n","      self.bert = bert \r\n","      self.bilstm = nn.LSTM(embed_dim, hidden_dim,  bidirectional=True, batch_first=True)\r\n","      self.dropout = nn.Dropout(drop_prob)\r\n","      self.fc1 = nn.Linear(2*hidden_dim,out_dim)\r\n","      self.sigmoid = nn.Sigmoid()\r\n","\r\n","    #define the forward pass\r\n","    def forward(self, sent_id, mask):\r\n","\r\n","      #pass the inputs to the model  \r\n","      x = self.bert(sent_id, attention_mask=mask)\r\n","      embed = x[0]\r\n","      cls_hs = x[1]\r\n","      x,_ = self.bilstm(embed)\r\n","      x = self.dropout(x)\r\n","      x = self.fc1(x)\r\n","      x = self.sigmoid(x)\r\n","\r\n","      return x\r\n","      \r\n","bert_model = BERT_Arch(model_BioBert, 768, 256, 0.5, 1,1)\r\n","bert_model = bert_model.to(device)\r\n"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"32OjC-RMEO7Z"},"source":["Measured Property"]},{"cell_type":"code","metadata":{"id":"lxgzE2oCBc8Y"},"source":["bert_model = torch.load(MEASURED_PROPERTY_MODEL_PATH)\r\n","Measured_Property=[]\r\n","\r\n","for s in test_text:\r\n","  tokenized_text = tokenizer_BioBert.tokenize(s)\r\n","  test_inputs = tokenizer_BioBert(s,max_length = 512,padding='max_length',truncation=True, return_tensors=\"pt\")\r\n","  bert_model.eval()\r\n","  temp = bert_model(test_inputs['input_ids'].to(device), test_inputs['attention_mask'].to(device))>=0.30\r\n","  Measured_Property.append(temp[0])"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"LoCvQsmMCba2","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1608028003844,"user_tz":-330,"elapsed":1463,"user":{"displayName":"Akash Gangwar","photoUrl":"","userId":"02081727011117790241"}},"outputId":"b68e68cc-1b29-4df8-d7df-444168b2a081"},"source":["Property=[]\r\n","for i in range(len(test_text)):\r\n","  tokenized_text = tokenizer_BioBert.tokenize(test_text[i])\r\n","  txt=[]\r\n","  label=[]\r\n","  # print(test_text[i])\r\n","  combine=\"\"\r\n","  for j in range(len(tokenized_text)):\r\n","    if(Measured_Property[i][j+1] and Measured_Property[i][j] and j!=0):\r\n","      label.append(tokenized_text[j])\r\n","      if(len(tokenized_text[j])>=2 and tokenized_text[j][:2]==\"##\"):\r\n","          combine=combine+tokenized_text[j][2:]\r\n","      \r\n","      else:\r\n","        if(len(combine)==0):\r\n","          combine=tokenized_text[j]\r\n","          continue\r\n","        \r\n","        if(tokenized_text[j] in (\"!\", \",\" ,\"\\'\" ,\";\" ,\"\\\"\", \".\", \"-\" ,\"?\",\"+\",\"-\",\"*\",\"/\",\"%\")):\r\n","          combine=combine+tokenized_text[j]\r\n","          continue\r\n","        if(tokenized_text[j] in (\"(\",\")\") or combine in (\"(\",\")\")):\r\n","          continue\r\n","  \r\n","        start=[m.start() for m in re.finditer(combine,test_text[i])]\r\n","        end=[m.start() for m in re.finditer(tokenized_text[j],test_text[i])]\r\n","        flag=-1\r\n","        \r\n","        for s in start:\r\n","          for e in end:\r\n","            if(flag!=-1):\r\n","              break\r\n","            \r\n","            \r\n","            if(e-s-len(combine)+1==1):\r\n","              flag=1\r\n","            elif(e-s-len(combine)+1==2):\r\n","              flag=2\r\n","        \r\n","        if(flag==2):\r\n","          combine=combine+\" \"+tokenized_text[j]\r\n","        elif(flag==1):\r\n","          combine=combine+tokenized_text[j]\r\n","          \r\n","        else:\r\n","          txt.append(combine)\r\n","          combine=\"\"\r\n","\r\n","    elif(Measured_Property[i][j+1]):\r\n","      label.append(tokenized_text[j])\r\n","      \r\n","      if(len(combine)!=0):\r\n","        txt.append(combine)\r\n","      \r\n","      if(len(tokenized_text[j])>=2 and tokenized_text[j][:2]==\"##\"):\r\n","        tokenized_text[j]=tokenized_text[j][2:]\r\n","      \r\n","      combine=tokenized_text[j]\r\n","    else:\r\n","      if(len(combine)!=0):\r\n","        txt.append(combine)\r\n","      \r\n","      combine=\"\"\r\n","  if(len(combine)!=0):\r\n","        txt.append(combine)\r\n","  \r\n","  print(test_text[i])\r\n","  print(txt)\r\n","  Property.append(txt)\r\n"],"execution_count":null,"outputs":[{"output_type":"stream","text":["(F) Adult hRPE cultures display expression of markers typical of native RPE in their polarized localization.\n","[]\n","DAPI is cyan, whereas all other immunofluorescence is gold.\n","[]\n","Claudin 19, ezrin, ZO1, and MCT1 are preferentially located on the apical side.\n","[]\n","RPE65 and CRALBP are cytoplasmic.\n","[]\n","The scale bars represent 10 Î¼m.\n","['scale bars']\n","GFP-expressing H9, BBHX8, and A1ATD-1 cells were generated by stable transfection using lipofectamine 2000 (Invitrogen) as described previously (Vallier et al., 2001).\n","[]\n","GFP-positive cells were differentiated into foregut cells and then dissociated into single cells.\n","[]\n","An individually isolated GFP cell was then transferred into a well containing non-GFP-positive hFSCs.\n","[]\n","Wells were visually inspected 12 hr after plating, and wells containing a single GFP-positive hFSC were selected for clonal expansion.\n","[]\n","For group analyses, the following procedures were applied.\n","[]\n","(1) T1-weighted sMRIs from each subject (Fig.\n","[]\n","1A) were registered to MNI space (Montreal Neurological Institute, MNI-152 atlas as in Fig.\n","[]\n","1B) using an affine transformation (FLIRTâ€“FMRIB's Linear Image Registration Tool) (Jenkinson and Smith, 2001) in FSL (www.fmrib.ox.ac.uk/fsl/).\n","[]\n","(2) The cortical (Fig.\n","[]\n","1C) and subcortical masks with pre-defined brain regions from the standard atlas were transferred to the individual's headspace (Fig.\n","[]\n","1D), using the inverse of the transformation obtained in the first step: the Harvard-Oxford Atlas, part of the FSL software with masks of 96 cortical gray-matter regions (48 regions in each hemisphere), 21 sub-cortical regions, and cerebellum, was used.\n","[]\n","(3) The regional masks were down-sampled to a cubic source grid with voxels of 5 mm per side (Fig.\n","['grid with voxels']\n","1E).\n","[]\n","(4) VESTAL MEG source imaging used the source grid from step 3.\n","[]\n","This step permits group-based analyses.\n","[]\n","In the shown example, MEG responses evoked by S1 localized to left and right Heschl's gyri (Fig.\n","[]\n","1F).\n","[]\n","(5) Finally, for regions of interest (ROIs), the source time course was obtained by summing activity from all ROI voxels.\n","[]\n","Fig.\n","[]\n","1H shows the time course from left Heschl's gyrus (dark blue region in Fig.\n","[]\n","1C and D).\n","[]\n","Another important finding is that, in spite of the immune responses mounted by the host brain, a substantial number of TH+ cells survived in the allografts.\n","[]\n","This is consistent with previous clinical reports of human fetal cell transplantation.\n","[]\n","Postmortem analyses of the patients revealed robust survival of DA neurons in spite of the fact that numerous immune cells were present around the graft (Kordower et al., 1997).\n","[]\n","In two double-blind clinical trials, immunosuppressive drugs were never used (Freed et al., 2001) or were withdrawn after 6 months (Olanow et al., 2003).\n","['used', 'withdrawn']\n","In these cases, the cells from multiple fetuses were used without HLA matching, but more than 50,000 TH+ cells had survived after several years.\n","['TH', 'survived']\n","Our quantitative PCR (qPCR) study in vitro showed that the expression of MHC-I increased in response to IFN-Î³, but the expression level was still 1/10 that of untreated monkey peripheral blood cells (Figure 2B).\n","['expression', 'MHC-', 'increased', 'response', 'IFN', 'expression level']\n","The in vivo studies revealed that the serum level of IFN-Î³ increased at 2 months, and CD45+ cells (including CD8+ cells) accumulated in the allografts 3.5â€“4 months after the transplant.\n","['IFN-Î³ increased']\n","On the other hand, the levels of INF-Î³ in the cerebrospinal fluid (CSF) and the levels of tumor necrosis factor Î± (TNF-Î±) in both the serum and CSF were below the limit of detection by ELISA (data not shown).\n","[]\n","An immunofluorescence study did not reveal any apparent expression of MHC-I by the grafted cells (Figure S4A).\n","[]\n","Therefore, it is possible that the immune response in the primate brain was not strong enough to reject all of the donor cells.\n","[]\n","These findings closely correlate with the results of previous murine experiments (Hudson et al., 1994; Shinoda et al., 1995).\n","[]\n","To apply our findings to a more clinically relevant setting, we investigated the expression of HLA-I during neural differentiation of human ESCs (hESCs) and iPSCs by qPCR (Figure S4D).\n","[]\n","The expression level was 1/100 compared with that of human peripheral blood cells in both hESCs and iPSCs, and it was similarly elevated in response to IFN-Î³.\n","['The expression level']\n","It is difficult to precisely compare immunogenicity in monkeys with that in humans, but the low expression level of MHC-I by the donor cells may account for the mild rejection in both monkey and human neural transplantation.\n","[]\n","We next addressed whether hESC-derived trigeminal neurons can engraft in the adult mouse CNS and project toward their physiological target.\n","[]\n","The trigeminal nuclei in the brainstem receive afferent innervation from the trigeminal sensory ganglion that is relayed to the contralateral thalamus.\n","[]\n","The pons was selected as site for transplantation, because it is surgically accessible and located within proximity of the trigeminal brain stem nuclei that receive afferent input from the trigeminal ganglia.\n","[]\n","Hence, GFP+ human trigeminal neuron clusters were injected into adult NOD/SCID mice via stereotactic surgery (see Experimental Procedures).\n","[]\n","Histological analysis 4 weeks after transplantation showed survival of GFP+ human cell graft in the ventral pons (Figure S5E).\n","[]\n","Although GFP+ cell bodies remained tightly clustered at injection site, GFP+ fibers showed extensive projections into the host brain (n = 6) including the endogenous trigeminal nuclei (Figure S5F).\n","[]\n","Expression of BRN3A confirmed the sensory neuron identity of the cells (Figure S5G).\n","[]\n","Graft-derived human fiber bundles (hNCAM+ and GFP+) were observed emanating from the graft core (Figure S5H).\n","[]\n","These data demonstrate in vivo survival of trigeminal placode derivatives, differentiation along sensory neuron lineage, and the establishment of axonal projections toward relevant endogenous targets in the embryonic chick and adult mouse brain.\n","[]\n","(B) Fetal hRPE stained for SC121 (red) and MCT1 (green; scale bars, 125 Î¼m and 25 Î¼m [inset]).\n","[]\n","(F) Intracellular PGRN levels in iPSC-derived neurons after medium collection.\n","[]\n","Values of control line 20 were set to 100% (n = 3â€“4 independent cultures).\n","['Values', '20']\n","Prior to VESTAL analyses, a 5â€“55 Hz bandpass filter was applied.\n","[]\n","VESTAL analyses examined activity 30â€“130 ms post-stimulus producing a 4D activation map (3D volumes across time) as well as a 2D source time-course matrix.\n","[]\n","The average percent variance explained for gradiometer data using VESTAL was 95.81% for HC and 94.38% for SZ.\n","[]\n","The average percent variance explained for magnetometer data using VESTAL program was 96.24% for HC and 93.17% for SZ.\n","['percent variance']\n","There were no group differences in percent variance explained for gradiometer data (t(39) = 1.16, p = 0.25) or magnetometer data (t(39) = 1.31, p = 0.20).\n","[]\n","(C) Control lines 16 and 17, sporadic lines 12 and 23, and PGRN S116X lines 1 and 14 were immunostained for AFP (endoderm), desmin (mesoderm), and Î²III-tubulin (ectoderm), and counterstained with DAPI (nuclei).\n","['tain']\n","All lines showed a normal karyotype.\n","[]\n","Scale bar: 50 Î¼m.\n","['Scale bar']\n","(Iâ€“M) Teratoma formation at 3 months after transplantation in the testes of SCID mice.\n","[]\n","H&E staining of the sections showed histological features of the neuroepithelium (J), cartilage (K), muscle (L) and gut-like epithelium (M).\n","[]\n","Using a recombinant adeno-associated viral (rAAV)-based gene-targeting method, we inserted the gene-encoding GFP into the SOX2 locus in H9 hESCs (Figure 1A).\n","[]\n","Proper homologous recombination led to the replacement of the SOX2 open reading frame with that of GFP and a neomycin selection cassette (SV40-Neo).\n","[]\n","After infection with rAAV and G418 drug selection, a total of 36 clones were expanded and screened by Southern blotting for homologous recombination events.\n","[]\n","Among these clones, 26 (72%) were found to carry the GFP-Neo cassette in the SOX2 locus (Figure S1A available online).\n","['s', 'found', 'cassette']\n","No clones in which both SOX2 alleles were disrupted were isolated.\n","[]\n","Our subsequent analysis focused on one of these clones, clone 23 (hSOX2-23).\n","[]\n","We confirmed appropriate gene targeting in this clone using multiple restriction digests followed by Southern blotting (Figures 1B, S1B, and S1C).\n","[]\n","We did not observe nontargeted insertions of the rAAV sequences, and cells exhibited a normal karyotype (data not shown).\n","[]\n","Flow cytometry of hSOX2-23 revealed that the majority of the cells expressed GFP (Figure 1C).\n","[]\n","By comparison, a drug-selected clone, hSOX2-25, which was negative for targeted insertion (Figure S1A), showed no detectable GFP (Figure S2A).\n","['G']\n","Despite only having one copy of SOX2, hSOX2-23 had similar levels of SOX2, OCT4, and NANOG expression as hSOX2-25 and wild-type (WT) hESCs (Figure S2B).\n","[]\n","Moreover, the percentage of GFP-positive (GFP+) cells in hSOX2-23 was constant over more than 20 passages.\n","['GFP-positive', 'FP+', 'constant']\n","Immunofluorescence (IF) staining of hSOX2-23 showed that 100% of GFP+ cells expressed SOX2 protein (Figure S2C).\n","['GFP+']\n","Additionally, hSOX2-23 colonies had characteristic hESC morphology (Figure S2D) and expressed markers of the undifferentiated state, such as NANOG (Figure S2E).\n","[]\n","These results show that this rAAV-based gene-targeting method can be used to efficiently disrupt genes by homologous recombination.\n","[]\n","In addition, the SOX2-GFP hESC marker line can be used to monitor SOX2 expression in undifferentiated hESCs.\n","[]\n","Based on several previous publications (Khan et al., 2010, 2011), we explored the utility of adeno-associated virus (AAV) as a method to improve gene targeting efficiencies in hPSCs.\n","[]\n","For SOX2, a gene that is highly expressed in undifferentiated hPSCs, gene targeting rates were greater than 70%.\n","['gene targeting rates']\n","Similar targeting efficiencies in hPSCs using AAV have been reported by others (Asuri et al., 2012; Khan et al., 2010, 2011; Smith-Arica et al., 2003), indicating that AAV offers a highly efficient and robust approach to target genes for HR in hPSCs.\n","[]\n","(C) C-peptide and PDX1 expression was confirmed by immunocytochemistry of cells differentiated for 25 days.\n","['immunocytochemistry', 'cells differentiated']\n","â€¢Adult hRPESC-derived RPE had comparable in vitro characteristics to fetal hRPEâ€¢hRPE monolayers survived 4 weeks on PET carriers under the rabbit retinaâ€¢Better xenograft survival may be due to the maintained hRPE cell polarityâ€¢Atrophy of the retina overlaying the hRPE xenograft remains a future challenge\n","['survived']\n","(O) GH plasma levels using a human specific ELISA (6 weeks after transplantation).\n","[]\n","(R) Quantification of immunocytochemical analyses for each iPSC line.\n","[]\n","Data are shown as the means Â± SD (n = 3 independent experiments).\n","[]\n","SER, serotonin, TUBÎ²III, Î²-tubulin class III.\n","[', Î²-tubulin']\n","Scale bars: 200 Î¼m in (A)â€“(H), 50 Î¼m in insets of (C)â€“(F), 100 Î¼m in (J)â€“(M) and (Q).\n","[]\n","The two FTD patients under investigation in this study were part of a longitudinal dementia research program at the Memory and Aging Center, University of California, San Francisco.\n","[]\n","Both had an 8-year history of behavioral changes and memory impairment at the time of tissue collection for this study.\n","[]\n","One patient, a 67-year-old male with sporadic FTD, tested negative for mutations in GRN, MAPT, and C9ORF72.\n","['tested negative']\n","The other patient, a 64-year-old male with a significant family history of dementia, had behavioral variant FTD.\n","[]\n","MRI in this patient demonstrated severe bifrontal and temporal atrophy associated with gliosis in the frontal lobes (greater on the right).\n","[]\n","One year later, MRI scans showed progression of atrophy and gliosis.\n","[]\n","Genetic testing revealed a novel nonsense mutation in GRN, p.S116X (g.4627C > A, c.347C > A), which is predicted to result in a premature stop codon.\n","[]\n","Both FTD patients had parkinsonism, which is typical of all FTD patients with PGRN mutations.\n","[]\n","An age-matched subject, a clinically normal 64-year-old male with no mutations in GRN, MAPT, or C9ORF72, served as a control.\n","[]\n","(Qâ€“S) Magnetic resonance images of a representative animal (No.\n","[]\n","6, autograft) at 3 months after the transplant.\n","[]\n","The arrowheads indicate the directions of the cell injections.\n","[]\n","(Q) coronal, (R) axial, and (S) sagittal.\n","[]\n","The letter L indicates the left side.\n","[]\n","(A) Fetal hRPE stained for pan-cytokeratin (scale bar, 50 Î¼m); inset shows section overview stained with hematoxylin/eosin (scale bar, 200 Î¼m).\n","['stained for pan-cyt']\n","To assess the in vivo properties of hESC-derived trigeminal placode precursors, PIP-induced neuronal clusters, derived from a constitutively GFP-positive hESC line (Figures S5A and S5B), were injected into the developing chick embryo targeting the early trigeminal anlage at H&H stage 10â€“12 (Figure S5C).\n","[]\n","Human cells were identified based on GFP expression and use of human specific antibodies against cytoplasmic antigen (hCA).\n","[]\n","Two days after in ovo transplantation, surviving GFP+ cells were found dispersed in the area of the endogenous chick trigeminal ganglion (Figure 4P).\n","[]\n","We observed extensive GFP+ human fiber bundles coexpressing hCA and peripherin (Figures 4Q and 4R).\n","[]\n","In contrast, no hCA or peripherin expression was detected in the neural tube of the embryo (Figure S5D).\n","[]\n","The in vivo fiber outgrowth 2 days after transplantation was reminiscent of the extensive in vitro fiber outgrowth of replated trigeminal neuron clusters (Figure S5A).\n","['fiber outgro']\n","Peripherin expression in vivo (Figure 4S) confirmed the peripheral neuron identity of the grafted cells.\n","[]\n","For the first two animals (Nos.\n","[]\n","1 and 4), we established iPSCs from fibroblasts derived from the oral mucosa using retroviral vectors (Okita et al., 2011).\n","[]\n","For the other two animals (Nos.\n","[]\n","6 and 8), we used peripheral blood mononuclear cells (PBMCs) with nonintegrating episomal vectors (Okita et al., 2013).\n","[]\n","We selected the best clone from each animal according to the following criteria: a stable embryonic stem cell (ESC)-like morphology of the colonies after passaging, expression of pluripotent markers, few or no integrated transgenes (Figures 1Aâ€“1F; Figure S1 available online), and the potential for stable neural differentiation.\n","[]\n","A PCR analysis revealed that all of the clones with retroviral vectors showed apparent expression of remaining transgenes (Figures S1C and S1D), whereas the clones with episomal vectors never did (Figure S1F).\n","[]\n","To detect the iPSC-derived cells in a brain, we introduced GFP (Figures 1G and 1H).\n","[]\n","The selected clones of iPSCs had the potential to generate teratomas in the testes of a severe combined immunodeficiency (SCID) mouse within 12 weeks (Figures 1Iâ€“1M).\n","[]\n","(A) Flow-cytometric analyses for MHC-I (HLA-A, HLA-B, and HLA-C).\n","[]\n","Incubation of the cells with IFN-Î³ for 48 hr increased the MHC-I expression (green).\n","['IFN-Î³']\n","Our group has developed a defined culture system to direct the differentiation of hPSCs into a near-homogenous population of definitive endoderm (DE) cells that have the capacity to differentiate into hepatocytes and pancreatic progenitors (Brown et al., 2011; Cho et al., 2012; Rashid et al., 2010; Touboul et al., 2010; Vallier et al., 2009a; Yusa et al., 2011).\n","[]\n","Cells grown in these culture conditions successively express primitive streak markers (T and Mixl1), downregulate pluripotency markers (NANOG, SOX2, and POU5F1) and progressively upregulate definitive endoderm markers (CXCR4, FOXA2, GATA4, CERB, and SOX17) (Figures S1Aâ€“S1C available online).\n","[]\n","Flow cytometry analyses showed that 80% of the resulting DE population coexpresses CXCR4 and SOX17 (Figure S1D).\n","[]\n","Interestingly, the resulting population of DE cells is negative for genes marking the foregut (SOX2), the midgut/hindgut (CDX2), the pancreas (PDX1), the liver (AFP), and the lungs (HOXA1) (Figures S1E and S1F) This confirms that DE cells generated in vitro could correspond to early endoderm progenitor cells prior to anteroposterior patterning or organogenesis.\n","[]\n","In sequential PET studies, we observed increased uptake of [11C]PK11195 in one allograft (animal No.\n","['up']\n","10) at 3 months (Figures 3A and 3B).\n","[]\n","We could not detect any apparent uptake in the other animals or at any other time points (Figure S2).\n","['up']\n","Intriguingly, the serum level of IFN-Î³ temporarily increased at 2 months after the transplant in three animals (Figure 3C).\n","['IFN-Î³', 'increased']\n","An immunofluorescence study conducted at 3.5â€“4 months showed that MHC-II+ cells were more frequently found in allografts than in autografts, especially in the monkey with increased uptake of [11C]PK11195 (Figure 3D, No.\n","['up']\n","10).\n","[]\n","The MHC-II staining never overlapped with that of GFP of the donor cells (Figure 3F), whereas it generally overlapped with that of IBA1 (Figure 3G), indicating that MHC-II was expressed by host-derived microglia.\n","[]\n","Consistently, the number and density of IBA1+ cells were higher in allografts than in autografts (Figures 3E, 3H, and S4C).\n","[]\n","An increase in the expression of MHC might trigger the recruitment of circulating immune cells, including T cells.\n","[]\n","An immunofluorescence study revealed that more CD45+ cells (a marker for pan-leukocytes) accumulated in allografts compared with autografts (Figures 3I and 3J).\n","[]\n","Most of the CD45+ cells were CD3+ T cells, and 60% of them were CD8+ killer T cells (Figures 3K and 3L).\n","['CD', '+', '+', 'T']\n","These findings suggest that an acquired immune response was elicited only in the allografts in the primate brain.\n","[]\n","This study was approved by the Institutional Review Board and Ethics Committees of the University of California, San Francisco, and written informed consent was obtained in all cases.\n","[]\n","The patient with the PGRN S116X mutation followed the classic clinical progression for FTD and developed parkinsonism, as do all FTD patients with PGRN mutations, but he did not show typical features of PD dementia.\n","[]\n","The patient with sporadic FTD also showed parkinsonism.\n","[]\n","Skin biopsies were collected, cut into small pieces, and placed on culture dishes to allow the fibroblasts to expand.\n","[]\n","The cells were maintained in Dulbecco's modified Eagle's medium supplemented with 10% fetal bovine serum, 1X nonessential amino acids, and penicillin/streptomycin (100 U/ml).\n","['ssential amino acids, and penicillin/stre']\n","iPSCs were generated as described previously (Takahashi et al., 2007).\n","[]\n","Please see Supplemental Information for more details.\n","[]\n","(Dâ€“H) Adult hRPE stained for SC121 (red).\n","[]\n","(D) Adult hRPE stained for MCT1 (green).\n","[]\n","(E) Human adult RPE stained for ezrin (green).\n","[]\n","hRPEs transplanted into rabbit SRS show absence of expression of ki67 (F), phosphohistone H3 (G), and caspase-3 (H).\n","[]\n","Polarized fetal and adult hRPE cells were found in TEM (I and J).\n","[]\n","Nuclei with regular chromatin were found in the basal compartment, a basal lamina ([I], large black arrowhead) had formed between the xenograft and PET carrier (black asterisks).\n","[]\n","Melanosomes (M) in multiple stages, some microvilli abutting to the atrophic neural retina (NR), and junctional structures with desmosomes (small black arrowhead) and tight junctions (red arrowhead) were discerned apically.\n","[]\n","Mitochondria (MC) were seen in the basolateral part of the cell.\n","[]\n","Detachment from cell carrier (asterisk) in (J) is a histologic processing artifact.\n","[]\n","Left images in (I) and (J) taken at 10,500Ã— magnification; right micrographs are rectangular zone in left at 25,000Ã—; scale bars represent 2 Î¼m/inset 0.2 Î¼m distance in (I) and (J).\n","['magnification', 'distance']\n","iPSC colonies were detached with accutase (Millipore) and grown as embryoid bodies (EBs) in suspension for 5â€“6 days in iPSC medium without basic fibroblast growth factor.\n","['suspension']\n","EBs were allowed to attach and form rosettes.\n","[]\n","Ten-day-old rosettes were collected and grown in suspension as neurospheres.\n","[]\n","Neurospheres were dissociated after 3â€“4 weeks, and the cells were plated on glass coverslips (BD Biosciences) or plates coated with poly-D-lysine (0.1 mg/ml) and laminin (10 Î¼g/m).\n","['di', 'cia']\n","Neurons were used after 2-4 weeks in culture.\n","[]\n","For cell differentiation, FAP-specific iPS cells were cultured with serial changes of media as shown in Fig.\n","[]\n","3A.\n","[]\n","To test whether FAP-specific iPS cells can differentiate into hepatocyte-like cells, we analyzed several markers via real-time PCR analysis on Day 5 (D5), D13 and D20 differentiated FAP-specific iPS cells (Fig.\n","[]\n","3B).\n","[]\n","A decrease in expression of the pluripotency marker Oct3/4 was accompanied by differentiation of FAP-specific iPS cells.\n","[]\n","Expression of the endoderm marker Sox17 was observed on D5 differentiation and decreased gradually after the medium was changed to hepatic differentiation medium on D7 (Fig.\n","['endoderm marker Sox17', 'changed']\n","3B).\n","[]\n","The hepatic progenitor marker AFP and the mature hepatocyte marker ALB were obviously expressed on D13 and D20.\n","[]\n","In addition, immunocytochemical analyses showed Sox17 expression on D5, both HNF-4Î± and AFP expression on D13, and ALB cytoplasmic staining on D20 (Fig.\n","['17', 'ALB cytoplasmic stain']\n","3C).\n","[]\n","Quantitative imaging analysis revealed that approximately 78 Â± 0.6% of cells were Sox17-positive on D5 and approximately 88 Â± 1.1% of cells were AFP-positive on D13 and approximately 29 Â± 0.9% of cells were ALB-positive on D20 (Fig.\n","['cells were Sox17-positive', 'AFP-positive', 'ALB-positive']\n","3D).\n","[]\n","The ALB secretion in the media of differentiated FAP-specific iPS cells on D20 was approximately 20 Î¼g/ml (Fig.\n","['ALB secretion', 'media', 'differentiated FAP-specific iPS cells on D']\n","3E).\n","[]\n","Moreover, these D20 differentiated FAP-specific iPS cells were also periodic acid-Schiff (PAS)-positive, indicating cytoplasmic glycogen storage (Fig.\n","[]\n","3F).\n","[]\n","These results clearly indicated that FAP-specific iPS cells had the potential to differentiate into hepatocyte-like cells.\n","[]\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"345-i1CqD6zv"},"source":["Measured Entity"]},{"cell_type":"code","metadata":{"id":"yaau-0i4DjHP"},"source":["bert_model = torch.load(MEASURED_ENTITY_MODEL_PATH)\r\n","Measured_Entity=[]\r\n","for s in test_text:\r\n","  tokenized_text = tokenizer_BioBert.tokenize(s)\r\n","  test_inputs = tokenizer_BioBert(s,max_length = 512,padding='max_length',truncation=True, return_tensors=\"pt\")\r\n","  bert_model.eval()\r\n","  temp = bert_model(test_inputs['input_ids'].to(device), test_inputs['attention_mask'].to(device))>=0.30\r\n","  Measured_Entity.append(temp[0])"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"Undf0Qf4EBkV","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1608028021106,"user_tz":-330,"elapsed":11103,"user":{"displayName":"Akash Gangwar","photoUrl":"","userId":"02081727011117790241"}},"outputId":"78c9716b-9171-4f97-9d86-ae6788f9be8a"},"source":["Entity=[]\r\n","for i in range(len(test_text)):\r\n","  tokenized_text = tokenizer_BioBert.tokenize(test_text[i])\r\n","  txt=[]\r\n","  label=[]\r\n","  \r\n","  combine=\"\"\r\n","  for j in range(len(tokenized_text)):\r\n","    if(Measured_Entity[i][j+1] and Measured_Entity[i][j] and j!=0):\r\n","      label.append(tokenized_text[j])\r\n","      if(len(tokenized_text[j])>=2 and tokenized_text[j][:2]==\"##\"):\r\n","          combine=combine+tokenized_text[j][2:]\r\n","      \r\n","      else:\r\n","        if(len(combine)==0):\r\n","          combine=tokenized_text[j]\r\n","          continue\r\n","        \r\n","        if(tokenized_text[j] in (\"!\", \",\" ,\"\\'\" ,\";\" ,\"\\\"\", \".\", \"-\" ,\"?\",\"+\",\"-\",\"*\",\"/\",\"%\")):\r\n","          combine=combine+tokenized_text[j]\r\n","          continue\r\n","        if(tokenized_text[j] in (\"(\",\")\") or combine in (\"(\",\")\")):\r\n","          continue\r\n","  \r\n","        start=[m.start() for m in re.finditer(combine,test_text[i])]\r\n","        end=[m.start() for m in re.finditer(tokenized_text[j],test_text[i])]\r\n","        flag=-1\r\n","        \r\n","        for s in start:\r\n","          for e in end:\r\n","            if(flag!=-1):\r\n","              break\r\n","            \r\n","            \r\n","            if(e-s-len(combine)+1==1):\r\n","              flag=1\r\n","            elif(e-s-len(combine)+1==2):\r\n","              flag=2\r\n","        \r\n","        if(flag==2):\r\n","          combine=combine+\" \"+tokenized_text[j]\r\n","        elif(flag==1):\r\n","          combine=combine+tokenized_text[j]\r\n","          \r\n","        else:\r\n","          txt.append(combine)\r\n","          combine=\"\"\r\n","\r\n","    elif(Measured_Entity[i][j+1]):\r\n","      label.append(tokenized_text[j])\r\n","      \r\n","      if(len(combine)!=0):\r\n","        txt.append(combine)\r\n","      \r\n","      if(len(tokenized_text[j])>=2 and tokenized_text[j][:2]==\"##\"):\r\n","        tokenized_text[j]=tokenized_text[j][2:]\r\n","      \r\n","      combine=tokenized_text[j]\r\n","    else:\r\n","      if(len(combine)!=0):\r\n","        txt.append(combine)\r\n","      \r\n","      combine=\"\"\r\n","  if(len(combine)!=0):\r\n","        txt.append(combine)\r\n","  \r\n","  print(test_text[i])\r\n","  print(txt)  \r\n","  Entity.append(txt)\r\n","        \r\n"],"execution_count":null,"outputs":[{"output_type":"stream","text":["(F) Adult hRPE cultures display expression of markers typical of native RPE in their polarized localization.\n","[]\n","DAPI is cyan, whereas all other immunofluorescence is gold.\n","[]\n","Claudin 19, ezrin, ZO1, and MCT1 are preferentially located on the apical side.\n","[]\n","RPE65 and CRALBP are cytoplasmic.\n","[]\n","The scale bars represent 10 Î¼m.\n","['scale bars', '.']\n","GFP-expressing H9, BBHX8, and A1ATD-1 cells were generated by stable transfection using lipofectamine 2000 (Invitrogen) as described previously (Vallier et al., 2001).\n","['GFP-', ',']\n","GFP-positive cells were differentiated into foregut cells and then dissociated into single cells.\n","[]\n","An individually isolated GFP cell was then transferred into a well containing non-GFP-positive hFSCs.\n","['GFP cell']\n","Wells were visually inspected 12 hr after plating, and wells containing a single GFP-positive hFSC were selected for clonal expansion.\n","['Wells']\n","For group analyses, the following procedures were applied.\n","[]\n","(1) T1-weighted sMRIs from each subject (Fig.\n","['T1-weighted sMRIs']\n","1A) were registered to MNI space (Montreal Neurological Institute, MNI-152 atlas as in Fig.\n","[]\n","1B) using an affine transformation (FLIRTâ€“FMRIB's Linear Image Registration Tool) (Jenkinson and Smith, 2001) in FSL (www.fmrib.ox.ac.uk/fsl/).\n","[]\n","(2) The cortical (Fig.\n","[]\n","1C) and subcortical masks with pre-defined brain regions from the standard atlas were transferred to the individual's headspace (Fig.\n","[]\n","1D), using the inverse of the transformation obtained in the first step: the Harvard-Oxford Atlas, part of the FSL software with masks of 96 cortical gray-matter regions (48 regions in each hemisphere), 21 sub-cortical regions, and cerebellum, was used.\n","['regions']\n","(3) The regional masks were down-sampled to a cubic source grid with voxels of 5 mm per side (Fig.\n","['regional masks', 'source grid', 'voxels', 'side']\n","1E).\n","[]\n","(4) VESTAL MEG source imaging used the source grid from step 3.\n","[]\n","This step permits group-based analyses.\n","[]\n","In the shown example, MEG responses evoked by S1 localized to left and right Heschl's gyri (Fig.\n","['S1']\n","1F).\n","[]\n","(5) Finally, for regions of interest (ROIs), the source time course was obtained by summing activity from all ROI voxels.\n","['regions', 'interest', 'ROIs']\n","Fig.\n","[]\n","1H shows the time course from left Heschl's gyrus (dark blue region in Fig.\n","[]\n","1C and D).\n","[]\n","Another important finding is that, in spite of the immune responses mounted by the host brain, a substantial number of TH+ cells survived in the allografts.\n","[]\n","This is consistent with previous clinical reports of human fetal cell transplantation.\n","[]\n","Postmortem analyses of the patients revealed robust survival of DA neurons in spite of the fact that numerous immune cells were present around the graft (Kordower et al., 1997).\n","[]\n","In two double-blind clinical trials, immunosuppressive drugs were never used (Freed et al., 2001) or were withdrawn after 6 months (Olanow et al., 2003).\n","['double-blind clinical trials', 'immunosuppressive']\n","In these cases, the cells from multiple fetuses were used without HLA matching, but more than 50,000 TH+ cells had survived after several years.\n","['TH']\n","Our quantitative PCR (qPCR) study in vitro showed that the expression of MHC-I increased in response to IFN-Î³, but the expression level was still 1/10 that of untreated monkey peripheral blood cells (Figure 2B).\n","['HC', 'untreated monkey peripheral blood cells']\n","The in vivo studies revealed that the serum level of IFN-Î³ increased at 2 months, and CD45+ cells (including CD8+ cells) accumulated in the allografts 3.5â€“4 months after the transplant.\n","['IFN']\n","On the other hand, the levels of INF-Î³ in the cerebrospinal fluid (CSF) and the levels of tumor necrosis factor Î± (TNF-Î±) in both the serum and CSF were below the limit of detection by ELISA (data not shown).\n","[]\n","An immunofluorescence study did not reveal any apparent expression of MHC-I by the grafted cells (Figure S4A).\n","[]\n","Therefore, it is possible that the immune response in the primate brain was not strong enough to reject all of the donor cells.\n","[]\n","These findings closely correlate with the results of previous murine experiments (Hudson et al., 1994; Shinoda et al., 1995).\n","[]\n","To apply our findings to a more clinically relevant setting, we investigated the expression of HLA-I during neural differentiation of human ESCs (hESCs) and iPSCs by qPCR (Figure S4D).\n","[]\n","The expression level was 1/100 compared with that of human peripheral blood cells in both hESCs and iPSCs, and it was similarly elevated in response to IFN-Î³.\n","['expression', 'peripheral blood cells', 'hESCs and iPSCs']\n","It is difficult to precisely compare immunogenicity in monkeys with that in humans, but the low expression level of MHC-I by the donor cells may account for the mild rejection in both monkey and human neural transplantation.\n","[]\n","We next addressed whether hESC-derived trigeminal neurons can engraft in the adult mouse CNS and project toward their physiological target.\n","['eminal neurons']\n","The trigeminal nuclei in the brainstem receive afferent innervation from the trigeminal sensory ganglion that is relayed to the contralateral thalamus.\n","['trigeminal nuclei']\n","The pons was selected as site for transplantation, because it is surgically accessible and located within proximity of the trigeminal brain stem nuclei that receive afferent input from the trigeminal ganglia.\n","['pons']\n","Hence, GFP+ human trigeminal neuron clusters were injected into adult NOD/SCID mice via stereotactic surgery (see Experimental Procedures).\n","['GFP+ human', 'rigeminal neuron clusters']\n","Histological analysis 4 weeks after transplantation showed survival of GFP+ human cell graft in the ventral pons (Figure S5E).\n","[]\n","Although GFP+ cell bodies remained tightly clustered at injection site, GFP+ fibers showed extensive projections into the host brain (n = 6) including the endogenous trigeminal nuclei (Figure S5F).\n","['em']\n","Expression of BRN3A confirmed the sensory neuron identity of the cells (Figure S5G).\n","[]\n","Graft-derived human fiber bundles (hNCAM+ and GFP+) were observed emanating from the graft core (Figure S5H).\n","[]\n","These data demonstrate in vivo survival of trigeminal placode derivatives, differentiation along sensory neuron lineage, and the establishment of axonal projections toward relevant endogenous targets in the embryonic chick and adult mouse brain.\n","['em']\n","(B) Fetal hRPE stained for SC121 (red) and MCT1 (green; scale bars, 125 Î¼m and 25 Î¼m [inset]).\n","['Fetal hRPE']\n","(F) Intracellular PGRN levels in iPSC-derived neurons after medium collection.\n","[]\n","Values of control line 20 were set to 100% (n = 3â€“4 independent cultures).\n","['Value', 'control line 20', 'cultures']\n","Prior to VESTAL analyses, a 5â€“55 Hz bandpass filter was applied.\n","['pass filter']\n","VESTAL analyses examined activity 30â€“130 ms post-stimulus producing a 4D activation map (3D volumes across time) as well as a 2D source time-course matrix.\n","['VESTAL', 'volumes']\n","The average percent variance explained for gradiometer data using VESTAL was 95.81% for HC and 94.38% for SZ.\n","['HC', 'SZ']\n","The average percent variance explained for magnetometer data using VESTAL program was 96.24% for HC and 93.17% for SZ.\n","['ESTA', 'HC', 'SZ']\n","There were no group differences in percent variance explained for gradiometer data (t(39) = 1.16, p = 0.25) or magnetometer data (t(39) = 1.31, p = 0.20).\n","['no group differences', 'percent variance explained', 'grad', '39', 'magnetometer data']\n","(C) Control lines 16 and 17, sporadic lines 12 and 23, and PGRN S116X lines 1 and 14 were immunostained for AFP (endoderm), desmin (mesoderm), and Î²III-tubulin (ectoderm), and counterstained with DAPI (nuclei).\n","['Control lines']\n","All lines showed a normal karyotype.\n","['lines']\n","Scale bar: 50 Î¼m.\n","['Scale bar']\n","(Iâ€“M) Teratoma formation at 3 months after transplantation in the testes of SCID mice.\n","[]\n","H&E staining of the sections showed histological features of the neuroepithelium (J), cartilage (K), muscle (L) and gut-like epithelium (M).\n","[]\n","Using a recombinant adeno-associated viral (rAAV)-based gene-targeting method, we inserted the gene-encoding GFP into the SOX2 locus in H9 hESCs (Figure 1A).\n","['H9 hESCs']\n","Proper homologous recombination led to the replacement of the SOX2 open reading frame with that of GFP and a neomycin selection cassette (SV40-Neo).\n","[]\n","After infection with rAAV and G418 drug selection, a total of 36 clones were expanded and screened by Southern blotting for homologous recombination events.\n","['36 clones']\n","Among these clones, 26 (72%) were found to carry the GFP-Neo cassette in the SOX2 locus (Figure S1A available online).\n","['clones']\n","No clones in which both SOX2 alleles were disrupted were isolated.\n","[]\n","Our subsequent analysis focused on one of these clones, clone 23 (hSOX2-23).\n","['these clones, clone 23', 'SO']\n","We confirmed appropriate gene targeting in this clone using multiple restriction digests followed by Southern blotting (Figures 1B, S1B, and S1C).\n","[]\n","We did not observe nontargeted insertions of the rAAV sequences, and cells exhibited a normal karyotype (data not shown).\n","[]\n","Flow cytometry of hSOX2-23 revealed that the majority of the cells expressed GFP (Figure 1C).\n","['SO', 'cells']\n","By comparison, a drug-selected clone, hSOX2-25, which was negative for targeted insertion (Figure S1A), showed no detectable GFP (Figure S2A).\n","['drug-selected clone, hSO']\n","Despite only having one copy of SOX2, hSOX2-23 had similar levels of SOX2, OCT4, and NANOG expression as hSOX2-25 and wild-type (WT) hESCs (Figure S2B).\n","['of SOX2, hSOX2', 'hSOX2-25 and wild-type']\n","Moreover, the percentage of GFP-positive (GFP+) cells in hSOX2-23 was constant over more than 20 passages.\n","['GFP-positive', 'FP+', 'in hSOX2-23']\n","Immunofluorescence (IF) staining of hSOX2-23 showed that 100% of GFP+ cells expressed SOX2 protein (Figure S2C).\n","['unofluores', 'SOX2-', 'GFP+ cells']\n","Additionally, hSOX2-23 colonies had characteristic hESC morphology (Figure S2D) and expressed markers of the undifferentiated state, such as NANOG (Figure S2E).\n","['X2-']\n","These results show that this rAAV-based gene-targeting method can be used to efficiently disrupt genes by homologous recombination.\n","[]\n","In addition, the SOX2-GFP hESC marker line can be used to monitor SOX2 expression in undifferentiated hESCs.\n","[]\n","Based on several previous publications (Khan et al., 2010, 2011), we explored the utility of adeno-associated virus (AAV) as a method to improve gene targeting efficiencies in hPSCs.\n","[]\n","For SOX2, a gene that is highly expressed in undifferentiated hPSCs, gene targeting rates were greater than 70%.\n","[]\n","Similar targeting efficiencies in hPSCs using AAV have been reported by others (Asuri et al., 2012; Khan et al., 2010, 2011; Smith-Arica et al., 2003), indicating that AAV offers a highly efficient and robust approach to target genes for HR in hPSCs.\n","[]\n","(C) C-peptide and PDX1 expression was confirmed by immunocytochemistry of cells differentiated for 25 days.\n","[]\n","â€¢Adult hRPESC-derived RPE had comparable in vitro characteristics to fetal hRPEâ€¢hRPE monolayers survived 4 weeks on PET carriers under the rabbit retinaâ€¢Better xenograft survival may be due to the maintained hRPE cell polarityâ€¢Atrophy of the retina overlaying the hRPE xenograft remains a future challenge\n","['E', 'RPE monolayers']\n","(O) GH plasma levels using a human specific ELISA (6 weeks after transplantation).\n","[]\n","(R) Quantification of immunocytochemical analyses for each iPSC line.\n","[]\n","Data are shown as the means Â± SD (n = 3 independent experiments).\n","['Data', '3 independent experiments']\n","SER, serotonin, TUBÎ²III, Î²-tubulin class III.\n","[]\n","Scale bars: 200 Î¼m in (A)â€“(H), 50 Î¼m in insets of (C)â€“(F), 100 Î¼m in (J)â€“(M) and (Q).\n","['bars', '(', 'insets of']\n","The two FTD patients under investigation in this study were part of a longitudinal dementia research program at the Memory and Aging Center, University of California, San Francisco.\n","['FTD patients']\n","Both had an 8-year history of behavioral changes and memory impairment at the time of tissue collection for this study.\n","[]\n","One patient, a 67-year-old male with sporadic FTD, tested negative for mutations in GRN, MAPT, and C9ORF72.\n","['patient,']\n","The other patient, a 64-year-old male with a significant family history of dementia, had behavioral variant FTD.\n","['patient,']\n","MRI in this patient demonstrated severe bifrontal and temporal atrophy associated with gliosis in the frontal lobes (greater on the right).\n","[]\n","One year later, MRI scans showed progression of atrophy and gliosis.\n","[]\n","Genetic testing revealed a novel nonsense mutation in GRN, p.S116X (g.4627C > A, c.347C > A), which is predicted to result in a premature stop codon.\n","[]\n","Both FTD patients had parkinsonism, which is typical of all FTD patients with PGRN mutations.\n","['FT']\n","An age-matched subject, a clinically normal 64-year-old male with no mutations in GRN, MAPT, or C9ORF72, served as a control.\n","['subject,']\n","(Qâ€“S) Magnetic resonance images of a representative animal (No.\n","['a representative animal']\n","6, autograft) at 3 months after the transplant.\n","[]\n","The arrowheads indicate the directions of the cell injections.\n","[]\n","(Q) coronal, (R) axial, and (S) sagittal.\n","[]\n","The letter L indicates the left side.\n","[]\n","(A) Fetal hRPE stained for pan-cytokeratin (scale bar, 50 Î¼m); inset shows section overview stained with hematoxylin/eosin (scale bar, 200 Î¼m).\n","['Fetal hRPE stained']\n","To assess the in vivo properties of hESC-derived trigeminal placode precursors, PIP-induced neuronal clusters, derived from a constitutively GFP-positive hESC line (Figures S5A and S5B), were injected into the developing chick embryo targeting the early trigeminal anlage at H&H stage 10â€“12 (Figure S5C).\n","['ne', 'clusters', 'eminal anlage']\n","Human cells were identified based on GFP expression and use of human specific antibodies against cytoplasmic antigen (hCA).\n","['cells']\n","Two days after in ovo transplantation, surviving GFP+ cells were found dispersed in the area of the endogenous chick trigeminal ganglion (Figure 4P).\n","['transplant', 'G']\n","We observed extensive GFP+ human fiber bundles coexpressing hCA and peripherin (Figures 4Q and 4R).\n","[]\n","In contrast, no hCA or peripherin expression was detected in the neural tube of the embryo (Figure S5D).\n","[]\n","The in vivo fiber outgrowth 2 days after transplantation was reminiscent of the extensive in vitro fiber outgrowth of replated trigeminal neuron clusters (Figure S5A).\n","[]\n","Peripherin expression in vivo (Figure 4S) confirmed the peripheral neuron identity of the grafted cells.\n","[]\n","For the first two animals (Nos.\n","['two animals']\n","1 and 4), we established iPSCs from fibroblasts derived from the oral mucosa using retroviral vectors (Okita et al., 2011).\n","[]\n","For the other two animals (Nos.\n","['two animals']\n","6 and 8), we used peripheral blood mononuclear cells (PBMCs) with nonintegrating episomal vectors (Okita et al., 2013).\n","['r cells']\n","We selected the best clone from each animal according to the following criteria: a stable embryonic stem cell (ESC)-like morphology of the colonies after passaging, expression of pluripotent markers, few or no integrated transgenes (Figures 1Aâ€“1F; Figure S1 available online), and the potential for stable neural differentiation.\n","[]\n","A PCR analysis revealed that all of the clones with retroviral vectors showed apparent expression of remaining transgenes (Figures S1C and S1D), whereas the clones with episomal vectors never did (Figure S1F).\n","['clones']\n","To detect the iPSC-derived cells in a brain, we introduced GFP (Figures 1G and 1H).\n","[]\n","The selected clones of iPSCs had the potential to generate teratomas in the testes of a severe combined immunodeficiency (SCID) mouse within 12 weeks (Figures 1Iâ€“1M).\n","['deficiency']\n","(A) Flow-cytometric analyses for MHC-I (HLA-A, HLA-B, and HLA-C).\n","[]\n","Incubation of the cells with IFN-Î³ for 48 hr increased the MHC-I expression (green).\n","['cells with IF']\n","Our group has developed a defined culture system to direct the differentiation of hPSCs into a near-homogenous population of definitive endoderm (DE) cells that have the capacity to differentiate into hepatocytes and pancreatic progenitors (Brown et al., 2011; Cho et al., 2012; Rashid et al., 2010; Touboul et al., 2010; Vallier et al., 2009a; Yusa et al., 2011).\n","['definitive endoderm']\n","Cells grown in these culture conditions successively express primitive streak markers (T and Mixl1), downregulate pluripotency markers (NANOG, SOX2, and POU5F1) and progressively upregulate definitive endoderm markers (CXCR4, FOXA2, GATA4, CERB, and SOX17) (Figures S1Aâ€“S1C available online).\n","[]\n","Flow cytometry analyses showed that 80% of the resulting DE population coexpresses CXCR4 and SOX17 (Figure S1D).\n","['Flow cytome', 'resulting DE population']\n","Interestingly, the resulting population of DE cells is negative for genes marking the foregut (SOX2), the midgut/hindgut (CDX2), the pancreas (PDX1), the liver (AFP), and the lungs (HOXA1) (Figures S1E and S1F) This confirms that DE cells generated in vitro could correspond to early endoderm progenitor cells prior to anteroposterior patterning or organogenesis.\n","[]\n","In sequential PET studies, we observed increased uptake of [11C]PK11195 in one allograft (animal No.\n","['PET studies', 'allograft']\n","10) at 3 months (Figures 3A and 3B).\n","[]\n","We could not detect any apparent uptake in the other animals or at any other time points (Figure S2).\n","[]\n","Intriguingly, the serum level of IFN-Î³ temporarily increased at 2 months after the transplant in three animals (Figure 3C).\n","['IF', 'animals']\n","An immunofluorescence study conducted at 3.5â€“4 months showed that MHC-II+ cells were more frequently found in allografts than in autografts, especially in the monkey with increased uptake of [11C]PK11195 (Figure 3D, No.\n","[]\n","10).\n","[]\n","The MHC-II staining never overlapped with that of GFP of the donor cells (Figure 3F), whereas it generally overlapped with that of IBA1 (Figure 3G), indicating that MHC-II was expressed by host-derived microglia.\n","[]\n","Consistently, the number and density of IBA1+ cells were higher in allografts than in autografts (Figures 3E, 3H, and S4C).\n","[]\n","An increase in the expression of MHC might trigger the recruitment of circulating immune cells, including T cells.\n","[]\n","An immunofluorescence study revealed that more CD45+ cells (a marker for pan-leukocytes) accumulated in allografts compared with autografts (Figures 3I and 3J).\n","[]\n","Most of the CD45+ cells were CD3+ T cells, and 60% of them were CD8+ killer T cells (Figures 3K and 3L).\n","[]\n","These findings suggest that an acquired immune response was elicited only in the allografts in the primate brain.\n","[]\n","This study was approved by the Institutional Review Board and Ethics Committees of the University of California, San Francisco, and written informed consent was obtained in all cases.\n","[]\n","The patient with the PGRN S116X mutation followed the classic clinical progression for FTD and developed parkinsonism, as do all FTD patients with PGRN mutations, but he did not show typical features of PD dementia.\n","['PGRN S11']\n","The patient with sporadic FTD also showed parkinsonism.\n","['poradic FTD']\n","Skin biopsies were collected, cut into small pieces, and placed on culture dishes to allow the fibroblasts to expand.\n","[]\n","The cells were maintained in Dulbecco's modified Eagle's medium supplemented with 10% fetal bovine serum, 1X nonessential amino acids, and penicillin/streptomycin (100 U/ml).\n","['cells', \"Dulbecco's modified Eagle's medium supplemented\", 'etal bovine serum', 'nonessential amino acids,', 'ici']\n","iPSCs were generated as described previously (Takahashi et al., 2007).\n","[]\n","Please see Supplemental Information for more details.\n","[]\n","(Dâ€“H) Adult hRPE stained for SC121 (red).\n","[]\n","(D) Adult hRPE stained for MCT1 (green).\n","[]\n","(E) Human adult RPE stained for ezrin (green).\n","['adult RPE']\n","hRPEs transplanted into rabbit SRS show absence of expression of ki67 (F), phosphohistone H3 (G), and caspase-3 (H).\n","[]\n","Polarized fetal and adult hRPE cells were found in TEM (I and J).\n","[]\n","Nuclei with regular chromatin were found in the basal compartment, a basal lamina ([I], large black arrowhead) had formed between the xenograft and PET carrier (black asterisks).\n","[]\n","Melanosomes (M) in multiple stages, some microvilli abutting to the atrophic neural retina (NR), and junctional structures with desmosomes (small black arrowhead) and tight junctions (red arrowhead) were discerned apically.\n","[]\n","Mitochondria (MC) were seen in the basolateral part of the cell.\n","[]\n","Detachment from cell carrier (asterisk) in (J) is a histologic processing artifact.\n","[]\n","Left images in (I) and (J) taken at 10,500Ã— magnification; right micrographs are rectangular zone in left at 25,000Ã—; scale bars represent 2 Î¼m/inset 0.2 Î¼m distance in (I) and (J).\n","['Left images', 'right micrographs', '/inset']\n","iPSC colonies were detached with accutase (Millipore) and grown as embryoid bodies (EBs) in suspension for 5â€“6 days in iPSC medium without basic fibroblast growth factor.\n","[]\n","EBs were allowed to attach and form rosettes.\n","[]\n","Ten-day-old rosettes were collected and grown in suspension as neurospheres.\n","['rosettes']\n","Neurospheres were dissociated after 3â€“4 weeks, and the cells were plated on glass coverslips (BD Biosciences) or plates coated with poly-D-lysine (0.1 mg/ml) and laminin (10 Î¼g/m).\n","[]\n","Neurons were used after 2-4 weeks in culture.\n","['Neurons']\n","For cell differentiation, FAP-specific iPS cells were cultured with serial changes of media as shown in Fig.\n","[]\n","3A.\n","[]\n","To test whether FAP-specific iPS cells can differentiate into hepatocyte-like cells, we analyzed several markers via real-time PCR analysis on Day 5 (D5), D13 and D20 differentiated FAP-specific iPS cells (Fig.\n","['PS cells']\n","3B).\n","[]\n","A decrease in expression of the pluripotency marker Oct3/4 was accompanied by differentiation of FAP-specific iPS cells.\n","[]\n","Expression of the endoderm marker Sox17 was observed on D5 differentiation and decreased gradually after the medium was changed to hepatic differentiation medium on D7 (Fig.\n","[]\n","3B).\n","[]\n","The hepatic progenitor marker AFP and the mature hepatocyte marker ALB were obviously expressed on D13 and D20.\n","[]\n","In addition, immunocytochemical analyses showed Sox17 expression on D5, both HNF-4Î± and AFP expression on D13, and ALB cytoplasmic staining on D20 (Fig.\n","[]\n","3C).\n","[]\n","Quantitative imaging analysis revealed that approximately 78 Â± 0.6% of cells were Sox17-positive on D5 and approximately 88 Â± 1.1% of cells were AFP-positive on D13 and approximately 29 Â± 0.9% of cells were ALB-positive on D20 (Fig.\n","['uantitative imaging analysis', 'cells', '17', 'cells', 'cells']\n","3D).\n","[]\n","The ALB secretion in the media of differentiated FAP-specific iPS cells on D20 was approximately 20 Î¼g/ml (Fig.\n","[]\n","3E).\n","[]\n","Moreover, these D20 differentiated FAP-specific iPS cells were also periodic acid-Schiff (PAS)-positive, indicating cytoplasmic glycogen storage (Fig.\n","[]\n","3F).\n","[]\n","These results clearly indicated that FAP-specific iPS cells had the potential to differentiate into hepatocyte-like cells.\n","[]\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"kIp0vLvDEag2"},"source":["Quantity"]},{"cell_type":"code","metadata":{"id":"WvR3aM47ELAs","colab":{"base_uri":"https://localhost:8080/","height":100,"referenced_widgets":["80bfb292f3cd4d79a710be148ac5d740","c666525d535e4626b8e55900c77e677b","80be2dfba24641a6be4ebc43c33554d6","f182eb364fca48d59d0cdc609a54a5b1","43b0937d8fa643f78490f55a387a6677","afc82981315f4ab89e515075ed3c3c91","c9f5f1994985410c8eed0f9406423f14","5acfe71daf5d42b093ccc963b6d65b94","482ce836853e40d3a3836257a5cc7293"]},"executionInfo":{"status":"ok","timestamp":1608028048262,"user_tz":-330,"elapsed":8957,"user":{"displayName":"Akash Gangwar","photoUrl":"","userId":"02081727011117790241"}},"outputId":"a8922607-54fe-4575-81b9-f64d139d8617"},"source":["tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')\r\n","model = BertModel.from_pretrained('bert-base-uncased')"],"execution_count":null,"outputs":[{"output_type":"display_data","data":{"application/vnd.jupyter.widget-view+json":{"model_id":"80bfb292f3cd4d79a710be148ac5d740","version_minor":0,"version_major":2},"text/plain":["HBox(children=(FloatProgress(value=0.0, description='Downloading', max=433.0, style=ProgressStyle(description_â€¦"]},"metadata":{"tags":[]}},{"output_type":"stream","text":["\n"],"name":"stdout"},{"output_type":"display_data","data":{"application/vnd.jupyter.widget-view+json":{"model_id":"c666525d535e4626b8e55900c77e677b","version_minor":0,"version_major":2},"text/plain":["HBox(children=(FloatProgress(value=0.0, description='Downloading', max=440473133.0, style=ProgressStyle(descriâ€¦"]},"metadata":{"tags":[]}},{"output_type":"stream","text":["\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"WKwv9_zmEeIQ"},"source":["bert_model = BERT_Arch(model, 768, 256, 0.5, 1,1)\r\n","bert_model = bert_model.to(device)\r\n","bert_model = torch.load(MEASURED_QUANTITY_MODEL_PATH)\r\n","Measured_Quantity=[]\r\n","for s in test_text:\r\n","  tokenized_text = tokenizer.tokenize(s)\r\n","  test_inputs = tokenizer(s,max_length = 512,padding='max_length',truncation=True, return_tensors=\"pt\")\r\n","  bert_model.eval()\r\n","  temp = bert_model(test_inputs['input_ids'].to(device), test_inputs['attention_mask'].to(device))>=0.30\r\n","  Measured_Quantity.append(temp[0])\r\n","  "],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"Nsii1AlrEuqm","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1608028062114,"user_tz":-330,"elapsed":16107,"user":{"displayName":"Akash Gangwar","photoUrl":"","userId":"02081727011117790241"}},"outputId":"7228b951-6104-4c5f-b963-ee08d2839e97"},"source":["Quantity=[]\r\n","for i in range(len(test_text)):\r\n","  tokenized_text = tokenizer.tokenize(test_text[i])\r\n","  txt=[]\r\n","  label=[]\r\n","  \r\n","  combine=\"\"\r\n","  for j in range(len(tokenized_text)):\r\n","    if(Measured_Quantity[i][j+1] and Measured_Quantity[i][j] and j!=0):\r\n","      label.append(tokenized_text[j])\r\n","      if(len(tokenized_text[j])>=2 and tokenized_text[j][:2]==\"##\"):\r\n","          combine=combine+tokenized_text[j][2:]\r\n","      \r\n","      else:\r\n","        if(len(combine)==0):\r\n","          combine=tokenized_text[j]\r\n","          continue\r\n","        \r\n","        if(tokenized_text[j] in (\"!\", \",\" ,\"\\'\" ,\";\" ,\"\\\"\", \".\", \"-\" ,\"?\",\"+\",\"-\",\"*\",\"/\",\"%\")):\r\n","          combine=combine+tokenized_text[j]\r\n","          continue\r\n","        if(tokenized_text[j] in (\"(\",\")\") or combine in (\"(\",\")\")):\r\n","          print(combine)\r\n","          continue\r\n","        start=[m.start() for m in re.finditer(combine,test_text[i])]\r\n","        end=[m.start() for m in re.finditer(tokenized_text[j],test_text[i])]\r\n","        flag=-1\r\n","        \r\n","        for s in start:\r\n","          for e in end:\r\n","            if(flag!=-1):\r\n","              break\r\n","            \r\n","            \r\n","            if(e-s-len(combine)+1==1):\r\n","              flag=1\r\n","            elif(e-s-len(combine)+1==2):\r\n","              flag=2\r\n","        \r\n","        if(flag==2):\r\n","          combine=combine+\" \"+tokenized_text[j]\r\n","        elif(flag==1):\r\n","          combine=combine+tokenized_text[j]\r\n","          \r\n","        else:\r\n","          txt.append(combine)\r\n","          combine=\"\"\r\n","\r\n","    elif(Measured_Quantity[i][j+1]):\r\n","      label.append(tokenized_text[j])\r\n","      \r\n","      if(len(combine)!=0):\r\n","        txt.append(combine)\r\n","      \r\n","      if(len(tokenized_text[j])>=2 and tokenized_text[j][:2]==\"##\"):\r\n","        tokenized_text[j]=tokenized_text[j][2:]\r\n","      \r\n","      combine=tokenized_text[j]\r\n","    else:\r\n","      if(len(combine)!=0):\r\n","        txt.append(combine)\r\n","      \r\n","      combine=\"\"\r\n","        \r\n","  if(len(combine)!=0):\r\n","        txt.append(combine)\r\n","  \r\n","               \r\n","  print(test_text[i])\r\n","  print(txt)\r\n","  Quantity.append(txt)\r\n","  \r\n","  "],"execution_count":null,"outputs":[{"output_type":"stream","text":["(F) Adult hRPE cultures display expression of markers typical of native RPE in their polarized localization.\n","[]\n","DAPI is cyan, whereas all other immunofluorescence is gold.\n","[]\n","Claudin 19, ezrin, ZO1, and MCT1 are preferentially located on the apical side.\n","[]\n","RPE65 and CRALBP are cytoplasmic.\n","[]\n","The scale bars represent 10 Î¼m.\n","['10 Î¼m.']\n","GFP-expressing H9, BBHX8, and A1ATD-1 cells were generated by stable transfection using lipofectamine 2000 (Invitrogen) as described previously (Vallier et al., 2001).\n","[]\n","GFP-positive cells were differentiated into foregut cells and then dissociated into single cells.\n","[]\n","An individually isolated GFP cell was then transferred into a well containing non-GFP-positive hFSCs.\n","[]\n","Wells were visually inspected 12 hr after plating, and wells containing a single GFP-positive hFSC were selected for clonal expansion.\n","['12 hr']\n","For group analyses, the following procedures were applied.\n","[]\n","(1) T1-weighted sMRIs from each subject (Fig.\n","[]\n","1A) were registered to MNI space (Montreal Neurological Institute, MNI-152 atlas as in Fig.\n","[]\n","1B) using an affine transformation (FLIRTâ€“FMRIB's Linear Image Registration Tool) (Jenkinson and Smith, 2001) in FSL (www.fmrib.ox.ac.uk/fsl/).\n","[]\n","(2) The cortical (Fig.\n","[]\n","1C) and subcortical masks with pre-defined brain regions from the standard atlas were transferred to the individual's headspace (Fig.\n","[]\n","1D), using the inverse of the transformation obtained in the first step: the Harvard-Oxford Atlas, part of the FSL software with masks of 96 cortical gray-matter regions (48 regions in each hemisphere), 21 sub-cortical regions, and cerebellum, was used.\n","['96']\n","(3) The regional masks were down-sampled to a cubic source grid with voxels of 5 mm per side (Fig.\n","['5 mm per side']\n","1E).\n","[]\n","(4) VESTAL MEG source imaging used the source grid from step 3.\n","[]\n","This step permits group-based analyses.\n","[]\n","In the shown example, MEG responses evoked by S1 localized to left and right Heschl's gyri (Fig.\n","[]\n","1F).\n","[]\n","(5) Finally, for regions of interest (ROIs), the source time course was obtained by summing activity from all ROI voxels.\n","[]\n","Fig.\n","[]\n","1H shows the time course from left Heschl's gyrus (dark blue region in Fig.\n","[]\n","1C and D).\n","[]\n","Another important finding is that, in spite of the immune responses mounted by the host brain, a substantial number of TH+ cells survived in the allografts.\n","[]\n","This is consistent with previous clinical reports of human fetal cell transplantation.\n","[]\n","Postmortem analyses of the patients revealed robust survival of DA neurons in spite of the fact that numerous immune cells were present around the graft (Kordower et al., 1997).\n","[]\n","In two double-blind clinical trials, immunosuppressive drugs were never used (Freed et al., 2001) or were withdrawn after 6 months (Olanow et al., 2003).\n","['two', 'after 6 months']\n","In these cases, the cells from multiple fetuses were used without HLA matching, but more than 50,000 TH+ cells had survived after several years.\n","['more than 50,000']\n","Our quantitative PCR (qPCR) study in vitro showed that the expression of MHC-I increased in response to IFN-Î³, but the expression level was still 1/10 that of untreated monkey peripheral blood cells (Figure 2B).\n","['1/10 that']\n","The in vivo studies revealed that the serum level of IFN-Î³ increased at 2 months, and CD45+ cells (including CD8+ cells) accumulated in the allografts 3.5â€“4 months after the transplant.\n","['2 months,', '3.5â€“4 months']\n","On the other hand, the levels of INF-Î³ in the cerebrospinal fluid (CSF) and the levels of tumor necrosis factor Î± (TNF-Î±) in both the serum and CSF were below the limit of detection by ELISA (data not shown).\n","[]\n","An immunofluorescence study did not reveal any apparent expression of MHC-I by the grafted cells (Figure S4A).\n","[]\n","Therefore, it is possible that the immune response in the primate brain was not strong enough to reject all of the donor cells.\n","[]\n","These findings closely correlate with the results of previous murine experiments (Hudson et al., 1994; Shinoda et al., 1995).\n","[]\n","To apply our findings to a more clinically relevant setting, we investigated the expression of HLA-I during neural differentiation of human ESCs (hESCs) and iPSCs by qPCR (Figure S4D).\n","[]\n","The expression level was 1/100 compared with that of human peripheral blood cells in both hESCs and iPSCs, and it was similarly elevated in response to IFN-Î³.\n","['1/100']\n","It is difficult to precisely compare immunogenicity in monkeys with that in humans, but the low expression level of MHC-I by the donor cells may account for the mild rejection in both monkey and human neural transplantation.\n","[]\n","We next addressed whether hESC-derived trigeminal neurons can engraft in the adult mouse CNS and project toward their physiological target.\n","[]\n","The trigeminal nuclei in the brainstem receive afferent innervation from the trigeminal sensory ganglion that is relayed to the contralateral thalamus.\n","[]\n","The pons was selected as site for transplantation, because it is surgically accessible and located within proximity of the trigeminal brain stem nuclei that receive afferent input from the trigeminal ganglia.\n","[]\n","Hence, GFP+ human trigeminal neuron clusters were injected into adult NOD/SCID mice via stereotactic surgery (see Experimental Procedures).\n","[]\n","Histological analysis 4 weeks after transplantation showed survival of GFP+ human cell graft in the ventral pons (Figure S5E).\n","['4 weeks']\n","Although GFP+ cell bodies remained tightly clustered at injection site, GFP+ fibers showed extensive projections into the host brain (n = 6) including the endogenous trigeminal nuclei (Figure S5F).\n","['6']\n","Expression of BRN3A confirmed the sensory neuron identity of the cells (Figure S5G).\n","[]\n","Graft-derived human fiber bundles (hNCAM+ and GFP+) were observed emanating from the graft core (Figure S5H).\n","[]\n","These data demonstrate in vivo survival of trigeminal placode derivatives, differentiation along sensory neuron lineage, and the establishment of axonal projections toward relevant endogenous targets in the embryonic chick and adult mouse brain.\n","[]\n","(B) Fetal hRPE stained for SC121 (red) and MCT1 (green; scale bars, 125 Î¼m and 25 Î¼m [inset]).\n","['125 Î¼m', '25 Î¼m']\n","(F) Intracellular PGRN levels in iPSC-derived neurons after medium collection.\n","[]\n","Values of control line 20 were set to 100% (n = 3â€“4 independent cultures).\n","['100%', '3â€“4']\n","Prior to VESTAL analyses, a 5â€“55 Hz bandpass filter was applied.\n","['5â€“55']\n","VESTAL analyses examined activity 30â€“130 ms post-stimulus producing a 4D activation map (3D volumes across time) as well as a 2D source time-course matrix.\n","['30â€“130 ms', '4d']\n","The average percent variance explained for gradiometer data using VESTAL was 95.81% for HC and 94.38% for SZ.\n","['95.81%', '94.38%']\n","The average percent variance explained for magnetometer data using VESTAL program was 96.24% for HC and 93.17% for SZ.\n","['96.24%', '93.17%']\n","0.25\n","There were no group differences in percent variance explained for gradiometer data (t(39) = 1.16, p = 0.25) or magnetometer data (t(39) = 1.31, p = 0.20).\n","['1.16,', '0.25', '1.31,']\n","(C) Control lines 16 and 17, sporadic lines 12 and 23, and PGRN S116X lines 1 and 14 were immunostained for AFP (endoderm), desmin (mesoderm), and Î²III-tubulin (ectoderm), and counterstained with DAPI (nuclei).\n","[]\n","All lines showed a normal karyotype.\n","[]\n","Scale bar: 50 Î¼m.\n","['50 Î¼m.']\n","(Iâ€“M) Teratoma formation at 3 months after transplantation in the testes of SCID mice.\n","['3 months']\n","H&E staining of the sections showed histological features of the neuroepithelium (J), cartilage (K), muscle (L) and gut-like epithelium (M).\n","[]\n","Using a recombinant adeno-associated viral (rAAV)-based gene-targeting method, we inserted the gene-encoding GFP into the SOX2 locus in H9 hESCs (Figure 1A).\n","[]\n","Proper homologous recombination led to the replacement of the SOX2 open reading frame with that of GFP and a neomycin selection cassette (SV40-Neo).\n","[]\n","After infection with rAAV and G418 drug selection, a total of 36 clones were expanded and screened by Southern blotting for homologous recombination events.\n","['36']\n","Among these clones, 26 (72%) were found to carry the GFP-Neo cassette in the SOX2 locus (Figure S1A available online).\n","['26']\n","No clones in which both SOX2 alleles were disrupted were isolated.\n","[]\n","Our subsequent analysis focused on one of these clones, clone 23 (hSOX2-23).\n","[]\n","We confirmed appropriate gene targeting in this clone using multiple restriction digests followed by Southern blotting (Figures 1B, S1B, and S1C).\n","[]\n","We did not observe nontargeted insertions of the rAAV sequences, and cells exhibited a normal karyotype (data not shown).\n","[]\n","Flow cytometry of hSOX2-23 revealed that the majority of the cells expressed GFP (Figure 1C).\n","[]\n","By comparison, a drug-selected clone, hSOX2-25, which was negative for targeted insertion (Figure S1A), showed no detectable GFP (Figure S2A).\n","[]\n","Despite only having one copy of SOX2, hSOX2-23 had similar levels of SOX2, OCT4, and NANOG expression as hSOX2-25 and wild-type (WT) hESCs (Figure S2B).\n","['one']\n","Moreover, the percentage of GFP-positive (GFP+) cells in hSOX2-23 was constant over more than 20 passages.\n","['more than 20 passages.']\n","Immunofluorescence (IF) staining of hSOX2-23 showed that 100% of GFP+ cells expressed SOX2 protein (Figure S2C).\n","['100%']\n","Additionally, hSOX2-23 colonies had characteristic hESC morphology (Figure S2D) and expressed markers of the undifferentiated state, such as NANOG (Figure S2E).\n","[]\n","These results show that this rAAV-based gene-targeting method can be used to efficiently disrupt genes by homologous recombination.\n","[]\n","In addition, the SOX2-GFP hESC marker line can be used to monitor SOX2 expression in undifferentiated hESCs.\n","[]\n","Based on several previous publications (Khan et al., 2010, 2011), we explored the utility of adeno-associated virus (AAV) as a method to improve gene targeting efficiencies in hPSCs.\n","[]\n","For SOX2, a gene that is highly expressed in undifferentiated hPSCs, gene targeting rates were greater than 70%.\n","['greater than 70%.']\n","Similar targeting efficiencies in hPSCs using AAV have been reported by others (Asuri et al., 2012; Khan et al., 2010, 2011; Smith-Arica et al., 2003), indicating that AAV offers a highly efficient and robust approach to target genes for HR in hPSCs.\n","[]\n","(C) C-peptide and PDX1 expression was confirmed by immunocytochemistry of cells differentiated for 25 days.\n","['25 days.']\n","â€¢Adult hRPESC-derived RPE had comparable in vitro characteristics to fetal hRPEâ€¢hRPE monolayers survived 4 weeks on PET carriers under the rabbit retinaâ€¢Better xenograft survival may be due to the maintained hRPE cell polarityâ€¢Atrophy of the retina overlaying the hRPE xenograft remains a future challenge\n","['4 weeks']\n","(O) GH plasma levels using a human specific ELISA (6 weeks after transplantation).\n","['6 weeks']\n","(R) Quantification of immunocytochemical analyses for each iPSC line.\n","[]\n","Data are shown as the means Â± SD (n = 3 independent experiments).\n","['3']\n","SER, serotonin, TUBÎ²III, Î²-tubulin class III.\n","[]\n","Scale bars: 200 Î¼m in (A)â€“(H), 50 Î¼m in insets of (C)â€“(F), 100 Î¼m in (J)â€“(M) and (Q).\n","['200 Î¼m', '50 Î¼m', '100 Î¼m']\n","The two FTD patients under investigation in this study were part of a longitudinal dementia research program at the Memory and Aging Center, University of California, San Francisco.\n","['two']\n","Both had an 8-year history of behavioral changes and memory impairment at the time of tissue collection for this study.\n","['8-year']\n","One patient, a 67-year-old male with sporadic FTD, tested negative for mutations in GRN, MAPT, and C9ORF72.\n","['67-year-']\n","The other patient, a 64-year-old male with a significant family history of dementia, had behavioral variant FTD.\n","['64-year-']\n","MRI in this patient demonstrated severe bifrontal and temporal atrophy associated with gliosis in the frontal lobes (greater on the right).\n","[]\n","One year later, MRI scans showed progression of atrophy and gliosis.\n","[]\n","Genetic testing revealed a novel nonsense mutation in GRN, p.S116X (g.4627C > A, c.347C > A), which is predicted to result in a premature stop codon.\n","[]\n","Both FTD patients had parkinsonism, which is typical of all FTD patients with PGRN mutations.\n","[]\n","An age-matched subject, a clinically normal 64-year-old male with no mutations in GRN, MAPT, or C9ORF72, served as a control.\n","['64-year-']\n","(Qâ€“S) Magnetic resonance images of a representative animal (No.\n","[]\n","6, autograft) at 3 months after the transplant.\n","['3 months']\n","The arrowheads indicate the directions of the cell injections.\n","[]\n","(Q) coronal, (R) axial, and (S) sagittal.\n","[]\n","The letter L indicates the left side.\n","[]\n","50 Î¼m\n","(A) Fetal hRPE stained for pan-cytokeratin (scale bar, 50 Î¼m); inset shows section overview stained with hematoxylin/eosin (scale bar, 200 Î¼m).\n","['50 Î¼m', '200 Î¼m']\n","To assess the in vivo properties of hESC-derived trigeminal placode precursors, PIP-induced neuronal clusters, derived from a constitutively GFP-positive hESC line (Figures S5A and S5B), were injected into the developing chick embryo targeting the early trigeminal anlage at H&H stage 10â€“12 (Figure S5C).\n","[]\n","Human cells were identified based on GFP expression and use of human specific antibodies against cytoplasmic antigen (hCA).\n","[]\n","Two days after in ovo transplantation, surviving GFP+ cells were found dispersed in the area of the endogenous chick trigeminal ganglion (Figure 4P).\n","[]\n","We observed extensive GFP+ human fiber bundles coexpressing hCA and peripherin (Figures 4Q and 4R).\n","[]\n","In contrast, no hCA or peripherin expression was detected in the neural tube of the embryo (Figure S5D).\n","[]\n","The in vivo fiber outgrowth 2 days after transplantation was reminiscent of the extensive in vitro fiber outgrowth of replated trigeminal neuron clusters (Figure S5A).\n","['2 days']\n","Peripherin expression in vivo (Figure 4S) confirmed the peripheral neuron identity of the grafted cells.\n","[]\n","For the first two animals (Nos.\n","['two']\n","1 and 4), we established iPSCs from fibroblasts derived from the oral mucosa using retroviral vectors (Okita et al., 2011).\n","[]\n","For the other two animals (Nos.\n","['two']\n","6 and 8), we used peripheral blood mononuclear cells (PBMCs) with nonintegrating episomal vectors (Okita et al., 2013).\n","[]\n","We selected the best clone from each animal according to the following criteria: a stable embryonic stem cell (ESC)-like morphology of the colonies after passaging, expression of pluripotent markers, few or no integrated transgenes (Figures 1Aâ€“1F; Figure S1 available online), and the potential for stable neural differentiation.\n","[]\n","A PCR analysis revealed that all of the clones with retroviral vectors showed apparent expression of remaining transgenes (Figures S1C and S1D), whereas the clones with episomal vectors never did (Figure S1F).\n","[]\n","To detect the iPSC-derived cells in a brain, we introduced GFP (Figures 1G and 1H).\n","[]\n","The selected clones of iPSCs had the potential to generate teratomas in the testes of a severe combined immunodeficiency (SCID) mouse within 12 weeks (Figures 1Iâ€“1M).\n","['12 weeks']\n","(A) Flow-cytometric analyses for MHC-I (HLA-A, HLA-B, and HLA-C).\n","[]\n","Incubation of the cells with IFN-Î³ for 48 hr increased the MHC-I expression (green).\n","['48 hr']\n","Our group has developed a defined culture system to direct the differentiation of hPSCs into a near-homogenous population of definitive endoderm (DE) cells that have the capacity to differentiate into hepatocytes and pancreatic progenitors (Brown et al., 2011; Cho et al., 2012; Rashid et al., 2010; Touboul et al., 2010; Vallier et al., 2009a; Yusa et al., 2011).\n","[]\n","Cells grown in these culture conditions successively express primitive streak markers (T and Mixl1), downregulate pluripotency markers (NANOG, SOX2, and POU5F1) and progressively upregulate definitive endoderm markers (CXCR4, FOXA2, GATA4, CERB, and SOX17) (Figures S1Aâ€“S1C available online).\n","[]\n","Flow cytometry analyses showed that 80% of the resulting DE population coexpresses CXCR4 and SOX17 (Figure S1D).\n","['80%']\n","Interestingly, the resulting population of DE cells is negative for genes marking the foregut (SOX2), the midgut/hindgut (CDX2), the pancreas (PDX1), the liver (AFP), and the lungs (HOXA1) (Figures S1E and S1F) This confirms that DE cells generated in vitro could correspond to early endoderm progenitor cells prior to anteroposterior patterning or organogenesis.\n","[]\n","In sequential PET studies, we observed increased uptake of [11C]PK11195 in one allograft (animal No.\n","[]\n","10) at 3 months (Figures 3A and 3B).\n","['3 months']\n","We could not detect any apparent uptake in the other animals or at any other time points (Figure S2).\n","[]\n","Intriguingly, the serum level of IFN-Î³ temporarily increased at 2 months after the transplant in three animals (Figure 3C).\n","['at 2 months', 'three']\n","An immunofluorescence study conducted at 3.5â€“4 months showed that MHC-II+ cells were more frequently found in allografts than in autografts, especially in the monkey with increased uptake of [11C]PK11195 (Figure 3D, No.\n","['3.5â€“4 months']\n","10).\n","[]\n","The MHC-II staining never overlapped with that of GFP of the donor cells (Figure 3F), whereas it generally overlapped with that of IBA1 (Figure 3G), indicating that MHC-II was expressed by host-derived microglia.\n","[]\n","Consistently, the number and density of IBA1+ cells were higher in allografts than in autografts (Figures 3E, 3H, and S4C).\n","[]\n","An increase in the expression of MHC might trigger the recruitment of circulating immune cells, including T cells.\n","[]\n","An immunofluorescence study revealed that more CD45+ cells (a marker for pan-leukocytes) accumulated in allografts compared with autografts (Figures 3I and 3J).\n","[]\n","Most of the CD45+ cells were CD3+ T cells, and 60% of them were CD8+ killer T cells (Figures 3K and 3L).\n","['60%']\n","These findings suggest that an acquired immune response was elicited only in the allografts in the primate brain.\n","[]\n","This study was approved by the Institutional Review Board and Ethics Committees of the University of California, San Francisco, and written informed consent was obtained in all cases.\n","[]\n","The patient with the PGRN S116X mutation followed the classic clinical progression for FTD and developed parkinsonism, as do all FTD patients with PGRN mutations, but he did not show typical features of PD dementia.\n","[]\n","The patient with sporadic FTD also showed parkinsonism.\n","[]\n","Skin biopsies were collected, cut into small pieces, and placed on culture dishes to allow the fibroblasts to expand.\n","[]\n","The cells were maintained in Dulbecco's modified Eagle's medium supplemented with 10% fetal bovine serum, 1X nonessential amino acids, and penicillin/streptomycin (100 U/ml).\n","['10%', '1x']\n","iPSCs were generated as described previously (Takahashi et al., 2007).\n","[]\n","Please see Supplemental Information for more details.\n","[]\n","(Dâ€“H) Adult hRPE stained for SC121 (red).\n","[]\n","(D) Adult hRPE stained for MCT1 (green).\n","[]\n","(E) Human adult RPE stained for ezrin (green).\n","[]\n","hRPEs transplanted into rabbit SRS show absence of expression of ki67 (F), phosphohistone H3 (G), and caspase-3 (H).\n","[]\n","Polarized fetal and adult hRPE cells were found in TEM (I and J).\n","[]\n","Nuclei with regular chromatin were found in the basal compartment, a basal lamina ([I], large black arrowhead) had formed between the xenograft and PET carrier (black asterisks).\n","[]\n","Melanosomes (M) in multiple stages, some microvilli abutting to the atrophic neural retina (NR), and junctional structures with desmosomes (small black arrowhead) and tight junctions (red arrowhead) were discerned apically.\n","[]\n","Mitochondria (MC) were seen in the basolateral part of the cell.\n","[]\n","Detachment from cell carrier (asterisk) in (J) is a histologic processing artifact.\n","[]\n","Left images in (I) and (J) taken at 10,500Ã— magnification; right micrographs are rectangular zone in left at 25,000Ã—; scale bars represent 2 Î¼m/inset 0.2 Î¼m distance in (I) and (J).\n","['10,500Ã—', '25,000Ã—', '2 Î¼m/', '0.2 Î¼m']\n","iPSC colonies were detached with accutase (Millipore) and grown as embryoid bodies (EBs) in suspension for 5â€“6 days in iPSC medium without basic fibroblast growth factor.\n","['5â€“6 days']\n","EBs were allowed to attach and form rosettes.\n","[]\n","Ten-day-old rosettes were collected and grown in suspension as neurospheres.\n","['ten-']\n","0.1 mg/ml\n","Neurospheres were dissociated after 3â€“4 weeks, and the cells were plated on glass coverslips (BD Biosciences) or plates coated with poly-D-lysine (0.1 mg/ml) and laminin (10 Î¼g/m).\n","['3â€“4 weeks,', '0.1 mg/ml', '10 Î¼g']\n","Neurons were used after 2-4 weeks in culture.\n","['after 2-4 weeks']\n","For cell differentiation, FAP-specific iPS cells were cultured with serial changes of media as shown in Fig.\n","[]\n","3A.\n","[]\n","To test whether FAP-specific iPS cells can differentiate into hepatocyte-like cells, we analyzed several markers via real-time PCR analysis on Day 5 (D5), D13 and D20 differentiated FAP-specific iPS cells (Fig.\n","[]\n","3B).\n","[]\n","A decrease in expression of the pluripotency marker Oct3/4 was accompanied by differentiation of FAP-specific iPS cells.\n","[]\n","Expression of the endoderm marker Sox17 was observed on D5 differentiation and decreased gradually after the medium was changed to hepatic differentiation medium on D7 (Fig.\n","[]\n","3B).\n","[]\n","The hepatic progenitor marker AFP and the mature hepatocyte marker ALB were obviously expressed on D13 and D20.\n","[]\n","In addition, immunocytochemical analyses showed Sox17 expression on D5, both HNF-4Î± and AFP expression on D13, and ALB cytoplasmic staining on D20 (Fig.\n","[]\n","3C).\n","[]\n","Quantitative imaging analysis revealed that approximately 78 Â± 0.6% of cells were Sox17-positive on D5 and approximately 88 Â± 1.1% of cells were AFP-positive on D13 and approximately 29 Â± 0.9% of cells were ALB-positive on D20 (Fig.\n","['approximately 78 Â± 0.6%', 'approximately 88 Â± 1.1%', 'approximately 29 Â± 0.9%']\n","3D).\n","[]\n","The ALB secretion in the media of differentiated FAP-specific iPS cells on D20 was approximately 20 Î¼g/ml (Fig.\n","['approximately 20 Î¼g/ml']\n","3E).\n","[]\n","Moreover, these D20 differentiated FAP-specific iPS cells were also periodic acid-Schiff (PAS)-positive, indicating cytoplasmic glycogen storage (Fig.\n","[]\n","3F).\n","[]\n","These results clearly indicated that FAP-specific iPS cells had the potential to differentiate into hepatocyte-like cells.\n","[]\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"tBoVMN2SLMxF"},"source":["Qualifier"]},{"cell_type":"code","metadata":{"id":"6aY6HM7YG8Rn"},"source":["bert_model = BERT_Arch(model, 768, 256, 0.5, 1,1)\r\n","bert_model = bert_model.to(device)\r\n","bert_model = torch.load(QUALIFIER_PATH)\r\n","Qual=[]\r\n","for s in test_text:\r\n","  tokenized_text = tokenizer.tokenize(s)\r\n","  test_inputs = tokenizer(s,max_length = 512,padding='max_length',truncation=True, return_tensors=\"pt\")\r\n","  bert_model.eval()\r\n","  temp = bert_model(test_inputs['input_ids'].to(device), test_inputs['attention_mask'].to(device))>=0.995\r\n","  \r\n","  Qual.append(temp[0])"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"HZ9El1LUMJ0m","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1608028076857,"user_tz":-330,"elapsed":20708,"user":{"displayName":"Akash Gangwar","photoUrl":"","userId":"02081727011117790241"}},"outputId":"36a10c5a-f2b3-4b5b-ace9-118c93fb261a"},"source":["Qualifier=[]\r\n","for i in range(len(test_text)):\r\n","  tokenized_text = tokenizer.tokenize(test_text[i])\r\n","  txt=[]\r\n","  label=[]\r\n","  \r\n","  combine=\"\"\r\n","  for j in range(len(tokenized_text)):\r\n","    if(Qual[i][j+1] and Qual[i][j] and j!=0):\r\n","      label.append(tokenized_text[j])\r\n","      if(len(tokenized_text[j])>=2 and tokenized_text[j][:2]==\"##\"):\r\n","          combine=combine+tokenized_text[j][2:]\r\n","      \r\n","      else:\r\n","        if(len(combine)==0):\r\n","          combine=tokenized_text[j]\r\n","          continue\r\n","        \r\n","        if(tokenized_text[j] in (\"!\", \",\" ,\"\\'\" ,\";\" ,\"\\\"\", \".\", \"-\" ,\"?\",\"+\",\"-\",\"*\",\"/\",\"%\")):\r\n","          combine=combine+tokenized_text[j]\r\n","          continue\r\n","        if(tokenized_text[j] in (\"(\",\")\") or combine in (\"(\",\")\")):\r\n","          print(combine)\r\n","          continue\r\n","        start=[m.start() for m in re.finditer(combine,test_text[i])]\r\n","        end=[m.start() for m in re.finditer(tokenized_text[j],test_text[i])]\r\n","        flag=-1\r\n","        \r\n","        for s in start:\r\n","          for e in end:\r\n","            if(flag!=-1):\r\n","              break\r\n","            \r\n","            \r\n","            if(e-s-len(combine)+1==1):\r\n","              flag=1\r\n","            elif(e-s-len(combine)+1==2):\r\n","              flag=2\r\n","        \r\n","        if(flag==2):\r\n","          combine=combine+\" \"+tokenized_text[j]\r\n","        elif(flag==1):\r\n","          combine=combine+tokenized_text[j]\r\n","          \r\n","        else:\r\n","          txt.append(combine)\r\n","          combine=\"\"\r\n","\r\n","    elif(Qual[i][j+1]):\r\n","      label.append(tokenized_text[j])\r\n","      \r\n","      if(len(combine)!=0):\r\n","        txt.append(combine)\r\n","      \r\n","      if(len(tokenized_text[j])>=2 and tokenized_text[j][:2]==\"##\"):\r\n","        tokenized_text[j]=tokenized_text[j][2:]\r\n","      \r\n","      combine=tokenized_text[j]\r\n","    else:\r\n","      if(len(combine)!=0):\r\n","        txt.append(combine)\r\n","      \r\n","      combine=\"\"\r\n","        \r\n","  if(len(combine)!=0):\r\n","        txt.append(combine)\r\n","  \r\n","               \r\n","  print(test_text[i])\r\n","  print(txt)\r\n","  Quantity.append(txt)\r\n","  \r\n","  "],"execution_count":null,"outputs":[{"output_type":"stream","text":["(F) Adult hRPE cultures display expression of markers typical of native RPE in their polarized localization.\n","[]\n","DAPI is cyan, whereas all other immunofluorescence is gold.\n","[]\n","Claudin 19, ezrin, ZO1, and MCT1 are preferentially located on the apical side.\n","[]\n","RPE65 and CRALBP are cytoplasmic.\n","[]\n","The scale bars represent 10 Î¼m.\n","[]\n","GFP-expressing H9, BBHX8, and A1ATD-1 cells were generated by stable transfection using lipofectamine 2000 (Invitrogen) as described previously (Vallier et al., 2001).\n","[]\n","GFP-positive cells were differentiated into foregut cells and then dissociated into single cells.\n","[]\n","An individually isolated GFP cell was then transferred into a well containing non-GFP-positive hFSCs.\n","[]\n","Wells were visually inspected 12 hr after plating, and wells containing a single GFP-positive hFSC were selected for clonal expansion.\n","[]\n","For group analyses, the following procedures were applied.\n","[]\n","(1) T1-weighted sMRIs from each subject (Fig.\n","[]\n","1A) were registered to MNI space (Montreal Neurological Institute, MNI-152 atlas as in Fig.\n","[]\n","1B) using an affine transformation (FLIRTâ€“FMRIB's Linear Image Registration Tool) (Jenkinson and Smith, 2001) in FSL (www.fmrib.ox.ac.uk/fsl/).\n","[]\n","(2) The cortical (Fig.\n","[]\n","1C) and subcortical masks with pre-defined brain regions from the standard atlas were transferred to the individual's headspace (Fig.\n","[]\n","1D), using the inverse of the transformation obtained in the first step: the Harvard-Oxford Atlas, part of the FSL software with masks of 96 cortical gray-matter regions (48 regions in each hemisphere), 21 sub-cortical regions, and cerebellum, was used.\n","[]\n","(3) The regional masks were down-sampled to a cubic source grid with voxels of 5 mm per side (Fig.\n","[]\n","1E).\n","[]\n","(4) VESTAL MEG source imaging used the source grid from step 3.\n","[]\n","This step permits group-based analyses.\n","[]\n","In the shown example, MEG responses evoked by S1 localized to left and right Heschl's gyri (Fig.\n","[]\n","1F).\n","[]\n","(5) Finally, for regions of interest (ROIs), the source time course was obtained by summing activity from all ROI voxels.\n","[]\n","Fig.\n","[]\n","1H shows the time course from left Heschl's gyrus (dark blue region in Fig.\n","[]\n","1C and D).\n","[]\n","Another important finding is that, in spite of the immune responses mounted by the host brain, a substantial number of TH+ cells survived in the allografts.\n","[]\n","This is consistent with previous clinical reports of human fetal cell transplantation.\n","[]\n","Postmortem analyses of the patients revealed robust survival of DA neurons in spite of the fact that numerous immune cells were present around the graft (Kordower et al., 1997).\n","[]\n","In two double-blind clinical trials, immunosuppressive drugs were never used (Freed et al., 2001) or were withdrawn after 6 months (Olanow et al., 2003).\n","[]\n","In these cases, the cells from multiple fetuses were used without HLA matching, but more than 50,000 TH+ cells had survived after several years.\n","['+', 'survived after']\n","Our quantitative PCR (qPCR) study in vitro showed that the expression of MHC-I increased in response to IFN-Î³, but the expression level was still 1/10 that of untreated monkey peripheral blood cells (Figure 2B).\n","[]\n","The in vivo studies revealed that the serum level of IFN-Î³ increased at 2 months, and CD45+ cells (including CD8+ cells) accumulated in the allografts 3.5â€“4 months after the transplant.\n","['after the transplant']\n","On the other hand, the levels of INF-Î³ in the cerebrospinal fluid (CSF) and the levels of tumor necrosis factor Î± (TNF-Î±) in both the serum and CSF were below the limit of detection by ELISA (data not shown).\n","[]\n","An immunofluorescence study did not reveal any apparent expression of MHC-I by the grafted cells (Figure S4A).\n","[]\n","Therefore, it is possible that the immune response in the primate brain was not strong enough to reject all of the donor cells.\n","[]\n","These findings closely correlate with the results of previous murine experiments (Hudson et al., 1994; Shinoda et al., 1995).\n","[]\n","To apply our findings to a more clinically relevant setting, we investigated the expression of HLA-I during neural differentiation of human ESCs (hESCs) and iPSCs by qPCR (Figure S4D).\n","[]\n","The expression level was 1/100 compared with that of human peripheral blood cells in both hESCs and iPSCs, and it was similarly elevated in response to IFN-Î³.\n","[]\n","It is difficult to precisely compare immunogenicity in monkeys with that in humans, but the low expression level of MHC-I by the donor cells may account for the mild rejection in both monkey and human neural transplantation.\n","[]\n","We next addressed whether hESC-derived trigeminal neurons can engraft in the adult mouse CNS and project toward their physiological target.\n","[]\n","The trigeminal nuclei in the brainstem receive afferent innervation from the trigeminal sensory ganglion that is relayed to the contralateral thalamus.\n","[]\n","The pons was selected as site for transplantation, because it is surgically accessible and located within proximity of the trigeminal brain stem nuclei that receive afferent input from the trigeminal ganglia.\n","[]\n","Hence, GFP+ human trigeminal neuron clusters were injected into adult NOD/SCID mice via stereotactic surgery (see Experimental Procedures).\n","[]\n","Histological analysis 4 weeks after transplantation showed survival of GFP+ human cell graft in the ventral pons (Figure S5E).\n","[]\n","Although GFP+ cell bodies remained tightly clustered at injection site, GFP+ fibers showed extensive projections into the host brain (n = 6) including the endogenous trigeminal nuclei (Figure S5F).\n","[]\n","Expression of BRN3A confirmed the sensory neuron identity of the cells (Figure S5G).\n","[]\n","Graft-derived human fiber bundles (hNCAM+ and GFP+) were observed emanating from the graft core (Figure S5H).\n","[]\n","These data demonstrate in vivo survival of trigeminal placode derivatives, differentiation along sensory neuron lineage, and the establishment of axonal projections toward relevant endogenous targets in the embryonic chick and adult mouse brain.\n","[]\n","(B) Fetal hRPE stained for SC121 (red) and MCT1 (green; scale bars, 125 Î¼m and 25 Î¼m [inset]).\n","[]\n","(F) Intracellular PGRN levels in iPSC-derived neurons after medium collection.\n","[]\n","Values of control line 20 were set to 100% (n = 3â€“4 independent cultures).\n","[]\n","Prior to VESTAL analyses, a 5â€“55 Hz bandpass filter was applied.\n","[]\n","VESTAL analyses examined activity 30â€“130 ms post-stimulus producing a 4D activation map (3D volumes across time) as well as a 2D source time-course matrix.\n","[]\n","The average percent variance explained for gradiometer data using VESTAL was 95.81% for HC and 94.38% for SZ.\n","['for', 'for', 'z']\n","The average percent variance explained for magnetometer data using VESTAL program was 96.24% for HC and 93.17% for SZ.\n","['ometer', 'vestal', 'for', 'for', 'z']\n","There were no group differences in percent variance explained for gradiometer data (t(39) = 1.16, p = 0.25) or magnetometer data (t(39) = 1.31, p = 0.20).\n","['ometer']\n","(C) Control lines 16 and 17, sporadic lines 12 and 23, and PGRN S116X lines 1 and 14 were immunostained for AFP (endoderm), desmin (mesoderm), and Î²III-tubulin (ectoderm), and counterstained with DAPI (nuclei).\n","[]\n","All lines showed a normal karyotype.\n","[]\n","Scale bar: 50 Î¼m.\n","[]\n","(Iâ€“M) Teratoma formation at 3 months after transplantation in the testes of SCID mice.\n","['after transplantation in the test', 'sci']\n","H&E staining of the sections showed histological features of the neuroepithelium (J), cartilage (K), muscle (L) and gut-like epithelium (M).\n","[]\n","Using a recombinant adeno-associated viral (rAAV)-based gene-targeting method, we inserted the gene-encoding GFP into the SOX2 locus in H9 hESCs (Figure 1A).\n","[]\n","Proper homologous recombination led to the replacement of the SOX2 open reading frame with that of GFP and a neomycin selection cassette (SV40-Neo).\n","[]\n","After infection with rAAV and G418 drug selection, a total of 36 clones were expanded and screened by Southern blotting for homologous recombination events.\n","[]\n","Among these clones, 26 (72%) were found to carry the GFP-Neo cassette in the SOX2 locus (Figure S1A available online).\n","[]\n","No clones in which both SOX2 alleles were disrupted were isolated.\n","[]\n","Our subsequent analysis focused on one of these clones, clone 23 (hSOX2-23).\n","[]\n","We confirmed appropriate gene targeting in this clone using multiple restriction digests followed by Southern blotting (Figures 1B, S1B, and S1C).\n","[]\n","We did not observe nontargeted insertions of the rAAV sequences, and cells exhibited a normal karyotype (data not shown).\n","[]\n","Flow cytometry of hSOX2-23 revealed that the majority of the cells expressed GFP (Figure 1C).\n","[]\n","By comparison, a drug-selected clone, hSOX2-25, which was negative for targeted insertion (Figure S1A), showed no detectable GFP (Figure S2A).\n","[]\n","Despite only having one copy of SOX2, hSOX2-23 had similar levels of SOX2, OCT4, and NANOG expression as hSOX2-25 and wild-type (WT) hESCs (Figure S2B).\n","[]\n","Moreover, the percentage of GFP-positive (GFP+) cells in hSOX2-23 was constant over more than 20 passages.\n","['20 passages']\n","Immunofluorescence (IF) staining of hSOX2-23 showed that 100% of GFP+ cells expressed SOX2 protein (Figure S2C).\n","[]\n","Additionally, hSOX2-23 colonies had characteristic hESC morphology (Figure S2D) and expressed markers of the undifferentiated state, such as NANOG (Figure S2E).\n","[]\n","These results show that this rAAV-based gene-targeting method can be used to efficiently disrupt genes by homologous recombination.\n","[]\n","In addition, the SOX2-GFP hESC marker line can be used to monitor SOX2 expression in undifferentiated hESCs.\n","[]\n","Based on several previous publications (Khan et al., 2010, 2011), we explored the utility of adeno-associated virus (AAV) as a method to improve gene targeting efficiencies in hPSCs.\n","[]\n","For SOX2, a gene that is highly expressed in undifferentiated hPSCs, gene targeting rates were greater than 70%.\n","[]\n","Similar targeting efficiencies in hPSCs using AAV have been reported by others (Asuri et al., 2012; Khan et al., 2010, 2011; Smith-Arica et al., 2003), indicating that AAV offers a highly efficient and robust approach to target genes for HR in hPSCs.\n","[]\n","(C) C-peptide and PDX1 expression was confirmed by immunocytochemistry of cells differentiated for 25 days.\n","[]\n","â€¢Adult hRPESC-derived RPE had comparable in vitro characteristics to fetal hRPEâ€¢hRPE monolayers survived 4 weeks on PET carriers under the rabbit retinaâ€¢Better xenograft survival may be due to the maintained hRPE cell polarityâ€¢Atrophy of the retina overlaying the hRPE xenograft remains a future challenge\n","[]\n","(O) GH plasma levels using a human specific ELISA (6 weeks after transplantation).\n","['plasma', 'weeks after transplantation']\n","(R) Quantification of immunocytochemical analyses for each iPSC line.\n","[]\n","Data are shown as the means Â± SD (n = 3 independent experiments).\n","[]\n","SER, serotonin, TUBÎ²III, Î²-tubulin class III.\n","[]\n","Scale bars: 200 Î¼m in (A)â€“(H), 50 Î¼m in insets of (C)â€“(F), 100 Î¼m in (J)â€“(M) and (Q).\n","[]\n","The two FTD patients under investigation in this study were part of a longitudinal dementia research program at the Memory and Aging Center, University of California, San Francisco.\n","[]\n","Both had an 8-year history of behavioral changes and memory impairment at the time of tissue collection for this study.\n","[]\n","One patient, a 67-year-old male with sporadic FTD, tested negative for mutations in GRN, MAPT, and C9ORF72.\n","[]\n","The other patient, a 64-year-old male with a significant family history of dementia, had behavioral variant FTD.\n","[]\n","MRI in this patient demonstrated severe bifrontal and temporal atrophy associated with gliosis in the frontal lobes (greater on the right).\n","[]\n","One year later, MRI scans showed progression of atrophy and gliosis.\n","[]\n","Genetic testing revealed a novel nonsense mutation in GRN, p.S116X (g.4627C > A, c.347C > A), which is predicted to result in a premature stop codon.\n","[]\n","Both FTD patients had parkinsonism, which is typical of all FTD patients with PGRN mutations.\n","[]\n","An age-matched subject, a clinically normal 64-year-old male with no mutations in GRN, MAPT, or C9ORF72, served as a control.\n","[]\n","(Qâ€“S) Magnetic resonance images of a representative animal (No.\n","[]\n","6, autograft) at 3 months after the transplant.\n","['at', 'months after the transplant.']\n","The arrowheads indicate the directions of the cell injections.\n","[]\n","(Q) coronal, (R) axial, and (S) sagittal.\n","[]\n","The letter L indicates the left side.\n","[]\n","(A) Fetal hRPE stained for pan-cytokeratin (scale bar, 50 Î¼m); inset shows section overview stained with hematoxylin/eosin (scale bar, 200 Î¼m).\n","[]\n","To assess the in vivo properties of hESC-derived trigeminal placode precursors, PIP-induced neuronal clusters, derived from a constitutively GFP-positive hESC line (Figures S5A and S5B), were injected into the developing chick embryo targeting the early trigeminal anlage at H&H stage 10â€“12 (Figure S5C).\n","[]\n","Human cells were identified based on GFP expression and use of human specific antibodies against cytoplasmic antigen (hCA).\n","[]\n","Two days after in ovo transplantation, surviving GFP+ cells were found dispersed in the area of the endogenous chick trigeminal ganglion (Figure 4P).\n","[]\n","We observed extensive GFP+ human fiber bundles coexpressing hCA and peripherin (Figures 4Q and 4R).\n","[]\n","In contrast, no hCA or peripherin expression was detected in the neural tube of the embryo (Figure S5D).\n","[]\n","The in vivo fiber outgrowth 2 days after transplantation was reminiscent of the extensive in vitro fiber outgrowth of replated trigeminal neuron clusters (Figure S5A).\n","[]\n","Peripherin expression in vivo (Figure 4S) confirmed the peripheral neuron identity of the grafted cells.\n","[]\n","For the first two animals (Nos.\n","[]\n","1 and 4), we established iPSCs from fibroblasts derived from the oral mucosa using retroviral vectors (Okita et al., 2011).\n","[]\n","For the other two animals (Nos.\n","[]\n","6 and 8), we used peripheral blood mononuclear cells (PBMCs) with nonintegrating episomal vectors (Okita et al., 2013).\n","[]\n","We selected the best clone from each animal according to the following criteria: a stable embryonic stem cell (ESC)-like morphology of the colonies after passaging, expression of pluripotent markers, few or no integrated transgenes (Figures 1Aâ€“1F; Figure S1 available online), and the potential for stable neural differentiation.\n","[]\n","A PCR analysis revealed that all of the clones with retroviral vectors showed apparent expression of remaining transgenes (Figures S1C and S1D), whereas the clones with episomal vectors never did (Figure S1F).\n","[]\n","To detect the iPSC-derived cells in a brain, we introduced GFP (Figures 1G and 1H).\n","[]\n","The selected clones of iPSCs had the potential to generate teratomas in the testes of a severe combined immunodeficiency (SCID) mouse within 12 weeks (Figures 1Iâ€“1M).\n","[]\n","(A) Flow-cytometric analyses for MHC-I (HLA-A, HLA-B, and HLA-C).\n","[]\n","Incubation of the cells with IFN-Î³ for 48 hr increased the MHC-I expression (green).\n","[]\n","Our group has developed a defined culture system to direct the differentiation of hPSCs into a near-homogenous population of definitive endoderm (DE) cells that have the capacity to differentiate into hepatocytes and pancreatic progenitors (Brown et al., 2011; Cho et al., 2012; Rashid et al., 2010; Touboul et al., 2010; Vallier et al., 2009a; Yusa et al., 2011).\n","[]\n","Cells grown in these culture conditions successively express primitive streak markers (T and Mixl1), downregulate pluripotency markers (NANOG, SOX2, and POU5F1) and progressively upregulate definitive endoderm markers (CXCR4, FOXA2, GATA4, CERB, and SOX17) (Figures S1Aâ€“S1C available online).\n","[]\n","Flow cytometry analyses showed that 80% of the resulting DE population coexpresses CXCR4 and SOX17 (Figure S1D).\n","[]\n","Interestingly, the resulting population of DE cells is negative for genes marking the foregut (SOX2), the midgut/hindgut (CDX2), the pancreas (PDX1), the liver (AFP), and the lungs (HOXA1) (Figures S1E and S1F) This confirms that DE cells generated in vitro could correspond to early endoderm progenitor cells prior to anteroposterior patterning or organogenesis.\n","[]\n","In sequential PET studies, we observed increased uptake of [11C]PK11195 in one allograft (animal No.\n","['all', 't']\n","10) at 3 months (Figures 3A and 3B).\n","[]\n","We could not detect any apparent uptake in the other animals or at any other time points (Figure S2).\n","[]\n","Intriguingly, the serum level of IFN-Î³ temporarily increased at 2 months after the transplant in three animals (Figure 3C).\n","['n-', 'temporarily increased at', 'months after the transplant in']\n","An immunofluorescence study conducted at 3.5â€“4 months showed that MHC-II+ cells were more frequently found in allografts than in autografts, especially in the monkey with increased uptake of [11C]PK11195 (Figure 3D, No.\n","[]\n","10).\n","[]\n","The MHC-II staining never overlapped with that of GFP of the donor cells (Figure 3F), whereas it generally overlapped with that of IBA1 (Figure 3G), indicating that MHC-II was expressed by host-derived microglia.\n","[]\n","Consistently, the number and density of IBA1+ cells were higher in allografts than in autografts (Figures 3E, 3H, and S4C).\n","[]\n","An increase in the expression of MHC might trigger the recruitment of circulating immune cells, including T cells.\n","[]\n","An immunofluorescence study revealed that more CD45+ cells (a marker for pan-leukocytes) accumulated in allografts compared with autografts (Figures 3I and 3J).\n","[]\n","Most of the CD45+ cells were CD3+ T cells, and 60% of them were CD8+ killer T cells (Figures 3K and 3L).\n","[]\n","These findings suggest that an acquired immune response was elicited only in the allografts in the primate brain.\n","[]\n","This study was approved by the Institutional Review Board and Ethics Committees of the University of California, San Francisco, and written informed consent was obtained in all cases.\n","[]\n","The patient with the PGRN S116X mutation followed the classic clinical progression for FTD and developed parkinsonism, as do all FTD patients with PGRN mutations, but he did not show typical features of PD dementia.\n","[]\n","The patient with sporadic FTD also showed parkinsonism.\n","[]\n","Skin biopsies were collected, cut into small pieces, and placed on culture dishes to allow the fibroblasts to expand.\n","[]\n","The cells were maintained in Dulbecco's modified Eagle's medium supplemented with 10% fetal bovine serum, 1X nonessential amino acids, and penicillin/streptomycin (100 U/ml).\n","[]\n","iPSCs were generated as described previously (Takahashi et al., 2007).\n","[]\n","Please see Supplemental Information for more details.\n","[]\n","(Dâ€“H) Adult hRPE stained for SC121 (red).\n","[]\n","(D) Adult hRPE stained for MCT1 (green).\n","[]\n","(E) Human adult RPE stained for ezrin (green).\n","[]\n","hRPEs transplanted into rabbit SRS show absence of expression of ki67 (F), phosphohistone H3 (G), and caspase-3 (H).\n","[]\n","Polarized fetal and adult hRPE cells were found in TEM (I and J).\n","[]\n","Nuclei with regular chromatin were found in the basal compartment, a basal lamina ([I], large black arrowhead) had formed between the xenograft and PET carrier (black asterisks).\n","[]\n","Melanosomes (M) in multiple stages, some microvilli abutting to the atrophic neural retina (NR), and junctional structures with desmosomes (small black arrowhead) and tight junctions (red arrowhead) were discerned apically.\n","[]\n","Mitochondria (MC) were seen in the basolateral part of the cell.\n","[]\n","Detachment from cell carrier (asterisk) in (J) is a histologic processing artifact.\n","[]\n","Left images in (I) and (J) taken at 10,500Ã— magnification; right micrographs are rectangular zone in left at 25,000Ã—; scale bars represent 2 Î¼m/inset 0.2 Î¼m distance in (I) and (J).\n","[]\n","iPSC colonies were detached with accutase (Millipore) and grown as embryoid bodies (EBs) in suspension for 5â€“6 days in iPSC medium without basic fibroblast growth factor.\n","[]\n","EBs were allowed to attach and form rosettes.\n","[]\n","Ten-day-old rosettes were collected and grown in suspension as neurospheres.\n","[]\n","Neurospheres were dissociated after 3â€“4 weeks, and the cells were plated on glass coverslips (BD Biosciences) or plates coated with poly-D-lysine (0.1 mg/ml) and laminin (10 Î¼g/m).\n","[]\n","Neurons were used after 2-4 weeks in culture.\n","['weeks in culture']\n","For cell differentiation, FAP-specific iPS cells were cultured with serial changes of media as shown in Fig.\n","[]\n","3A.\n","[]\n","To test whether FAP-specific iPS cells can differentiate into hepatocyte-like cells, we analyzed several markers via real-time PCR analysis on Day 5 (D5), D13 and D20 differentiated FAP-specific iPS cells (Fig.\n","[]\n","3B).\n","[]\n","A decrease in expression of the pluripotency marker Oct3/4 was accompanied by differentiation of FAP-specific iPS cells.\n","[]\n","Expression of the endoderm marker Sox17 was observed on D5 differentiation and decreased gradually after the medium was changed to hepatic differentiation medium on D7 (Fig.\n","[]\n","3B).\n","[]\n","The hepatic progenitor marker AFP and the mature hepatocyte marker ALB were obviously expressed on D13 and D20.\n","[]\n","In addition, immunocytochemical analyses showed Sox17 expression on D5, both HNF-4Î± and AFP expression on D13, and ALB cytoplasmic staining on D20 (Fig.\n","[]\n","3C).\n","[]\n","Quantitative imaging analysis revealed that approximately 78 Â± 0.6% of cells were Sox17-positive on D5 and approximately 88 Â± 1.1% of cells were AFP-positive on D13 and approximately 29 Â± 0.9% of cells were ALB-positive on D20 (Fig.\n","[]\n","3D).\n","[]\n","The ALB secretion in the media of differentiated FAP-specific iPS cells on D20 was approximately 20 Î¼g/ml (Fig.\n","[]\n","3E).\n","[]\n","Moreover, these D20 differentiated FAP-specific iPS cells were also periodic acid-Schiff (PAS)-positive, indicating cytoplasmic glycogen storage (Fig.\n","[]\n","3F).\n","[]\n","These results clearly indicated that FAP-specific iPS cells had the potential to differentiate into hepatocyte-like cells.\n","[]\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"v3GxgBiLF-9J"},"source":["Count and Measurement"]},{"cell_type":"code","metadata":{"id":"HsUkwQKqGBCU"},"source":["class BERT_Arch(nn.Module):\r\n","\r\n","    def __init__(self, bert, embed_dim, hidden_dim, drop_prob, n_layers, out_dim):\r\n","      \r\n","      super(BERT_Arch, self).__init__()\r\n","\r\n","      self.bert = bert \r\n","      self.bilstm = nn.LSTM(embed_dim, hidden_dim, n_layers,dropout=drop_prob,  bidirectional=True, batch_first=True)\r\n","      self.dropout = nn.Dropout(drop_prob)\r\n","      self.fc1 = nn.Linear(2*hidden_dim,256)\r\n","      self.r1=nn.Tanh()\r\n","      self.fc2 = nn.Linear(256,out_dim)\r\n","      self.sf = nn.Sigmoid()\r\n","\r\n","    #define the forward pass\r\n","    def forward(self, sent_id, mask):\r\n","\r\n","      #pass the inputs to the model  \r\n","      embed, cls_hs = self.bert(sent_id, attention_mask=mask)\r\n","      x,_ = self.bilstm(embed)\r\n","      x = self.dropout(x)\r\n","      x = self.fc1(x)\r\n","      x = self.r1(x)\r\n","      x = self.fc2(x)\r\n","      x = self.sf(x)\r\n","\r\n","      return x\r\n","\r\n","bert_model = BERT_Arch(model, 768, 256, 0.5, 1,1)\r\n","bert_model = bert_model.to(device)\r\n"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"5rmuja4_G8xR"},"source":["bert_model = torch.load(COUNT_MEASUREMENT_MODEL_PATH)\r\n","count_measurement=[]\r\n","for s in test_text:\r\n","  tokenized_text = tokenizer.tokenize(s)\r\n","  test_inputs = tokenizer(s,max_length = 512,padding='max_length',truncation=True, return_tensors=\"pt\")\r\n","  bert_model.eval()\r\n","  temp = bert_model(test_inputs['input_ids'].to(device), test_inputs['attention_mask'].to(device))>=0.95\r\n","  count_measurement.append(temp[0])"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"wOROv1I4HCwJ"},"source":["for i in range(len(test_text)):\r\n","  tokenized_text = tokenizer.tokenize(test_text[i])\r\n","  txt=[]\r\n","  print(test_text[i])\r\n","  for j in range(len(tokenized_text)):\r\n","    if(count_measurement[i][j+1]):\r\n","      txt.append(tokenized_text[j])   \r\n","  print(txt)"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"856zb7R4HreM"},"source":["Extra Information"]},{"cell_type":"code","metadata":{"id":"CGUaHKM6HopM"},"source":["class BERT_Arch(nn.Module):\r\n","\r\n","    def __init__(self, bert, embed_dim, hidden_dim, drop_prob, n_layers, out_dim):\r\n","      \r\n","      super(BERT_Arch, self).__init__()\r\n","\r\n","      self.bert = bert \r\n","      self.n_layers = n_layers\r\n","      self.bilstm = nn.LSTM(embed_dim, hidden_dim,  bidirectional=True, batch_first=True)\r\n","      self.dropout = nn.Dropout(drop_prob)\r\n","      self.fc1 = nn.Linear(2*hidden_dim,out_dim)\r\n","      self.sigmoid = nn.Sigmoid()\r\n","\r\n","    #define the forward pass\r\n","    def forward(self, sent_id, mask):\r\n","\r\n","      \r\n","      # embed,cls_hs = self.bert(sent_id, attention_mask=mask)\r\n","      \r\n","      x = self.bert(sent_id, attention_mask=mask)\r\n","      embed=x[0]\r\n","      cls_hs=x[1]\r\n","      \r\n","      x,_ = self.bilstm(embed)\r\n","      \r\n","      x = self.dropout(x)\r\n","      x = self.fc1(x)\r\n","      x = self.sigmoid(x)\r\n","\r\n","      return x\r\n","\r\n","bert_model = BERT_Arch(model, 768, 256, 0.5, 1,1)\r\n","bert_model = bert_model.to(device)"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"v3BFoxi9HGv5"},"source":["bert_model = torch.load(INFORMATION_DETECTION_PATH)\r\n","info=[]\r\n","for s in test_text:\r\n","  tokenized_text = tokenizer.tokenize(s)\r\n","  test_inputs = tokenizer(s,max_length = 512,padding='max_length',truncation=True, return_tensors=\"pt\")\r\n","  bert_model.eval()\r\n","  temp = bert_model(test_inputs['input_ids'].to(device), test_inputs['attention_mask'].to(device))>=0.3\r\n","  info.append(temp[0])"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"91D5MrIjImF1"},"source":["idx2labels = {0:'HasTolerance',\r\n"," 1:'IsApproximate',\r\n"," 2:'IsCount',\r\n"," 3:'IsList',\r\n"," 4:'IsMean',\r\n"," 5:'IsMeanHasSD',\r\n"," 6:'IsMeanHasTolerance',\r\n"," 7:'IsMeanIsRange',\r\n"," 8:'IsMedian',\r\n"," 9:'IsRange',\r\n"," 10:'IsRangeHasTolerance'}"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"NKHEc2g8II-t","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1608028094257,"user_tz":-330,"elapsed":21531,"user":{"displayName":"Akash Gangwar","photoUrl":"","userId":"02081727011117790241"}},"outputId":"7fe54fbf-b546-4530-cd06-49badbaf21ec"},"source":["for i in range(len(test_text)):\r\n","  tokenized_text = tokenizer.tokenize(test_text[i])\r\n","  txt=[]\r\n","  print(test_text[i])\r\n","  \r\n","  for j in range(len(tokenized_text)):\r\n","    for k in range(11):\r\n","      if(info[i][j+1][k]):\r\n","        txt.append(tokenized_text[j]+\" \"+idx2labels[k])   \r\n","  print(txt)"],"execution_count":null,"outputs":[{"output_type":"stream","text":["(F) Adult hRPE cultures display expression of markers typical of native RPE in their polarized localization.\n","[]\n","DAPI is cyan, whereas all other immunofluorescence is gold.\n","[]\n","Claudin 19, ezrin, ZO1, and MCT1 are preferentially located on the apical side.\n","[]\n","RPE65 and CRALBP are cytoplasmic.\n","[]\n","The scale bars represent 10 Î¼m.\n","[]\n","GFP-expressing H9, BBHX8, and A1ATD-1 cells were generated by stable transfection using lipofectamine 2000 (Invitrogen) as described previously (Vallier et al., 2001).\n","[]\n","GFP-positive cells were differentiated into foregut cells and then dissociated into single cells.\n","[]\n","An individually isolated GFP cell was then transferred into a well containing non-GFP-positive hFSCs.\n","[]\n","Wells were visually inspected 12 hr after plating, and wells containing a single GFP-positive hFSC were selected for clonal expansion.\n","['hr IsRange']\n","For group analyses, the following procedures were applied.\n","[]\n","(1) T1-weighted sMRIs from each subject (Fig.\n","[]\n","1A) were registered to MNI space (Montreal Neurological Institute, MNI-152 atlas as in Fig.\n","[]\n","1B) using an affine transformation (FLIRTâ€“FMRIB's Linear Image Registration Tool) (Jenkinson and Smith, 2001) in FSL (www.fmrib.ox.ac.uk/fsl/).\n","[]\n","(2) The cortical (Fig.\n","[]\n","1C) and subcortical masks with pre-defined brain regions from the standard atlas were transferred to the individual's headspace (Fig.\n","[]\n","1D), using the inverse of the transformation obtained in the first step: the Harvard-Oxford Atlas, part of the FSL software with masks of 96 cortical gray-matter regions (48 regions in each hemisphere), 21 sub-cortical regions, and cerebellum, was used.\n","[]\n","(3) The regional masks were down-sampled to a cubic source grid with voxels of 5 mm per side (Fig.\n","[]\n","1E).\n","[]\n","(4) VESTAL MEG source imaging used the source grid from step 3.\n","[]\n","This step permits group-based analyses.\n","[]\n","In the shown example, MEG responses evoked by S1 localized to left and right Heschl's gyri (Fig.\n","[]\n","1F).\n","[]\n","(5) Finally, for regions of interest (ROIs), the source time course was obtained by summing activity from all ROI voxels.\n","[]\n","Fig.\n","[]\n","1H shows the time course from left Heschl's gyrus (dark blue region in Fig.\n","[]\n","1C and D).\n","[]\n","Another important finding is that, in spite of the immune responses mounted by the host brain, a substantial number of TH+ cells survived in the allografts.\n","[]\n","This is consistent with previous clinical reports of human fetal cell transplantation.\n","[]\n","Postmortem analyses of the patients revealed robust survival of DA neurons in spite of the fact that numerous immune cells were present around the graft (Kordower et al., 1997).\n","[]\n","In two double-blind clinical trials, immunosuppressive drugs were never used (Freed et al., 2001) or were withdrawn after 6 months (Olanow et al., 2003).\n","['two IsCount', '6 IsRange', 'months IsRange']\n","In these cases, the cells from multiple fetuses were used without HLA matching, but more than 50,000 TH+ cells had survived after several years.\n","['than IsRange', '50 IsRange', ', IsRange', '000 IsRange', 'th IsRange']\n","Our quantitative PCR (qPCR) study in vitro showed that the expression of MHC-I increased in response to IFN-Î³, but the expression level was still 1/10 that of untreated monkey peripheral blood cells (Figure 2B).\n","['/ IsApproximate', '10 IsApproximate']\n","The in vivo studies revealed that the serum level of IFN-Î³ increased at 2 months, and CD45+ cells (including CD8+ cells) accumulated in the allografts 3.5â€“4 months after the transplant.\n","['2 IsRange', '3 IsRange', '. IsRange', '5 IsRange', 'â€“ IsRange', '4 IsRange', 'months IsRange']\n","On the other hand, the levels of INF-Î³ in the cerebrospinal fluid (CSF) and the levels of tumor necrosis factor Î± (TNF-Î±) in both the serum and CSF were below the limit of detection by ELISA (data not shown).\n","[]\n","An immunofluorescence study did not reveal any apparent expression of MHC-I by the grafted cells (Figure S4A).\n","[]\n","Therefore, it is possible that the immune response in the primate brain was not strong enough to reject all of the donor cells.\n","[]\n","These findings closely correlate with the results of previous murine experiments (Hudson et al., 1994; Shinoda et al., 1995).\n","[]\n","To apply our findings to a more clinically relevant setting, we investigated the expression of HLA-I during neural differentiation of human ESCs (hESCs) and iPSCs by qPCR (Figure S4D).\n","[]\n","The expression level was 1/100 compared with that of human peripheral blood cells in both hESCs and iPSCs, and it was similarly elevated in response to IFN-Î³.\n","[]\n","It is difficult to precisely compare immunogenicity in monkeys with that in humans, but the low expression level of MHC-I by the donor cells may account for the mild rejection in both monkey and human neural transplantation.\n","[]\n","We next addressed whether hESC-derived trigeminal neurons can engraft in the adult mouse CNS and project toward their physiological target.\n","[]\n","The trigeminal nuclei in the brainstem receive afferent innervation from the trigeminal sensory ganglion that is relayed to the contralateral thalamus.\n","[]\n","The pons was selected as site for transplantation, because it is surgically accessible and located within proximity of the trigeminal brain stem nuclei that receive afferent input from the trigeminal ganglia.\n","[]\n","Hence, GFP+ human trigeminal neuron clusters were injected into adult NOD/SCID mice via stereotactic surgery (see Experimental Procedures).\n","[]\n","Histological analysis 4 weeks after transplantation showed survival of GFP+ human cell graft in the ventral pons (Figure S5E).\n","['weeks IsRange']\n","Although GFP+ cell bodies remained tightly clustered at injection site, GFP+ fibers showed extensive projections into the host brain (n = 6) including the endogenous trigeminal nuclei (Figure S5F).\n","[]\n","Expression of BRN3A confirmed the sensory neuron identity of the cells (Figure S5G).\n","[]\n","Graft-derived human fiber bundles (hNCAM+ and GFP+) were observed emanating from the graft core (Figure S5H).\n","[]\n","These data demonstrate in vivo survival of trigeminal placode derivatives, differentiation along sensory neuron lineage, and the establishment of axonal projections toward relevant endogenous targets in the embryonic chick and adult mouse brain.\n","[]\n","(B) Fetal hRPE stained for SC121 (red) and MCT1 (green; scale bars, 125 Î¼m and 25 Î¼m [inset]).\n","['##m IsRange']\n","(F) Intracellular PGRN levels in iPSC-derived neurons after medium collection.\n","[]\n","Values of control line 20 were set to 100% (n = 3â€“4 independent cultures).\n","[]\n","Prior to VESTAL analyses, a 5â€“55 Hz bandpass filter was applied.\n","[]\n","VESTAL analyses examined activity 30â€“130 ms post-stimulus producing a 4D activation map (3D volumes across time) as well as a 2D source time-course matrix.\n","['30 IsRange', 'â€“ IsRange', '130 IsRange', 'ms IsRange']\n","The average percent variance explained for gradiometer data using VESTAL was 95.81% for HC and 94.38% for SZ.\n","[]\n","The average percent variance explained for magnetometer data using VESTAL program was 96.24% for HC and 93.17% for SZ.\n","[]\n","There were no group differences in percent variance explained for gradiometer data (t(39) = 1.16, p = 0.25) or magnetometer data (t(39) = 1.31, p = 0.20).\n","[]\n","(C) Control lines 16 and 17, sporadic lines 12 and 23, and PGRN S116X lines 1 and 14 were immunostained for AFP (endoderm), desmin (mesoderm), and Î²III-tubulin (ectoderm), and counterstained with DAPI (nuclei).\n","[]\n","All lines showed a normal karyotype.\n","[]\n","Scale bar: 50 Î¼m.\n","[]\n","(Iâ€“M) Teratoma formation at 3 months after transplantation in the testes of SCID mice.\n","['3 IsRange', 'months IsRange']\n","H&E staining of the sections showed histological features of the neuroepithelium (J), cartilage (K), muscle (L) and gut-like epithelium (M).\n","[]\n","Using a recombinant adeno-associated viral (rAAV)-based gene-targeting method, we inserted the gene-encoding GFP into the SOX2 locus in H9 hESCs (Figure 1A).\n","[]\n","Proper homologous recombination led to the replacement of the SOX2 open reading frame with that of GFP and a neomycin selection cassette (SV40-Neo).\n","[]\n","After infection with rAAV and G418 drug selection, a total of 36 clones were expanded and screened by Southern blotting for homologous recombination events.\n","['36 IsCount']\n","Among these clones, 26 (72%) were found to carry the GFP-Neo cassette in the SOX2 locus (Figure S1A available online).\n","[]\n","No clones in which both SOX2 alleles were disrupted were isolated.\n","[]\n","Our subsequent analysis focused on one of these clones, clone 23 (hSOX2-23).\n","[]\n","We confirmed appropriate gene targeting in this clone using multiple restriction digests followed by Southern blotting (Figures 1B, S1B, and S1C).\n","[]\n","We did not observe nontargeted insertions of the rAAV sequences, and cells exhibited a normal karyotype (data not shown).\n","[]\n","Flow cytometry of hSOX2-23 revealed that the majority of the cells expressed GFP (Figure 1C).\n","[]\n","By comparison, a drug-selected clone, hSOX2-25, which was negative for targeted insertion (Figure S1A), showed no detectable GFP (Figure S2A).\n","[]\n","Despite only having one copy of SOX2, hSOX2-23 had similar levels of SOX2, OCT4, and NANOG expression as hSOX2-25 and wild-type (WT) hESCs (Figure S2B).\n","['one IsCount']\n","Moreover, the percentage of GFP-positive (GFP+) cells in hSOX2-23 was constant over more than 20 passages.\n","['more IsRange', 'than IsRange', '20 IsRange', 'passages IsRange', '. IsRange']\n","Immunofluorescence (IF) staining of hSOX2-23 showed that 100% of GFP+ cells expressed SOX2 protein (Figure S2C).\n","[]\n","Additionally, hSOX2-23 colonies had characteristic hESC morphology (Figure S2D) and expressed markers of the undifferentiated state, such as NANOG (Figure S2E).\n","[]\n","These results show that this rAAV-based gene-targeting method can be used to efficiently disrupt genes by homologous recombination.\n","[]\n","In addition, the SOX2-GFP hESC marker line can be used to monitor SOX2 expression in undifferentiated hESCs.\n","[]\n","Based on several previous publications (Khan et al., 2010, 2011), we explored the utility of adeno-associated virus (AAV) as a method to improve gene targeting efficiencies in hPSCs.\n","[]\n","For SOX2, a gene that is highly expressed in undifferentiated hPSCs, gene targeting rates were greater than 70%.\n","['than IsRange', '70 IsRange', '% IsRange']\n","Similar targeting efficiencies in hPSCs using AAV have been reported by others (Asuri et al., 2012; Khan et al., 2010, 2011; Smith-Arica et al., 2003), indicating that AAV offers a highly efficient and robust approach to target genes for HR in hPSCs.\n","[]\n","(C) C-peptide and PDX1 expression was confirmed by immunocytochemistry of cells differentiated for 25 days.\n","['25 IsRange', 'days IsRange']\n","â€¢Adult hRPESC-derived RPE had comparable in vitro characteristics to fetal hRPEâ€¢hRPE monolayers survived 4 weeks on PET carriers under the rabbit retinaâ€¢Better xenograft survival may be due to the maintained hRPE cell polarityâ€¢Atrophy of the retina overlaying the hRPE xenograft remains a future challenge\n","[]\n","(O) GH plasma levels using a human specific ELISA (6 weeks after transplantation).\n","[]\n","(R) Quantification of immunocytochemical analyses for each iPSC line.\n","[]\n","Data are shown as the means Â± SD (n = 3 independent experiments).\n","[]\n","SER, serotonin, TUBÎ²III, Î²-tubulin class III.\n","[]\n","Scale bars: 200 Î¼m in (A)â€“(H), 50 Î¼m in insets of (C)â€“(F), 100 Î¼m in (J)â€“(M) and (Q).\n","[]\n","The two FTD patients under investigation in this study were part of a longitudinal dementia research program at the Memory and Aging Center, University of California, San Francisco.\n","['two IsCount']\n","Both had an 8-year history of behavioral changes and memory impairment at the time of tissue collection for this study.\n","[]\n","One patient, a 67-year-old male with sporadic FTD, tested negative for mutations in GRN, MAPT, and C9ORF72.\n","[]\n","The other patient, a 64-year-old male with a significant family history of dementia, had behavioral variant FTD.\n","[]\n","MRI in this patient demonstrated severe bifrontal and temporal atrophy associated with gliosis in the frontal lobes (greater on the right).\n","[]\n","One year later, MRI scans showed progression of atrophy and gliosis.\n","[]\n","Genetic testing revealed a novel nonsense mutation in GRN, p.S116X (g.4627C > A, c.347C > A), which is predicted to result in a premature stop codon.\n","[]\n","Both FTD patients had parkinsonism, which is typical of all FTD patients with PGRN mutations.\n","[]\n","An age-matched subject, a clinically normal 64-year-old male with no mutations in GRN, MAPT, or C9ORF72, served as a control.\n","[]\n","(Qâ€“S) Magnetic resonance images of a representative animal (No.\n","[]\n","6, autograft) at 3 months after the transplant.\n","['3 IsRange', 'months IsRange']\n","The arrowheads indicate the directions of the cell injections.\n","[]\n","(Q) coronal, (R) axial, and (S) sagittal.\n","[]\n","The letter L indicates the left side.\n","[]\n","(A) Fetal hRPE stained for pan-cytokeratin (scale bar, 50 Î¼m); inset shows section overview stained with hematoxylin/eosin (scale bar, 200 Î¼m).\n","[]\n","To assess the in vivo properties of hESC-derived trigeminal placode precursors, PIP-induced neuronal clusters, derived from a constitutively GFP-positive hESC line (Figures S5A and S5B), were injected into the developing chick embryo targeting the early trigeminal anlage at H&H stage 10â€“12 (Figure S5C).\n","[]\n","Human cells were identified based on GFP expression and use of human specific antibodies against cytoplasmic antigen (hCA).\n","[]\n","Two days after in ovo transplantation, surviving GFP+ cells were found dispersed in the area of the endogenous chick trigeminal ganglion (Figure 4P).\n","[]\n","We observed extensive GFP+ human fiber bundles coexpressing hCA and peripherin (Figures 4Q and 4R).\n","[]\n","In contrast, no hCA or peripherin expression was detected in the neural tube of the embryo (Figure S5D).\n","[]\n","The in vivo fiber outgrowth 2 days after transplantation was reminiscent of the extensive in vitro fiber outgrowth of replated trigeminal neuron clusters (Figure S5A).\n","['days IsRange']\n","Peripherin expression in vivo (Figure 4S) confirmed the peripheral neuron identity of the grafted cells.\n","[]\n","For the first two animals (Nos.\n","['first IsCount', 'two IsCount']\n","1 and 4), we established iPSCs from fibroblasts derived from the oral mucosa using retroviral vectors (Okita et al., 2011).\n","[]\n","For the other two animals (Nos.\n","['two IsCount']\n","6 and 8), we used peripheral blood mononuclear cells (PBMCs) with nonintegrating episomal vectors (Okita et al., 2013).\n","[]\n","We selected the best clone from each animal according to the following criteria: a stable embryonic stem cell (ESC)-like morphology of the colonies after passaging, expression of pluripotent markers, few or no integrated transgenes (Figures 1Aâ€“1F; Figure S1 available online), and the potential for stable neural differentiation.\n","['following IsCount']\n","A PCR analysis revealed that all of the clones with retroviral vectors showed apparent expression of remaining transgenes (Figures S1C and S1D), whereas the clones with episomal vectors never did (Figure S1F).\n","['all IsCount']\n","To detect the iPSC-derived cells in a brain, we introduced GFP (Figures 1G and 1H).\n","[]\n","The selected clones of iPSCs had the potential to generate teratomas in the testes of a severe combined immunodeficiency (SCID) mouse within 12 weeks (Figures 1Iâ€“1M).\n","['12 IsRange']\n","(A) Flow-cytometric analyses for MHC-I (HLA-A, HLA-B, and HLA-C).\n","[]\n","Incubation of the cells with IFN-Î³ for 48 hr increased the MHC-I expression (green).\n","[]\n","Our group has developed a defined culture system to direct the differentiation of hPSCs into a near-homogenous population of definitive endoderm (DE) cells that have the capacity to differentiate into hepatocytes and pancreatic progenitors (Brown et al., 2011; Cho et al., 2012; Rashid et al., 2010; Touboul et al., 2010; Vallier et al., 2009a; Yusa et al., 2011).\n","[]\n","Cells grown in these culture conditions successively express primitive streak markers (T and Mixl1), downregulate pluripotency markers (NANOG, SOX2, and POU5F1) and progressively upregulate definitive endoderm markers (CXCR4, FOXA2, GATA4, CERB, and SOX17) (Figures S1Aâ€“S1C available online).\n","[]\n","Flow cytometry analyses showed that 80% of the resulting DE population coexpresses CXCR4 and SOX17 (Figure S1D).\n","['% IsApproximate']\n","Interestingly, the resulting population of DE cells is negative for genes marking the foregut (SOX2), the midgut/hindgut (CDX2), the pancreas (PDX1), the liver (AFP), and the lungs (HOXA1) (Figures S1E and S1F) This confirms that DE cells generated in vitro could correspond to early endoderm progenitor cells prior to anteroposterior patterning or organogenesis.\n","[]\n","In sequential PET studies, we observed increased uptake of [11C]PK11195 in one allograft (animal No.\n","[]\n","10) at 3 months (Figures 3A and 3B).\n","[]\n","We could not detect any apparent uptake in the other animals or at any other time points (Figure S2).\n","[]\n","Intriguingly, the serum level of IFN-Î³ temporarily increased at 2 months after the transplant in three animals (Figure 3C).\n","['2 IsRange', 'months IsRange', 'three IsCount']\n","An immunofluorescence study conducted at 3.5â€“4 months showed that MHC-II+ cells were more frequently found in allografts than in autografts, especially in the monkey with increased uptake of [11C]PK11195 (Figure 3D, No.\n","['3 IsRange', '. IsRange', '5 IsRange', 'â€“ IsRange', '4 IsRange', 'months IsRange']\n","10).\n","[]\n","The MHC-II staining never overlapped with that of GFP of the donor cells (Figure 3F), whereas it generally overlapped with that of IBA1 (Figure 3G), indicating that MHC-II was expressed by host-derived microglia.\n","[]\n","Consistently, the number and density of IBA1+ cells were higher in allografts than in autografts (Figures 3E, 3H, and S4C).\n","[]\n","An increase in the expression of MHC might trigger the recruitment of circulating immune cells, including T cells.\n","[]\n","An immunofluorescence study revealed that more CD45+ cells (a marker for pan-leukocytes) accumulated in allografts compared with autografts (Figures 3I and 3J).\n","[]\n","Most of the CD45+ cells were CD3+ T cells, and 60% of them were CD8+ killer T cells (Figures 3K and 3L).\n","[]\n","These findings suggest that an acquired immune response was elicited only in the allografts in the primate brain.\n","[]\n","This study was approved by the Institutional Review Board and Ethics Committees of the University of California, San Francisco, and written informed consent was obtained in all cases.\n","[]\n","The patient with the PGRN S116X mutation followed the classic clinical progression for FTD and developed parkinsonism, as do all FTD patients with PGRN mutations, but he did not show typical features of PD dementia.\n","[]\n","The patient with sporadic FTD also showed parkinsonism.\n","[]\n","Skin biopsies were collected, cut into small pieces, and placed on culture dishes to allow the fibroblasts to expand.\n","[]\n","The cells were maintained in Dulbecco's modified Eagle's medium supplemented with 10% fetal bovine serum, 1X nonessential amino acids, and penicillin/streptomycin (100 U/ml).\n","[]\n","iPSCs were generated as described previously (Takahashi et al., 2007).\n","[]\n","Please see Supplemental Information for more details.\n","[]\n","(Dâ€“H) Adult hRPE stained for SC121 (red).\n","[]\n","(D) Adult hRPE stained for MCT1 (green).\n","[]\n","(E) Human adult RPE stained for ezrin (green).\n","[]\n","hRPEs transplanted into rabbit SRS show absence of expression of ki67 (F), phosphohistone H3 (G), and caspase-3 (H).\n","[]\n","Polarized fetal and adult hRPE cells were found in TEM (I and J).\n","[]\n","Nuclei with regular chromatin were found in the basal compartment, a basal lamina ([I], large black arrowhead) had formed between the xenograft and PET carrier (black asterisks).\n","[]\n","Melanosomes (M) in multiple stages, some microvilli abutting to the atrophic neural retina (NR), and junctional structures with desmosomes (small black arrowhead) and tight junctions (red arrowhead) were discerned apically.\n","[]\n","Mitochondria (MC) were seen in the basolateral part of the cell.\n","[]\n","Detachment from cell carrier (asterisk) in (J) is a histologic processing artifact.\n","[]\n","Left images in (I) and (J) taken at 10,500Ã— magnification; right micrographs are rectangular zone in left at 25,000Ã—; scale bars represent 2 Î¼m/inset 0.2 Î¼m distance in (I) and (J).\n","[', IsList', '500 IsList', '000 IsList']\n","iPSC colonies were detached with accutase (Millipore) and grown as embryoid bodies (EBs) in suspension for 5â€“6 days in iPSC medium without basic fibroblast growth factor.\n","['â€“ IsRange', '6 IsRange']\n","EBs were allowed to attach and form rosettes.\n","[]\n","Ten-day-old rosettes were collected and grown in suspension as neurospheres.\n","[]\n","Neurospheres were dissociated after 3â€“4 weeks, and the cells were plated on glass coverslips (BD Biosciences) or plates coated with poly-D-lysine (0.1 mg/ml) and laminin (10 Î¼g/m).\n","['3 IsRange', 'â€“ IsRange', '4 IsRange', 'weeks IsRange', ', IsRange']\n","Neurons were used after 2-4 weeks in culture.\n","['2 IsRange', '- IsRange', '4 IsRange', 'weeks IsRange']\n","For cell differentiation, FAP-specific iPS cells were cultured with serial changes of media as shown in Fig.\n","[]\n","3A.\n","[]\n","To test whether FAP-specific iPS cells can differentiate into hepatocyte-like cells, we analyzed several markers via real-time PCR analysis on Day 5 (D5), D13 and D20 differentiated FAP-specific iPS cells (Fig.\n","[]\n","3B).\n","[]\n","A decrease in expression of the pluripotency marker Oct3/4 was accompanied by differentiation of FAP-specific iPS cells.\n","[]\n","Expression of the endoderm marker Sox17 was observed on D5 differentiation and decreased gradually after the medium was changed to hepatic differentiation medium on D7 (Fig.\n","[]\n","3B).\n","[]\n","The hepatic progenitor marker AFP and the mature hepatocyte marker ALB were obviously expressed on D13 and D20.\n","[]\n","In addition, immunocytochemical analyses showed Sox17 expression on D5, both HNF-4Î± and AFP expression on D13, and ALB cytoplasmic staining on D20 (Fig.\n","[]\n","3C).\n","[]\n","Quantitative imaging analysis revealed that approximately 78 Â± 0.6% of cells were Sox17-positive on D5 and approximately 88 Â± 1.1% of cells were AFP-positive on D13 and approximately 29 Â± 0.9% of cells were ALB-positive on D20 (Fig.\n","['approximately IsApproximate', '78 HasTolerance', 'Â± HasTolerance', '0 HasTolerance', '. HasTolerance', '6 HasTolerance', '% HasTolerance', '88 HasTolerance', 'Â± HasTolerance', 'Â± HasTolerance']\n","3D).\n","[]\n","The ALB secretion in the media of differentiated FAP-specific iPS cells on D20 was approximately 20 Î¼g/ml (Fig.\n","['approximately IsApproximate', '20 IsApproximate', 'Î¼ IsApproximate', '##g IsApproximate', '/ IsApproximate', 'ml IsApproximate']\n","3E).\n","[]\n","Moreover, these D20 differentiated FAP-specific iPS cells were also periodic acid-Schiff (PAS)-positive, indicating cytoplasmic glycogen storage (Fig.\n","[]\n","3F).\n","[]\n","These results clearly indicated that FAP-specific iPS cells had the potential to differentiate into hepatocyte-like cells.\n","[]\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"3onHFGI7JGNc"},"source":["Unit Detection"]},{"cell_type":"code","metadata":{"id":"zfTglERDIO9R"},"source":["file = open(VOCAB_PATH,'rb')\r\n","TEXT = pickle.load(file)\r\n","vocab_size=len(TEXT.vocab)\r\n","class bilstm(nn.Module):\r\n","    def __init__(self, vocab_size, output_size, embedding_dim, hidden_dim, n_layers):\r\n","        super().__init__()\r\n","        self.hidden_dim = hidden_dim\r\n","        self.n_layers = n_layers\r\n","        self.output_size = output_size\r\n","        \r\n","        self.embedding = nn.Embedding(vocab_size,embedding_dim)\r\n","        self.lstm = nn.LSTM(embedding_dim, hidden_dim, n_layers, \r\n","                         batch_first=True, bidirectional=True)\r\n","        \r\n","        self.dropout = nn.Dropout(0)\r\n","        self.fc = nn.Linear(hidden_dim*2,output_size)\r\n","        self.sigmoid = nn.Sigmoid()\r\n","        \r\n","    def forward(self,x):\r\n","        embeds = self.embedding(x)\r\n","        lstm, (hn,cn)= self.lstm(embeds)\r\n","        \r\n","        output = self.dropout(lstm)\r\n","        output = self.fc(output)\r\n","        output = self.sigmoid(output) \r\n","        return output\r\n","device = \"cpu\"\r\n","model = bilstm(vocab_size, 1, 100, 32, 1)\r\n","model = model.to(device)"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"H2bnUdAYJLzB"},"source":["model = torch.load(UNIT_DETECTION_PATH)"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"j8hvkoWfJRPv","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1608028102716,"user_tz":-330,"elapsed":2308,"user":{"displayName":"Akash Gangwar","photoUrl":"","userId":"02081727011117790241"}},"outputId":"588a838c-1a0c-42e7-813e-3238b9482c60"},"source":["UNITS=[]\r\n","\r\n","for i in Quantity:\r\n","  unit=[]\r\n","  for j in i:\r\n","    X_train=[]\r\n","    X_train.append(TEXT.process(j)[0].numpy())\r\n","    \r\n","    X_train=np.array(X_train)\r\n","\r\n","    \r\n","    X_train=torch.from_numpy(X_train)\r\n","    X_train=X_train[:,:,0]\r\n","\r\n","    \r\n","    row=X_train.shape[1]\r\n","    \r\n","    \r\n","    X_train=F.pad(input=X_train, pad=(0, 64-row, 0, 0), mode='constant', value=1)\r\n","    \r\n","    model.zero_grad()\r\n","    model.eval()\r\n","    y_pred = model(X_train.to(device))\r\n","    \r\n","    np_out = y_pred.cpu().data.numpy()\r\n","    \r\n","    np_out=np_out>=0.75\r\n","    un=\"\"\r\n","    for k in range(len(np_out[0])):\r\n","      if(np_out[0][k]):\r\n","        un=un+j[k]\r\n","    if(len(un)==0):\r\n","      print(\"Quantity-\"+j,\"Unit-None\")\r\n","    else:\r\n","      print(\"Quantity-\"+j,\"Unit-\"+un)\r\n","    unit.append(un)\r\n","\r\n","  UNITS.append(unit)"],"execution_count":null,"outputs":[{"output_type":"stream","text":["Quantity-10 Î¼m. Unit-Î¼m\n","Quantity-12 hr Unit-hr\n","Quantity-96 Unit-None\n","Quantity-5 mm per side Unit-mmper side\n","Quantity-two Unit-None\n","Quantity-after 6 months Unit-months\n","Quantity-more than 50,000 Unit-None\n","Quantity-1/10 that Unit-h\n","Quantity-2 months, Unit-months\n","Quantity-3.5â€“4 months Unit-months\n","Quantity-1/100 Unit-None\n","Quantity-4 weeks Unit-weeks\n","Quantity-6 Unit-None\n","Quantity-125 Î¼m Unit-Î¼m\n","Quantity-25 Î¼m Unit-Î¼m\n","Quantity-100% Unit-%\n","Quantity-3â€“4 Unit-None\n","Quantity-5â€“55 Unit-None\n","Quantity-30â€“130 ms Unit-ms\n","Quantity-4d Unit-None\n","Quantity-95.81% Unit-%\n","Quantity-94.38% Unit-%\n","Quantity-96.24% Unit-%\n","Quantity-93.17% Unit-%\n","Quantity-1.16, Unit-None\n","Quantity-0.25 Unit-None\n","Quantity-1.31, Unit-None\n","Quantity-50 Î¼m. Unit-Î¼m\n","Quantity-3 months Unit-months\n","Quantity-36 Unit-None\n","Quantity-26 Unit-None\n","Quantity-one Unit-None\n","Quantity-more than 20 passages. Unit-passages\n","Quantity-100% Unit-%\n","Quantity-greater than 70%. Unit-%\n","Quantity-25 days. Unit-days\n","Quantity-4 weeks Unit-weeks\n","Quantity-6 weeks Unit-eeks\n","Quantity-3 Unit-None\n","Quantity-200 Î¼m Unit-Î¼m\n","Quantity-50 Î¼m Unit-Î¼m\n","Quantity-100 Î¼m Unit-Î¼m\n","Quantity-two Unit-None\n","Quantity-8-year Unit-year\n","Quantity-67-year- Unit-year-\n","Quantity-64-year- Unit-year-\n","Quantity-64-year- Unit-year-\n","Quantity-3 months Unit-months\n","Quantity-50 Î¼m Unit-Î¼m\n","Quantity-200 Î¼m Unit-Î¼m\n","Quantity-2 days Unit-days\n","Quantity-two Unit-None\n","Quantity-two Unit-None\n","Quantity-12 weeks Unit-weeks\n","Quantity-48 hr Unit-hr\n","Quantity-80% Unit-%\n","Quantity-3 months Unit-months\n","Quantity-at 2 months Unit-months\n","Quantity-three Unit-None\n","Quantity-3.5â€“4 months Unit-months\n","Quantity-60% Unit-%\n","Quantity-10% Unit-%\n","Quantity-1x Unit-None\n","Quantity-10,500Ã— Unit-None\n","Quantity-25,000Ã— Unit-None\n","Quantity-2 Î¼m/ Unit-Î¼m/\n","Quantity-0.2 Î¼m Unit-Î¼m\n","Quantity-5â€“6 days Unit-days\n","Quantity-ten- Unit-None\n","Quantity-3â€“4 weeks, Unit-weeks\n","Quantity-0.1 mg/ml Unit-mg/ml\n","Quantity-10 Î¼g Unit-Î¼g\n","Quantity-after 2-4 weeks Unit-eeks\n","Quantity-approximately 78 Â± 0.6% Unit-%\n","Quantity-approximately 88 Â± 1.1% Unit-%\n","Quantity-approximately 29 Â± 0.9% Unit-%\n","Quantity-approximately 20 Î¼g/ml Unit-Î¼g/ml\n","Quantity-+ Unit-None\n","Quantity-survived after Unit-None\n","Quantity-after the transplant Unit-None\n","Quantity-for Unit-None\n","Quantity-for Unit-None\n","Quantity-z Unit-None\n","Quantity-ometer Unit-None\n","Quantity-vestal Unit-None\n","Quantity-for Unit-None\n","Quantity-for Unit-None\n","Quantity-z Unit-None\n","Quantity-ometer Unit-None\n","Quantity-after transplantation in the test Unit-None\n","Quantity-sci Unit-None\n","Quantity-20 passages Unit-passages\n","Quantity-plasma Unit-p\n","Quantity-weeks after transplantation Unit-None\n","Quantity-at Unit-None\n","Quantity-months after the transplant. Unit-None\n","Quantity-all Unit-None\n","Quantity-t Unit-None\n","Quantity-n- Unit-None\n","Quantity-temporarily increased at Unit-None\n","Quantity-months after the transplant in Unit-None\n","Quantity-weeks in culture Unit-ltre\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"VpK5Bmr7PYLv"},"source":["Relation Extraction"]},{"cell_type":"code","metadata":{"id":"5okh0jtzPebT"},"source":["device = torch.device(\"cuda\" if torch.cuda.is_available() else \"cpu\")"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"GJD1YZB3PrNS"},"source":["def tokenize_and_preserve_labels(sentence, text_labels, tokenizer):\r\n","\r\n","    \"\"\"\r\n","    Word piece tokenization makes it difficult to match word labels\r\n","    back up with individual word pieces. This function tokenizes each\r\n","    word one at a time so that it is easier to preserve the correct\r\n","    label for each subword. It is, of course, a bit slower in processing\r\n","    time, but it will help our model achieve higher accuracy.\r\n","    \"\"\"\r\n","    # print(len(sentence))\r\n","    # print(len(text_labels))\r\n","    tokenized_sentence = []\r\n","    labels = []\r\n","\r\n","    for word, label in zip(sentence, text_labels):\r\n","\r\n","        # Tokenize the word and count # of subwords the word is broken into\r\n","        tokenized_word = tokenizer.tokenize(word)\r\n","        n_subwords = len(tokenized_word)\r\n","\r\n","        # Add the tokenized word to the final tokenized word list\r\n","        tokenized_sentence.extend(tokenized_word)\r\n","\r\n","        # Add the same label to the new list of labels `n_subwords` times\r\n","        labels.extend([label] * n_subwords)\r\n","    # print(len(tokenized_sentence))\r\n","    # print(len(labels))\r\n","    return tokenized_sentence, labels"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"8PQTlX1HKGe2"},"source":["lis_en = []\r\n","tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')\r\n","model = BertModel.from_pretrained('bert-base-uncased')\r\n","for i in range(len(test_text)):\r\n","  tag={}\r\n","  embed={} \r\n","  for j in range(len(Quantity[i])):\r\n","    L=test_text[i].lower()\r\n","    if L.find(Quantity[i][j]):\r\n","      start=[m.start() for m in re.finditer(Quantity[i][j],L)]\r\n","      if(len(start)>0):\r\n","        start_q = start[0]\r\n","        end_q = len(Quantity[i][j]) + start_q - 1\r\n","        tag[\"Q-\"+str(j+1)]=[start_q,end_q,Quantity[i][j]]\r\n","        embed[\"Q-\"+str(j+1)]=np.zeros(768) \r\n","  for j in range(len(Property[i])):\r\n","    test_text[i]=test_text[i].lower()\r\n","    if test_text[i].find(Property[i][j]):\r\n","      \r\n","      if(Property[i][j] in (\"!\", \",\" ,\"\\'\" ,\";\" ,\"\\\"\", \".\", \"-\" ,\"?\",\")\",\"+\",\"-\",\"*\",\"/\",\"%\")):\r\n","        continue\r\n","      start=[m.start() for m in re.finditer(Property[i][j],test_text[i])]\r\n","      \r\n","      if(len(start)>0):\r\n","        start_q = start[0]\r\n","        end_q = len(Property[i][j]) + start_q - 1\r\n","        tag[\"P-\"+str(j+1)]=[start_q,end_q,Property[i][j]]\r\n","        embed[\"P-\"+str(j+1)]=np.zeros(768)\r\n","  for j in range(len(Entity[i])):\r\n","    \r\n","    if test_text[i].find(Entity[i][j]):\r\n","      if(Entity[i][j] in (\"!\", \",\" ,\"\\'\" ,\";\" ,'(',\"\\\"\", \".\", \"-\" ,\"?\",\")\",\"+\",\"-\",\"*\",\"/\",\"%\")):\r\n","        continue\r\n","      \r\n","      start=[m.start() for m in re.finditer(Entity[i][j],test_text[i])]\r\n","      # print(start,test_text[i],Entity[i][j])\r\n","      if(len(start)>0):\r\n","        start_q = start[0]\r\n","        end_q = len(Entity[i][j]) + start_q - 1\r\n","        tag[\"E-\"+str(j+1)]=[start_q,end_q,Entity[i][j]]\r\n","        embed[\"E-\"+str(j+1)]=np.zeros(768)\r\n","\r\n","  \r\n","  \r\n","  j=test_text[i]\r\n","  ret = [(m.group(0), m.start(), m.end() - 1) for m in re.finditer(r'\\S+', j)]\r\n","  entities=[]\r\n","  for word in ret:\r\n","    temp=[]\r\n","    for key,ent_doc in tag.items():\r\n","      if(word[1]>=ent_doc[0] and word[2]<=ent_doc[1]):\r\n","        temp.append(key)\r\n","    entities.append(temp)\r\n","  \r\n","\r\n","  sen,label=tokenize_and_preserve_labels(j.split(' '),entities,tokenizer)\r\n","  label.insert(0,[])\r\n","  label.append([])\r\n","  inputs = tokenizer(j, return_tensors=\"pt\")\r\n","  outputs = model(**inputs)\r\n","  \r\n","  last_hidden_states = outputs[0]\r\n","\r\n","  for i in range(len(label)):\r\n","    if(len(label[i])==0):\r\n","      continue\r\n","    else:\r\n","      for k in range(len(label[i])):\r\n","        embed[label[i][k]]=embed[label[i][k]]+last_hidden_states[0][i].detach().numpy()\r\n","  \r\n","  data=[]\r\n","\r\n","  for key,val in tag.items():\r\n","    if(key[0]=='Q'):\r\n","      data.append([\"Quantity\",embed[key],val[2]])\r\n","    elif(key[0]=='P'):\r\n","      data.append([\"MeasuredProperty\",embed[key],val[2]])\r\n","    else:\r\n","      data.append([\"MeasuredEntity\",embed[key],val[2]])    \r\n","  lis_en.append(data)\r\n","  "],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"RUuDc0WAQpy1"},"source":["import tensorflow as tf\r\n","from tensorflow import keras\r\n","from tensorflow.keras import layers\r\n","import numpy as np\r\n"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"VHDaAedSQv16"},"source":["inputA = layers.Input(shape=(768,))\r\n","inputB = layers.Input(shape=(768,))\r\n","inputC = layers.Input(shape=(1,))\r\n","inputD = layers.Input(shape = (1,))\r\n","\r\n","x = layers.Dense(128, activation=\"relu\")(inputA)\r\n","x = layers.Dense(64, activation=\"relu\")(x)\r\n","\r\n","y = layers.Dense(128, activation=\"relu\")(inputB)\r\n","y = layers.Dense(64, activation=\"relu\")(y)\r\n","\r\n","c = layers.Dense(1, activation=\"relu\")(inputC)\r\n","\r\n","d = layers.Dense(1, activation=\"relu\")(inputD)\r\n","\r\n","\r\n","com = layers.concatenate([x, y, c, d])\r\n","z = layers.Dense(32, activation=\"relu\")(com)\r\n","z = layers.Dense(4,  activation='softmax')(z)\r\n","model = keras.Model(inputs=[inputA, inputB, inputC, inputD], outputs=z)"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"UPD1n7G7QxE-"},"source":["weights = [15.593220338983052, 10.415094339622641, 33.65853658536585, 2.323232323232323]"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"DSLnwk6qQz71"},"source":["from keras import backend as K\r\n","def weighted_categorical_crossentropy(weights):\r\n","\r\n","    weights = K.variable(weights)\r\n","        \r\n","    def loss(y_true, y_pred):\r\n","\r\n","        y_pred /= K.sum(y_pred, axis=-1, keepdims=True)\r\n","\r\n","        y_pred = K.clip(y_pred, K.epsilon(), 1 - K.epsilon())\r\n","\r\n","        loss = y_true * K.log(y_pred) * weights\r\n","        loss = -K.sum(loss, -1)\r\n","        return loss\r\n","    \r\n","    return loss"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"waFJwVNMQ3Kv"},"source":["model.compile(optimizer='adam',\r\n","              loss= weighted_categorical_crossentropy(weights),\r\n","              metrics=['accuracy'])"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"t8rlNg3IQ6HZ"},"source":["from keras.models import load_model\r\n","model.load_weights('/content/drive/My Drive/Model/rel_wt_new.h5')"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"nb2NpDFFPlNq"},"source":["idx2tok = {}\r\n","idx2tok[\"Quantity\"] = 1\r\n","idx2tok[\"MeasuredEntity\"] = 2\r\n","idx2tok[\"MeasuredProperty\"] = 3\r\n","idx2tok[\"Qualifier\"] = 4"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"RnKIRh1mQZZM"},"source":["tok2idx = {}\r\n","tok2idx[0] = \"HasProperty\"\r\n","tok2idx[1] = \"HasQuantity\"\r\n","tok2idx[2] = \"Qualifies\"\r\n","tok2idx[3] = \"norelation\""],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"N3lpRs3FQcLR"},"source":["relation = []\r\n","for sen in lis_en:\r\n","  ans = []\r\n","  for i in range(len(sen)):\r\n","    for j in range(i+1, len(sen)):\r\n","      x1 = sen[i][1].reshape([1,768])\r\n","      x2 = sen[j][1].reshape([1,768])\r\n","      x3 = np.array([idx2tok[sen[i][0]]]).reshape([1,1])\r\n","      x4 = np.array([idx2tok[sen[j][0]]]).reshape([1,1])\r\n","      \r\n","      y_pred = model.predict([x1,x2,x3,x4])\r\n","      rel_pred = np.argmax(y_pred)\r\n","      if x3[0,0] == x4[0,0]:\r\n","        continue\r\n","      if tok2idx[rel_pred]==\"norelation\":\r\n","        continue\r\n","      if x3[0,0]==1 and x4[0,0]==2 and tok2idx[rel_pred]==\"HasQuantity\":\r\n","        ans.append([sen[j][2], sen[i][2], x4[0,0],x3[0,0], tok2idx[rel_pred]])\r\n","      if x3[0,0]==2 and x4[0,0]==1 and tok2idx[rel_pred]==\"HasQuantity\":\r\n","        ans.append([sen[i][2], sen[j][2],x3[0,0],x4[0,0], tok2idx[rel_pred]])\r\n","      if x3[0,0]==3 and x4[0,0]==2 and tok2idx[rel_pred]==\"HasProperty\":\r\n","        ans.append([sen[j][2], sen[i][2], x4[0,0],x3[0,0], tok2idx[rel_pred]])\r\n","      if x3[0,0]==2 and x4[0,0]==3 and tok2idx[rel_pred]==\"HasProperty\":\r\n","        ans.append([sen[i][2], sen[j][2],x3[0,0],x4[0,0], tok2idx[rel_pred]])\r\n","      if x3[0,0]==1 and x4[0,0]==4 and tok2idx[rel_pred]==\"Qualifies\":\r\n","        ans.append([sen[j][2], sen[i][2], x4[0,0],x3[0,0], tok2idx[rel_pred]])\r\n","      if x3[0,0]==4 and x4[0,0]==1 and tok2idx[rel_pred]==\"Qualifies\":\r\n","        ans.append([sen[i][2], sen[j][2],x3[0,0],x4[0,0], tok2idx[rel_pred]])\r\n","      \r\n","  \r\n","  relation.append(ans)"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"yA_KZdBSRF3c","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1608030701925,"user_tz":-330,"elapsed":1670,"user":{"displayName":"Akash Gangwar","photoUrl":"","userId":"02081727011117790241"}},"outputId":"12a93ec3-91e9-46f2-fc6c-feda6517e7c1"},"source":[""],"execution_count":null,"outputs":[{"output_type":"execute_result","data":{"text/plain":["[[],\n"," [],\n"," [],\n"," [],\n"," [['scale bars', '10 Î¼m.', 2, 1, 'HasQuantity'],\n","  ['scale bars', 'scale bars', 2, 3, 'HasProperty']],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [['regional masks', '5 mm per side', 2, 1, 'HasQuantity'],\n","  ['source grid', '5 mm per side', 2, 1, 'HasQuantity'],\n","  ['voxels', '5 mm per side', 2, 1, 'HasQuantity'],\n","  ['side', '5 mm per side', 2, 1, 'HasQuantity'],\n","  ['regional masks', 'grid with voxels', 2, 3, 'HasProperty'],\n","  ['source grid', 'grid with voxels', 2, 3, 'HasProperty'],\n","  ['voxels', 'grid with voxels', 2, 3, 'HasProperty'],\n","  ['side', 'grid with voxels', 2, 3, 'HasProperty']],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [['double-blind clinical trials', 'after 6 months', 2, 1, 'HasQuantity'],\n","  ['double-blind clinical trials', 'used', 2, 3, 'HasProperty'],\n","  ['immunosuppressive', 'used', 2, 3, 'HasProperty'],\n","  ['double-blind clinical trials', 'withdrawn', 2, 3, 'HasProperty'],\n","  ['immunosuppressive', 'withdrawn', 2, 3, 'HasProperty']],\n"," [],\n"," [['untreated monkey peripheral blood cells',\n","   '1/10 that',\n","   2,\n","   1,\n","   'HasQuantity'],\n","  ['untreated monkey peripheral blood cells',\n","   'expression',\n","   2,\n","   3,\n","   'HasProperty'],\n","  ['untreated monkey peripheral blood cells',\n","   'increased',\n","   2,\n","   3,\n","   'HasProperty'],\n","  ['untreated monkey peripheral blood cells', 'response', 2, 3, 'HasProperty'],\n","  ['untreated monkey peripheral blood cells',\n","   'expression level',\n","   2,\n","   3,\n","   'HasProperty']],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [['expression', '1/100', 2, 1, 'HasQuantity'],\n","  ['peripheral blood cells', '1/100', 2, 1, 'HasQuantity']],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [['control line 20', '100%', 2, 1, 'HasQuantity'],\n","  ['cultures', '100%', 2, 1, 'HasQuantity'],\n","  ['control line 20', '3â€“4', 2, 1, 'HasQuantity'],\n","  ['cultures', '3â€“4', 2, 1, 'HasQuantity']],\n"," [['pass filter', '5â€“55', 2, 1, 'HasQuantity']],\n"," [['volumes', '30â€“130 ms', 2, 1, 'HasQuantity'],\n","  ['volumes', '4d', 2, 1, 'HasQuantity']],\n"," [['HC', '95.81%', 2, 1, 'HasQuantity'],\n","  ['SZ', '95.81%', 2, 1, 'HasQuantity'],\n","  ['HC', '94.38%', 2, 1, 'HasQuantity'],\n","  ['SZ', '94.38%', 2, 1, 'HasQuantity']],\n"," [],\n"," [['no group differences', '1.16,', 2, 1, 'HasQuantity'],\n","  ['percent variance explained', '1.16,', 2, 1, 'HasQuantity'],\n","  ['grad', '1.16,', 2, 1, 'HasQuantity'],\n","  ['39', '1.16,', 2, 1, 'HasQuantity'],\n","  ['magnetometer data', '1.16,', 2, 1, 'HasQuantity'],\n","  ['no group differences', '1.31,', 2, 1, 'HasQuantity'],\n","  ['percent variance explained', '1.31,', 2, 1, 'HasQuantity'],\n","  ['grad', '1.31,', 2, 1, 'HasQuantity'],\n","  ['39', '1.31,', 2, 1, 'HasQuantity'],\n","  ['magnetometer data', '1.31,', 2, 1, 'HasQuantity']],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [['36 clones', '36', 2, 1, 'HasQuantity']],\n"," [['clones', '26', 2, 1, 'HasQuantity'],\n","  ['clones', 'found', 2, 3, 'HasProperty'],\n","  ['clones', 'cassette', 2, 3, 'HasProperty']],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [['unofluores', '100%', 2, 1, 'HasQuantity']],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [['3 independent experiments', '3', 2, 1, 'HasQuantity']],\n"," [],\n"," [['bars', '200 Î¼m', 2, 1, 'HasQuantity'],\n","  ['insets of', '200 Î¼m', 2, 1, 'HasQuantity'],\n","  ['bars', '50 Î¼m', 2, 1, 'HasQuantity'],\n","  ['insets of', '50 Î¼m', 2, 1, 'HasQuantity'],\n","  ['bars', '100 Î¼m', 2, 1, 'HasQuantity'],\n","  ['insets of', '100 Î¼m', 2, 1, 'HasQuantity']],\n"," [],\n"," [],\n"," [['patient,', 'tested negative', 2, 3, 'HasProperty']],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [['deficiency', '12 weeks', 2, 1, 'HasQuantity']],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [['resulting DE population', '80%', 2, 1, 'HasQuantity']],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [['animals', 'at 2 months', 2, 1, 'HasQuantity'],\n","  ['animals', 'increased', 2, 3, 'HasProperty']],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [['cells', '10%', 2, 1, 'HasQuantity'],\n","  ['nonessential amino acids,', '10%', 2, 1, 'HasQuantity'],\n","  ['ici', '10%', 2, 1, 'HasQuantity'],\n","  ['cells', '1x', 2, 1, 'HasQuantity'],\n","  ['etal bovine serum', '1x', 2, 1, 'HasQuantity'],\n","  ['nonessential amino acids,', '1x', 2, 1, 'HasQuantity'],\n","  ['ici', '1x', 2, 1, 'HasQuantity'],\n","  ['cells', 'ssential amino acids, and penicillin/stre', 2, 3, 'HasProperty'],\n","  ['etal bovine serum',\n","   'ssential amino acids, and penicillin/stre',\n","   2,\n","   3,\n","   'HasProperty'],\n","  ['nonessential amino acids,',\n","   'ssential amino acids, and penicillin/stre',\n","   2,\n","   3,\n","   'HasProperty'],\n","  ['ici', 'ssential amino acids, and penicillin/stre', 2, 3, 'HasProperty']],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [['/inset', '10,500Ã—', 2, 1, 'HasQuantity'],\n","  ['/inset', '0.2 Î¼m', 2, 1, 'HasQuantity'],\n","  ['right micrographs', 'distance', 2, 3, 'HasProperty'],\n","  ['/inset', 'distance', 2, 3, 'HasProperty']],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [['uantitative imaging analysis',\n","   'approximately 78 Â± 0.6%',\n","   2,\n","   1,\n","   'HasQuantity'],\n","  ['17', 'approximately 78 Â± 0.6%', 2, 1, 'HasQuantity'],\n","  ['uantitative imaging analysis',\n","   'approximately 88 Â± 1.1%',\n","   2,\n","   1,\n","   'HasQuantity'],\n","  ['cells', 'approximately 88 Â± 1.1%', 2, 1, 'HasQuantity'],\n","  ['17', 'approximately 88 Â± 1.1%', 2, 1, 'HasQuantity'],\n","  ['cells', 'approximately 88 Â± 1.1%', 2, 1, 'HasQuantity'],\n","  ['cells', 'approximately 88 Â± 1.1%', 2, 1, 'HasQuantity'],\n","  ['uantitative imaging analysis',\n","   'approximately 29 Â± 0.9%',\n","   2,\n","   1,\n","   'HasQuantity'],\n","  ['cells', 'approximately 29 Â± 0.9%', 2, 1, 'HasQuantity'],\n","  ['17', 'approximately 29 Â± 0.9%', 2, 1, 'HasQuantity'],\n","  ['cells', 'approximately 29 Â± 0.9%', 2, 1, 'HasQuantity'],\n","  ['cells', 'approximately 29 Â± 0.9%', 2, 1, 'HasQuantity']],\n"," [],\n"," [],\n"," [],\n"," [],\n"," [],\n"," []]"]},"metadata":{"tags":[]},"execution_count":71}]}]}
